<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>sSNAPPY: Singel Sample directioNAl Pathway Perturbation analYsis • sSNAPPY</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/Sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="sSNAPPY: Singel Sample directioNAl Pathway Perturbation analYsis">
<meta property="og:description" content="sSNAPPY">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sSNAPPY</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/sSNAPPY.html">Get Started</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="sSNAPPY_files/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="sSNAPPY_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="sSNAPPY_files/datatables-binding-0.21/datatables.js"></script><link href="sSNAPPY_files/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="sSNAPPY_files/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="sSNAPPY_files/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="sSNAPPY_files/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="sSNAPPY_files/nouislider-7.0.10/jquery.nouislider.min.js"></script><link href="sSNAPPY_files/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="sSNAPPY_files/selectize-0.12.0/selectize.min.js"></script><link href="sSNAPPY_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="sSNAPPY_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>sSNAPPY: Singel Sample directioNAl Pathway
Perturbation analYsis</h1>
                        <h4 data-toc-skip class="author">Wenjun Nora
Liu</h4>
            <address class="author_afil">
      Dame Roma Mitchell Cancer Research Laboratories, Adelaide Medical
School, University of
Adelaide<br><a class="author_email" href="mailto:#"></a><a href="mailto:wenjun.liu@adelaide.edu.au" class="email">wenjun.liu@adelaide.edu.au</a>
      </address>
                  
      
      
      <div class="hidden name"><code>sSNAPPY.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette demonstrates how to use the package
<code>sSNAPPY</code> to compute directional single sample pathway
perturbation scores by incorporating pathway topologies, utilize sample
permutation to test the significance of individual scores and compare
average pathway activities across treatments.</p>
<p>The package also provides a function to visualize overlap between
pathway genes contained in perturbed biological pathways as network
plots.</p>
</div>
<div class="section level2">
<h2 id="install-ssnappy-package-from-bioconductor">Install <code>sSNAPPY</code> package from
<code>Bioconductor</code><a class="anchor" aria-label="anchor" href="#install-ssnappy-package-from-bioconductor"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"sSNAPPY"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wenjun-liu.github.io/sSNAPPY/">sSNAPPY</a></span><span class="op">)</span></code></pre></div>
<p>Load the other packages used in this tutorial.</p>
</div>
<div class="section level2">
<h2 id="load-data">load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<p>The example dataset used for this tutorial has been built into the
<code>sSNAPPY</code> package. It’s a subset of data retrieved from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4928895/" class="external-link">Singhal H
et al. 2016</a>, where ER-positive primary breast cancer tumor tissues
collected from 12 patients were split into fragments of equal sizes for
different treatments. For the purpose of this tutorial, we only took the
RNA-seq data from samples that were treated with vehicle, E2 OR E2 +
R5020 for 48 hrs. They were from 5 different patients, giving rise to 15
samples in total. More detailed description of the dataset can be
assessed through the help page (<code><a href="../reference/logCPM_example.html">?logCPM_example</a></code> and
<code><a href="../reference/metadata_example.html">?metadata_example</a></code>).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># check if samples included in the logCPM matrix and metadata dataframe are identical</span>
<span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">setequal</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span>, <span class="va">metadata_example</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] TRUE</span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># View sample metadata</span>
<span class="fu">datatable</span><span class="op">(</span><span class="va">metadata_example</span>,  filter <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-4a769d20fe48b65120d2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4a769d20fe48b65120d2">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;N2&quot;,&quot;N3&quot;,&quot;P4&quot;,&quot;P5&quot;,&quot;P6&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Vehicle&quot;,&quot;E2&quot;,&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020_N2_48&quot;,&quot;E2+R5020_N3_48&quot;,&quot;E2+R5020_P4_48&quot;,&quot;E2+R5020_P5_48&quot;,&quot;E2+R5020_P6_48&quot;,&quot;R5020_N2_48&quot;,&quot;R5020_N3_48&quot;,&quot;R5020_P4_48&quot;,&quot;R5020_P5_48&quot;,&quot;R5020_P6_48&quot;,&quot;Vehicle_N2_48&quot;,&quot;Vehicle_N3_48&quot;,&quot;Vehicle_P4_48&quot;,&quot;Vehicle_P5_48&quot;,&quot;Vehicle_P6_48&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Negative&quot;,&quot;Positive&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["N2","N2","N2","N3","N3","N3","P4","P4","P4","P5","P5","P5","P6","P6","P6"],["Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020"],["Vehicle_N2_48","E2+R5020_N2_48","R5020_N2_48","Vehicle_N3_48","E2+R5020_N3_48","R5020_N3_48","Vehicle_P4_48","E2+R5020_P4_48","R5020_P4_48","Vehicle_P5_48","E2+R5020_P5_48","R5020_P5_48","Vehicle_P6_48","E2+R5020_P6_48","R5020_P6_48"],["Negative","Negative","Negative","Negative","Negative","Negative","Positive","Positive","Positive","Positive","Positive","Positive","Positive","Positive","Positive"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>patient<\/th>\n      <th>treatment<\/th>\n      <th>sample<\/th>\n      <th>PR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level2">
<h2 id="compute-weighted-single-sample-logfcs-sslogfcs">Compute weighted single-sample logFCs (ssLogFCs)<a class="anchor" aria-label="anchor" href="#compute-weighted-single-sample-logfcs-sslogfcs"></a>
</h2>
<p>It is expected that the logCPM matrix will be filtered to remove
undetectable genes and normalised to correct for library sizes or other
systematic artefacts, such as gene length or GC contents, prior to
applying this method. Filtration and normalisation has been performed on
the example dataset.</p>
<p>Before single-sample logFCs (ssLogFCs) can be computed, row names of
the logCPM matrix need to be converted to <code>entrez ID</code>. This
is because all the pathway topology information retrieved will be in
<code>entrez ID</code>. The conversion can be achieved through
bioconductor packages <code>AnnotationHub</code> and
<code>ensembldb</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##       Vehicle_N2_48 E2+R5020_N2_48 R5020_N2_48 Vehicle_N3_48 E2+R5020_N3_48</span>
<span class="co">## 7105       4.081669       5.109147    4.695391      4.673548       5.217774</span>
<span class="co">## 57147      5.211258       3.844185    4.343678      3.601125       3.427482</span>
<span class="co">## 55732      2.111865       1.366069    2.471200      1.975589       1.713835</span>
<span class="co">## 2268       6.930587       4.341791    4.388130      5.309135       5.066294</span>
<span class="co">## 90529      1.725384       2.997631    3.035200      3.161472       3.320000</span>
<span class="co">## 22875      4.099909       3.635736    3.845961      4.701901       4.886946</span>
<span class="co">##       R5020_N3_48 Vehicle_P4_48 E2+R5020_P4_48 R5020_P4_48 Vehicle_P5_48</span>
<span class="co">## 7105     4.296947      4.667555       4.741936    4.621193      5.484910</span>
<span class="co">## 57147    4.277160      5.128271       4.486209    4.624768      3.549639</span>
<span class="co">## 55732    2.222948      2.437893       3.147235    3.327889      2.368816</span>
<span class="co">## 2268     3.775364      2.912324       3.911222    4.258518      3.029545</span>
<span class="co">## 90529    2.719585      4.578478       3.624418    3.660976      3.477681</span>
<span class="co">## 22875    4.624176      5.453882       4.616053    5.200638      4.080996</span>
<span class="co">##       E2+R5020_P5_48 R5020_P5_48 Vehicle_P6_48 E2+R5020_P6_48 R5020_P6_48</span>
<span class="co">## 7105        5.678579    5.118092      5.636142       5.434636    5.753488</span>
<span class="co">## 57147       4.113434    3.802290      3.815933       3.965867    3.914067</span>
<span class="co">## 55732       2.438389    2.594323      1.028753       1.493969    2.365653</span>
<span class="co">## 2268        1.530079    3.044646      2.904204       3.796048    3.336565</span>
<span class="co">## 90529       3.307986    3.073607      3.577813       3.450174    3.613297</span>
<span class="co">## 22875       3.696430    3.267462      4.436550       4.363616    4.402518</span></code></pre>
<p>To compute the ssLogFCs, samples must be in matching pairs. In the
example, treated samples are matched to the corresponding control sample
that were derived from the same patients. So the <code>factor</code>
parameter of the <code><a href="../reference/weight_ssFC.html">weight_ssFC()</a></code> functions needs to be set to
be “patient”. The function also requires the control treatment level to
be specified, which was “Vehicle” in this case.</p>
<p><code><a href="../reference/weight_ssFC.html">weight_ssFC()</a></code> requires both the logCPM matrix and sample
metadata as input. The column names of the logCPM matrix should be
sample name, matching to a column called <code>sample</code> in the
metadata. The metadata must also contain a treatment column, and a
column corresponding to the <code>factor</code> parameter (i.e.. patient
in this case).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#compute weighted single sample logFCs</span>
<span class="va">weightedFC</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/weight_ssFC.html">weight_ssFC</a></span><span class="op">(</span><span class="va">logCPM_example</span>, metadata <span class="op">=</span> <span class="va">metadata_example</span>,
factor <span class="op">=</span> <span class="st">"patient"</span>, control <span class="op">=</span> <span class="st">"Vehicle"</span><span class="op">)</span></code></pre></div>
<p>The <code><a href="../reference/weight_ssFC.html">weight_ssFC()</a></code> function firstly computes raw ssLogFCs
for each gene by subtracting logCPM values of control sample from the
logCPM values of treated samples for each patient.</p>
<p>It has been demonstrated previously that in RNA-seq data, lowly
expressed genes turn to have larger variance, which is also demonstrated
by the plots below. To reduce the impact of this artefact,
<code>weight_ssFC</code> also weight each ssLogFCs by estimating the
relationship between the variance in ssLogFCs and mean logCPM, and
defining the gene-wise weight to be the inverse of the predicted
variance of that gene’s mean logCPM value.</p>
<p>The output of the <code><a href="../reference/weight_ssFC.html">weight_ssFC()</a></code> function is a list with
two element, where one is the weighted ssLogFCs matrix and the other is
a vector of gene-wise weights.</p>
<p><img src="sSNAPPY_files/figure-html/lowess-1.png" width="768"></p>
</div>
<div class="section level2">
<h2 id="retrieve-pathway-topologies-in-required-format">Retrieve pathway topologies in required format<a class="anchor" aria-label="anchor" href="#retrieve-pathway-topologies-in-required-format"></a>
</h2>
<p><em>sSNAPPY</em> adopts the pathway perturbation scoring algorithm
proposed in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2732297/" class="external-link"><em>SPIA</em>
(Tarca AL et al. 2009)</a>, which makes use of gene-set topologies and
gene-gene interaction to propagate pathway genes’ logFCs down the
topologies to compute pathway perturbation scores, where signs of scores
reflect the potential directions of changes.</p>
<p>Therefore, pathway topology information need to be firstly retrieved
from your chosen database and converted to weight adjacency matrices,
the format required to apply the scoring algorithm. This step is
achieved through a chain of functions that are part of the <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-13-20" class="external-link"><em>graphite</em>
package (Sales G et al. 2012)</a> and has been nested into one simple
function in this package: <code><a href="../reference/weightedAdjMatrix.html">weightedAdjMatrix()</a></code>. Databases
that are currently supported are:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"graphite"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"graphite"</span><span class="op">)</span>
<span class="fu">graphite</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graphite/man/pathwayDatabases.html" class="external-link">pathwayDatabases</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">species</span> <span class="op">==</span>  <span class="st">"hsapiens"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">pander</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pander/man/pander.html" class="external-link">pander</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table style="width:36%;" class="table">
<colgroup>
<col width="15%">
<col width="20%">
</colgroup>
<thead><tr class="header">
<th align="center">species</th>
<th align="center">database</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">kegg</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">panther</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">pathbank</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">pharmgkb</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">reactome</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">smpdb</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">wikipathways</td>
</tr>
</tbody>
</table>
<p>The retrieved topology information will be saved as an Rdata file in
the specified directory so this step only needs to be performed once for
each database. When loaded into the environment, the topologies will be
saved in a <code>list</code> called <code>gsTopology</code> by
default.</p>
<p>This vignette chose <em>KEGG</em> pathways as an example.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/weightedAdjMatrix.html">weightedAdjMatrix</a></span><span class="op">(</span>database <span class="op">=</span> <span class="st">"kegg"</span>, outputDir <span class="op">=</span> <span class="st">"gsTopology.rda"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"gsTopology.rda"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsTopology</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "Glycolysis / Gluconeogenesis"            </span>
<span class="co">## [2] "Citrate cycle (TCA cycle)"               </span>
<span class="co">## [3] "Pentose phosphate pathway"               </span>
<span class="co">## [4] "Pentose and glucuronate interconversions"</span>
<span class="co">## [5] "Fructose and mannose metabolism"         </span>
<span class="co">## [6] "Galactose metabolism"</span></code></pre>
<p>If only selected pathways are of interest, it’s possible to only
retrieve the topologies of those pathways by specifying the pathway
names.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/weightedAdjMatrix.html">weightedAdjMatrix</a></span><span class="op">(</span>
  database <span class="op">=</span> <span class="st">"kegg"</span>,
  pathwayName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>
    <span class="st">"Glycolysis / Gluconeogenesis"</span>, 
    <span class="st">"Citrate cycle (TCA cycle)"</span>,
    <span class="st">"Pentose phosphate pathway"</span>
    <span class="op">)</span>, 
  outputDir <span class="op">=</span> <span class="st">"gsTopology.rda"</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"gsTopology.rda"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsTopology</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="score-single-sample-pathway-perturbation">Score single sample pathway perturbation<a class="anchor" aria-label="anchor" href="#score-single-sample-pathway-perturbation"></a>
</h2>
<p>Once the expression matrix, sample metadata and pathway topologies
are all ready, single sample pathway perturbation scores (PS) can be
computed using function <code><a href="../reference/perturbationScore.html">perturbationScore()</a></code>, which returns
a data.frame containing the test perturbation scores for each sample
each pathway.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ssPertScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/perturbationScore.html">perturbationScore</a></span><span class="op">(</span><span class="va">weightedFC</span><span class="op">$</span><span class="va">logFC</span>, <span class="va">gsTopology</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ssPertScore</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##           sample           tA                                   gs_name</span>
<span class="co">## 1 E2+R5020_N2_48  0.006751152 EGFR tyrosine kinase inhibitor resistance</span>
<span class="co">## 2    R5020_N2_48  0.006744556 EGFR tyrosine kinase inhibitor resistance</span>
<span class="co">## 3 E2+R5020_N3_48  0.002683974 EGFR tyrosine kinase inhibitor resistance</span>
<span class="co">## 4    R5020_N3_48 -0.003974106 EGFR tyrosine kinase inhibitor resistance</span>
<span class="co">## 5 E2+R5020_P4_48  0.004130870 EGFR tyrosine kinase inhibitor resistance</span>
<span class="co">## 6    R5020_P4_48  0.015279018 EGFR tyrosine kinase inhibitor resistance</span></code></pre>
</div>
<div class="section level2">
<h2 id="generate-null-distributions-of-perturbation-scores">Generate null distributions of perturbation scores<a class="anchor" aria-label="anchor" href="#generate-null-distributions-of-perturbation-scores"></a>
</h2>
<p>To derive the empirical p-values for each single sample PS or
normalize the raw scores for comparing overall treatment effects, null
distributions of scores for each pathway is generated through a
sample-label permutation approach.</p>
<p>For each round of permutation, sample labels are randomly shuffled to
derive the permuted ssLogFCs, which are then used to score pathway
perturbation. We recommend to perform a minimum of 1000 rounds of
permutation, which means at least 8 samples are required. The
<code><a href="../reference/generate_PermutedScore.html">generate_PermutedScore()</a></code> function does not require sample
metadata but the number of treatments in the study design, including the
control treatment, need to be specified. In this example data, the
number of treatment was 3.</p>
<p>Output of the <code><a href="../reference/generate_PermutedScore.html">generate_PermutedScore()</a></code> function is a
list where each element is a vector of permuted perturbation scores for
a pathway.</p>
<p>The permutation step relies on the parallel computing feature
provided by <code>BiocParallel</code>. User can choose to customize the
parallel back-end or stick with the default one returned by
<code><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">BiocParallel::bpparam()</a></code>. Depending on the size of the data,
this step can take some time to complete. If the sample size is large,
we recommend users to consider performing this step on a HPC.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">permutedScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_PermutedScore.html">generate_PermutedScore</a></span><span class="op">(</span>
  logCPM <span class="op">=</span> <span class="va">logCPM_example</span>, 
  numOfTreat <span class="op">=</span> <span class="fl">3</span>, NB <span class="op">=</span> <span class="fl">1000</span>, 
  gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, 
  weight <span class="op">=</span> <span class="va">weightedFC</span><span class="op">$</span><span class="va">weight</span>
<span class="op">)</span></code></pre></div>
<p>Let’s randomly choose six pathways and examine the
permutation-derived empirical distribution of their perturbation scores.
They should all be approximately normal distributed.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pl</span> <span class="op">&lt;-</span> <span class="va">permutedScore</span> <span class="op">%&gt;%</span>
    <span class="fu">keep</span><span class="op">(</span><span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">.</span><span class="op">!=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="va">.</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, <span class="fl">6</span><span class="op">)</span><span class="op">]</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>
      <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span>
        <span class="fu">ggplot</span><span class="op">(</span>mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
          <span class="fu">geom_histogram</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
          <span class="fu">xlab</span><span class="op">(</span><span class="st">"Perturbation Score"</span><span class="op">)</span>
      <span class="op">}</span>
    <span class="op">)</span> 
<span class="fu">plot_grid</span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">pl</span>, nrow <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="sSNAPPY_files/figure-html/NullDistribution-1.png" width="768"></p>
<div class="section level3">
<h3 id="significance-of-individal-score">Significance of individal score<a class="anchor" aria-label="anchor" href="#significance-of-individal-score"></a>
</h3>
<p>After the empirical null distributions are generated, the median and
mad will be calculated for each pathway to convert the test
single-sample perturbation scores derived from the
<code><a href="../reference/perturbationScore.html">perturbationScore()</a></code> to robust z-scores: <span class="math display">\[ (Score - Median)/MAD\]</span> Two-sided p-values
associated with each robust z-scores are also computed and will be
corrected for multiple-testing using a user-define approach. The default
is <code>fdr</code>.</p>
<p>The pathways that were significant perturbed within individual
samples are:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">normalisedScores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normaliseByPermutation.html">normaliseByPermutation</a></span><span class="op">(</span><span class="va">permutedScore</span>, <span class="va">ssPertScore</span><span class="op">)</span>
<span class="va">normalisedScores</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">adjPvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">left_join</span><span class="op">(</span><span class="va">metadata_example</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">mutate_at</span><span class="op">(</span><span class="fu">vars</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"gs_name"</span><span class="op">)</span><span class="op">)</span>, <span class="va">as.factor</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">mutate_if</span><span class="op">(</span><span class="va">is.numeric</span>, <span class="va">sprintf</span>, fmt <span class="op">=</span> <span class="st">'%#.4f'</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">mutate</span><span class="op">(</span>Direction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">robustZ</span> <span class="op">&lt;</span> <span class="fl">0</span>, <span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span>
        <span class="va">sample</span>, <span class="va">patient</span>, Treatment <span class="op">=</span> <span class="va">treatment</span>,
        `Perturbation Score` <span class="op">=</span> <span class="va">robustZ</span>, <span class="va">Direction</span>,
        `Gene-set name` <span class="op">=</span> <span class="va">gs_name</span>, 
        `P-value` <span class="op">=</span> <span class="va">pvalue</span>, 
        FDR <span class="op">=</span> <span class="va">adjPvalue</span>
    <span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">datatable</span><span class="op">(</span>
        filter <span class="op">=</span> <span class="st">"top"</span>, 
        options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
            columnDefs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="st">"Direction"</span>, visible <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
        <span class="op">)</span>
    <span class="op">)</span> <span class="op">%&gt;%</span> 
    <span class="fu">formatStyle</span><span class="op">(</span>
        <span class="st">'Perturbation Score'</span>, <span class="st">'Direction'</span>,
        backgroundColor <span class="op">=</span> <span class="fu">styleEqual</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'lightblue'</span>, <span class="st">'indianred'</span><span class="op">)</span><span class="op">)</span>
    <span class="op">)</span></code></pre></div>
<div id="htmlwidget-5716e6ae4fdcdf8668cb" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5716e6ae4fdcdf8668cb">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020_N2_48&quot;,&quot;R5020_N2_48&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"disabled\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Vehicle&quot;,&quot;E2&quot;,&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Amoebiasis&quot;,&quot;Growth hormone synthesis, secretion and action&quot;,&quot;PI3K-Akt signaling pathway&quot;,&quot;TGF-beta signaling pathway&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8"],["E2+R5020_N2_48","E2+R5020_N2_48","E2+R5020_N2_48","E2+R5020_N2_48","R5020_N2_48","R5020_N2_48","R5020_N2_48","R5020_N2_48"],["N2","N2","N2","N2","N2","N2","N2","N2"],["E2+R5020","E2+R5020","E2+R5020","E2+R5020","R5020","R5020","R5020","R5020"],["3.4671","-9.5671","3.7926","3.7441","3.6851","-10.6115","4.1352","3.8332"],["Activation","Inhibited","Activation","Activation","Activation","Inhibited","Activation","Activation"],["PI3K-Akt signaling pathway","TGF-beta signaling pathway","Growth hormone synthesis, secretion and action","Amoebiasis","PI3K-Akt signaling pathway","TGF-beta signaling pathway","Growth hormone synthesis, secretion and action","Amoebiasis"],["0.0005","0.0000","0.0001","0.0002","0.0002","0.0000","0.0000","0.0001"],["0.0284","0.0000","0.0130","0.0130","0.0123","0.0000","0.0038","0.0091"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>patient<\/th>\n      <th>Treatment<\/th>\n      <th>Perturbation Score<\/th>\n      <th>Direction<\/th>\n      <th>Gene-set name<\/th>\n      <th>P-value<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":5,"visible":false},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[5]; $(this.api().cell(row, 4).node()).css({'background-color':value == \"Inhibited\" ? \"lightblue\" : value == \"Activation\" ? \"indianred\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script><div class="section level4">
<h4 id="visualisation">Visualisation<a class="anchor" aria-label="anchor" href="#visualisation"></a>
</h4>
<p>We can use the <code>plot_gsNetwork</code> function to visualise the
significantly perturbed biological pathways as networks, where edges
between gene-sets reflect how much overlap those two gene-sets share.
The function can take <code>normaliseByPermutation</code>’s output, or a
subset of it as its direct input.</p>
<p>Nodes in the network plots could be coloured by the predicted
direction of perturbation (i.e.. sign of robust z-score):</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pl</span> <span class="op">&lt;-</span> <span class="va">normalisedScores</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">adjPvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>
    <span class="va">plot_gsNetwork</span>, 
    <span class="co"># layout = "dh",</span>
    gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, 
    colorBy <span class="op">=</span> <span class="st">"robustZ"</span>
  <span class="op">)</span>
<span class="fu">plot_grid</span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">pl</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="sSNAPPY_files/figure-html/sigGS_nt_zscore-1.png" width="960"></p>
<p>Or p-values:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pl</span> <span class="op">&lt;-</span> <span class="va">normalisedScores</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">adjPvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>
        <span class="va">plot_gsNetwork</span>, 
        layout <span class="op">=</span> <span class="st">"dh"</span>,
        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, 
        colorBy <span class="op">=</span> <span class="st">"pvalue"</span>, 
        color_lg_title <span class="op">=</span> <span class="st">"P-value"</span>
    <span class="op">)</span>
<span class="fu">plot_grid</span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">pl</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="sSNAPPY_files/figure-html/sigGS_nt_pvalue-1.png" width="960"></p>
<p>The function allows you to customize the layout, colour, edge
transparency and other aesthetics of the graph. More information can be
found in the help page (<code><a href="../reference/plot_gsNetwork.html">?plot_gsNetwork</a></code>). Output of the
graph is a <code>ggplot</code> object and the theme of it can be changed
just as other <code>ggplot</code> figures.</p>
</div>
</div>
<div class="section level3">
<h3 id="significance-of-overall-treatment-effect">Significance of overall treatment effect<a class="anchor" aria-label="anchor" href="#significance-of-overall-treatment-effect"></a>
</h3>
<p>Normalised perturbation scores can also be used to model mean
treatment effects. An advantage of this method is that it has great
flexibility that allows you to incorporate other cofactors or covariate
in your modelling.</p>
<p>For example, in the example dataset, samples were collected from
patients with different progesterone receptor (PR) status and we can
include it as a cofactor to offset its confounding effect.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op">&lt;-</span> <span class="va">normalisedScores</span> <span class="op">%&gt;%</span>
    <span class="fu">left_join</span><span class="op">(</span><span class="va">metadata_example</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="co">#.["Estrogen signaling pathway"] %&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">robustZ</span> <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">treatment</span> <span class="op">+</span> <span class="va">PR</span>, data <span class="op">=</span> <span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">summary</span><span class="op">)</span>
<span class="va">treat_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>, 
  <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span>
    <span class="va">fit</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">coefficients</span> <span class="op">%&gt;%</span>
      <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="va">.</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,<span class="op">]</span> <span class="op">%&gt;%</span>
      <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Estimate</span>, pvalue <span class="op">=</span> <span class="va">`Pr(&gt;|t|)`</span> <span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu">rownames_to_column</span><span class="op">(</span><span class="st">"Treatment"</span><span class="op">)</span> <span class="op">%&gt;%</span>
      <span class="fu">mutate</span><span class="op">(</span>
        gs_name <span class="op">=</span> <span class="va">x</span>, 
        FDR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">pvalue</span>, <span class="st">"fdr"</span><span class="op">)</span>, 
        Treatment <span class="op">=</span> <span class="fu">str_remove_all</span><span class="op">(</span><span class="va">Treatment</span>, <span class="st">"treatment"</span><span class="op">)</span>
      <span class="op">)</span> 
  <span class="op">}</span>, 
  simplify <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">bind_rows</span><span class="op">(</span><span class="op">)</span> </code></pre></div>
<p>The pathways that were on average perturbed due to each treatment
were:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">treat_sig</span> <span class="op">%&gt;%</span> 
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">mutate_at</span><span class="op">(</span><span class="fu">vars</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Treatment"</span>, <span class="st">"gs_name"</span><span class="op">)</span><span class="op">)</span>, <span class="va">as.factor</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">mutate_if</span><span class="op">(</span><span class="va">is.numeric</span>, <span class="va">sprintf</span>, fmt <span class="op">=</span> <span class="st">'%#.4f'</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">mutate</span><span class="op">(</span>Direction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">Estimate</span> <span class="op">&lt;</span> <span class="fl">0</span>, <span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span>
        <span class="va">Treatment</span>, `Perturbation Score` <span class="op">=</span> <span class="va">Estimate</span>, <span class="va">Direction</span>,
        `Gene-set name` <span class="op">=</span> <span class="va">gs_name</span>, 
        `P-value` <span class="op">=</span> <span class="va">pvalue</span>, 
        <span class="va">FDR</span>
    <span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">datatable</span><span class="op">(</span>
        filter <span class="op">=</span> <span class="st">"top"</span>, 
        options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
            columnDefs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="st">"Direction"</span>, visible <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
        <span class="op">)</span>
    <span class="op">)</span> <span class="op">%&gt;%</span> 
    <span class="fu">formatStyle</span><span class="op">(</span>
        <span class="st">'Perturbation Score'</span>, <span class="st">'Direction'</span>,
        backgroundColor <span class="op">=</span> <span class="fu">styleEqual</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'lightblue'</span>, <span class="st">'indianred'</span><span class="op">)</span><span class="op">)</span>
    <span class="op">)</span></code></pre></div>
<div id="htmlwidget-bb64e620af0c3a2752f9" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-bb64e620af0c3a2752f9">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Aldosterone synthesis and secretion&quot;,&quot;Antigen processing and presentation&quot;,&quot;Cellular senescence&quot;,&quot;Cholinergic synapse&quot;,&quot;ECM-receptor interaction&quot;,&quot;Endocrine and other factor-regulated calcium reabsorption&quot;,&quot;Estrogen signaling pathway&quot;,&quot;Focal adhesion&quot;,&quot;Human papillomavirus infection&quot;,&quot;Malaria&quot;,&quot;Melanoma&quot;,&quot;Mineral absorption&quot;,&quot;Neutrophil extracellular trap formation&quot;,&quot;Ovarian steroidogenesis&quot;,&quot;PI3K-Akt signaling pathway&quot;,&quot;Prostate cancer&quot;,&quot;Ras signaling pathway&quot;,&quot;Renin secretion&quot;,&quot;Small cell lung cancer&quot;,&quot;Staphylococcus aureus infection&quot;,&quot;Tight junction&quot;,&quot;Toxoplasmosis&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39"],["E2+R5020","R5020","E2+R5020","R5020","E2+R5020","E2+R5020","R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","E2+R5020","R5020"],["0.9485","1.4044","-0.7342","-0.6872","-0.8185","-1.1526","-0.7144","1.4553","-1.0317","-1.0809","1.4769","1.6224","1.5205","1.7987","1.2589","1.6920","0.8863","1.0952","1.1894","1.4915","2.0458","2.0597","-0.6589","-0.6573","1.2543","1.1174","2.1320","2.1828","1.2983","1.4653","1.7263","1.5526","-1.3850","1.1033","0.5541","0.5233","-1.1231","-1.0981","-0.9856"],["Activation","Activation","Inhibited","Inhibited","Inhibited","Inhibited","Inhibited","Activation","Inhibited","Inhibited","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Inhibited","Inhibited","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Inhibited","Activation","Activation","Activation","Inhibited","Inhibited","Inhibited"],["Aldosterone synthesis and secretion","Aldosterone synthesis and secretion","Antigen processing and presentation","Antigen processing and presentation","Cellular senescence","Cholinergic synapse","Cholinergic synapse","ECM-receptor interaction","Endocrine and other factor-regulated calcium reabsorption","Endocrine and other factor-regulated calcium reabsorption","Estrogen signaling pathway","Estrogen signaling pathway","Focal adhesion","Focal adhesion","Human papillomavirus infection","Human papillomavirus infection","Malaria","Malaria","Melanoma","Melanoma","Mineral absorption","Mineral absorption","Neutrophil extracellular trap formation","Neutrophil extracellular trap formation","Ovarian steroidogenesis","Ovarian steroidogenesis","PI3K-Akt signaling pathway","PI3K-Akt signaling pathway","Prostate cancer","Prostate cancer","Ras signaling pathway","Ras signaling pathway","Renin secretion","Small cell lung cancer","Staphylococcus aureus infection","Staphylococcus aureus infection","Tight junction","Toxoplasmosis","Toxoplasmosis"],["0.0174","0.0025","0.0347","0.0443","0.0184","0.0012","0.0146","0.0128","0.0500","0.0423","0.0015","0.0009","0.0292","0.0144","0.0208","0.0052","0.0297","0.0120","0.0080","0.0025","0.0018","0.0017","0.0386","0.0390","0.0225","0.0356","0.0332","0.0303","0.0308","0.0187","0.0336","0.0496","0.0127","0.0154","0.0304","0.0378","0.0056","0.0146","0.0232"],["0.0174","0.0051","0.0443","0.0443","0.0368","0.0025","0.0146","0.0257","0.0500","0.0500","0.0015","0.0015","0.0292","0.0288","0.0208","0.0105","0.0297","0.0241","0.0080","0.0050","0.0018","0.0018","0.0390","0.0390","0.0356","0.0356","0.0332","0.0332","0.0308","0.0308","0.0496","0.0496","0.0255","0.0308","0.0378","0.0378","0.0112","0.0232","0.0232"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Treatment<\/th>\n      <th>Perturbation Score<\/th>\n      <th>Direction<\/th>\n      <th>Gene-set name<\/th>\n      <th>P-value<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":3,"visible":false},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[3]; $(this.api().cell(row, 2).node()).css({'background-color':value == \"Inhibited\" ? \"lightblue\" : value == \"Activation\" ? \"indianred\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script><p>Results from this analysis indicate that the estrogen signalling
pathway was significantly activated among both E2 and E2+R5020 treated
samples, which makes sense biologically.</p>
<div class="section level4">
<h4 id="visualisation-1">Visualisation<a class="anchor" aria-label="anchor" href="#visualisation-1"></a>
</h4>
<p>Results of <code>lm</code> can also be visualised using the
<code>plot_gsNetwork</code> function. We just need to change the name of
the <code>Estimate</code> column to <code>robustZ</code> to colour the
networks by the predicted directionality.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">treat_sig</span> <span class="op">%&gt;%</span> 
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"R5020"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>robustZ <span class="op">=</span> <span class="va">Estimate</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="../reference/plot_gsNetwork.html">plot_gsNetwork</a></span><span class="op">(</span>
        <span class="co"># layout = "stress",</span>
        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, 
        colorBy <span class="op">=</span> <span class="st">"robustZ"</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme</span><span class="op">(</span>
        panel.grid <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>, 
        panel.background <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>
    <span class="op">)</span> </code></pre></div>
<p><img src="sSNAPPY_files/figure-html/unnamed-chunk-3-1.png" width="576"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Sales G, Calura E, Cavalieri D, Romualdi C (2012). “graphite - a
Bioconductor package to convert pathway topology to gene network.” BMC
Bioinformatics. <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-13-20" class="external-link uri">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-13-20</a>.</li>
<li>Tarca, Adi Laurentiu et al. (2009). “A novel signaling pathway
impact analysis.” Bioinformatics vol. 25,1 : 75-82. <a href="doi:10.1093/bioinformatics/btn577" class="uri">doi:10.1093/bioinformatics/btn577</a>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">## Running under: macOS Big Sur/Monterey 10.16</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] org.Hs.eg.db_3.14.0  AnnotationDbi_1.56.2 IRanges_2.28.0      </span>
<span class="co">##  [4] S4Vectors_0.32.3     Biobase_2.54.0       BiocGenerics_0.40.0 </span>
<span class="co">##  [7] DT_0.21              cowplot_1.1.1        magrittr_2.0.2      </span>
<span class="co">## [10] forcats_0.5.1        stringr_1.4.0        dplyr_1.0.8         </span>
<span class="co">## [13] purrr_0.3.4          readr_2.1.2          tidyr_1.2.0         </span>
<span class="co">## [16] tibble_3.1.6         ggplot2_3.3.5        tidyverse_1.3.1     </span>
<span class="co">## [19] sSNAPPY_0.99.0       BiocStyle_2.22.0    </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] colorspace_2.0-3       ellipsis_0.3.2         rprojroot_2.0.2       </span>
<span class="co">##   [4] XVector_0.34.0         fs_1.5.2               rstudioapi_0.13       </span>
<span class="co">##   [7] farver_2.1.0           graphlayouts_0.8.0     ggrepel_0.9.1         </span>
<span class="co">##  [10] bit64_4.0.5            fansi_1.0.3            lubridate_1.8.0       </span>
<span class="co">##  [13] xml2_1.3.3             splines_4.1.3          cachem_1.0.6          </span>
<span class="co">##  [16] knitr_1.38             polyclip_1.10-0        jsonlite_1.8.0        </span>
<span class="co">##  [19] broom_0.7.12           dbplyr_2.1.1           png_0.1-7             </span>
<span class="co">##  [22] graph_1.72.0           ggforce_0.3.3          graphite_1.40.0       </span>
<span class="co">##  [25] BiocManager_1.30.16    compiler_4.1.3         httr_1.4.2            </span>
<span class="co">##  [28] backports_1.4.1        Matrix_1.4-1           assertthat_0.2.1      </span>
<span class="co">##  [31] fastmap_1.1.0          cli_3.2.0              tweenr_1.0.2          </span>
<span class="co">##  [34] htmltools_0.5.2        tools_4.1.3            igraph_1.2.11         </span>
<span class="co">##  [37] gtable_0.3.0           glue_1.6.2             GenomeInfoDbData_1.2.7</span>
<span class="co">##  [40] reshape2_1.4.4         rappdirs_0.3.3         Rcpp_1.0.8.3          </span>
<span class="co">##  [43] cellranger_1.1.0       jquerylib_0.1.4        pkgdown_2.0.2         </span>
<span class="co">##  [46] vctrs_0.3.8            Biostrings_2.62.0      nlme_3.1-155          </span>
<span class="co">##  [49] crosstalk_1.2.0        ggraph_2.0.5           xfun_0.30             </span>
<span class="co">##  [52] rvest_1.0.2            lifecycle_1.0.1        MASS_7.3-56           </span>
<span class="co">##  [55] zlibbioc_1.40.0        scales_1.1.1           tidygraph_1.2.0       </span>
<span class="co">##  [58] ragg_1.2.2             hms_1.1.1              parallel_4.1.3        </span>
<span class="co">##  [61] yaml_2.3.5             memoise_2.0.1          gridExtra_2.3         </span>
<span class="co">##  [64] pander_0.6.5           sass_0.4.1             stringi_1.7.6         </span>
<span class="co">##  [67] RSQLite_2.2.11         highr_0.9              desc_1.4.1            </span>
<span class="co">##  [70] checkmate_2.0.0        BiocParallel_1.28.3    GenomeInfoDb_1.30.1   </span>
<span class="co">##  [73] rlang_1.0.2            pkgconfig_2.0.3        systemfonts_1.0.4     </span>
<span class="co">##  [76] bitops_1.0-7           lattice_0.20-45        evaluate_0.15         </span>
<span class="co">##  [79] labeling_0.4.2         htmlwidgets_1.5.4      bit_4.0.4             </span>
<span class="co">##  [82] tidyselect_1.1.2       plyr_1.8.7             bookdown_0.25         </span>
<span class="co">##  [85] R6_2.5.1               generics_0.1.2         DBI_1.1.2             </span>
<span class="co">##  [88] mgcv_1.8-39            pillar_1.7.0           haven_2.4.3           </span>
<span class="co">##  [91] withr_2.5.0            KEGGREST_1.34.0        RCurl_1.98-1.6        </span>
<span class="co">##  [94] modelr_0.1.8           crayon_1.5.1           utf8_1.2.2            </span>
<span class="co">##  [97] tzdb_0.2.0             rmarkdown_2.13         viridis_0.6.2         </span>
<span class="co">## [100] grid_4.1.3             readxl_1.3.1           blob_1.2.2            </span>
<span class="co">## [103] reprex_2.0.1           digest_0.6.29          textshaping_0.3.6     </span>
<span class="co">## [106] munsell_0.5.0          viridisLite_0.4.0      bslib_0.3.1</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://github.com/Wenjun-Liu" class="external-link">Wenjun
Liu</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
