<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>sSNAPPY: Singel Sample directioNAl Pathway Perturbation analYsis • sSNAPPY</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="sSNAPPY: Singel Sample directioNAl Pathway Perturbation analYsis">
<meta property="og:description" content="sSNAPPY">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sSNAPPY</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/sSNAPPY.html">Get Started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/Wenjun-Liu/sSNAPPY" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="sSNAPPY_files/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="sSNAPPY_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="sSNAPPY_files/datatables-binding-0.22/datatables.js"></script><link href="sSNAPPY_files/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="sSNAPPY_files/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="sSNAPPY_files/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="sSNAPPY_files/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="sSNAPPY_files/nouislider-7.0.10/jquery.nouislider.min.js"></script><link href="sSNAPPY_files/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="sSNAPPY_files/selectize-0.12.0/selectize.min.js"></script><link href="sSNAPPY_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="sSNAPPY_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>sSNAPPY: Singel Sample directioNAl Pathway
Perturbation analYsis</h1>
                        <h4 data-toc-skip class="author">Wenjun Nora
Liu</h4>
            <address class="author_afil">
      Dame Roma Mitchell Cancer Research Laboratories, Adelaide Medical
School, University of
Adelaide<br><a class="author_email" href="mailto:#"></a><a href="mailto:wenjun.liu@adelaide.edu.au" class="email">wenjun.liu@adelaide.edu.au</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/Wenjun-Liu/sSNAPPY/blob/HEAD/vignettes/sSNAPPY.Rmd" class="external-link"><code>vignettes/sSNAPPY.Rmd</code></a></small>
      <div class="hidden name"><code>sSNAPPY.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette demonstrates how to use the package
<code>sSNAPPY</code> to compute directional single sample pathway
perturbation scores by incorporating pathway topologies, utilize sample
permutation to test the significance of individual scores and compare
average pathway activities across treatments.</p>
<p>The package also provides a function to visualize overlap between
pathway genes contained in perturbed biological pathways as network
plots.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The package <code>sSNAPPY</code> can be installed using the package
<code>BiocManager</code>, along with other packages required for this
vignette (<code>tidyverse</code>, <code>magrittr</code>,
<code>ggplot2</code>, <code>cowplot</code>, and <code>DT</code>).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"BiocManager"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="va">pkg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tidyverse"</span>, <span class="st">"magrittr"</span>, <span class="st">"ggplot2"</span>, <span class="st">"cowplot"</span>, <span class="st">"DT"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="va">pkg</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"sSNAPPY"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wenjun-liu.github.io/sSNAPPY/" class="external-link">sSNAPPY</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/DT" class="external-link">DT</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-data">load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h2>
<p>The example dataset used for this tutorial can loaded with
<code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code> as shown below. It’s a subset of data retrieved from
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4928895/" class="external-link">Singhal
H et al. 2016</a>, where ER-positive primary breast cancer tumor tissues
collected from 12 patients were split into fragments of equal sizes for
different treatments. For the purpose of this tutorial, we only took the
RNA-seq data from samples that were treated with vehicle, E2 OR E2 +
R5020 for 48 hrs. They were from 5 different patients, giving rise to 15
samples in total. More detailed description of the dataset can be
assessed through the help page (<code><a href="../reference/logCPM_example.html">?logCPM_example</a></code> and
<code><a href="../reference/metadata_example.html">?metadata_example</a></code>).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">)</span>
<span class="co"># check if samples included in the logCPM matrix and metadata dataframe are identical</span>
<span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setequal</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span>, <span class="va">metadata_example</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] TRUE</span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># View sample metadata</span>
<span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="va">metadata_example</span>,  filter <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-9557e72cc6da0b13481e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-9557e72cc6da0b13481e">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;N2&quot;,&quot;N3&quot;,&quot;P4&quot;,&quot;P5&quot;,&quot;P6&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Vehicle&quot;,&quot;E2&quot;,&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020_N2_48&quot;,&quot;E2+R5020_N3_48&quot;,&quot;E2+R5020_P4_48&quot;,&quot;E2+R5020_P5_48&quot;,&quot;E2+R5020_P6_48&quot;,&quot;R5020_N2_48&quot;,&quot;R5020_N3_48&quot;,&quot;R5020_P4_48&quot;,&quot;R5020_P5_48&quot;,&quot;R5020_P6_48&quot;,&quot;Vehicle_N2_48&quot;,&quot;Vehicle_N3_48&quot;,&quot;Vehicle_P4_48&quot;,&quot;Vehicle_P5_48&quot;,&quot;Vehicle_P6_48&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Negative&quot;,&quot;Positive&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["N2","N2","N2","N3","N3","N3","P4","P4","P4","P5","P5","P5","P6","P6","P6"],["Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020"],["Vehicle_N2_48","E2+R5020_N2_48","R5020_N2_48","Vehicle_N3_48","E2+R5020_N3_48","R5020_N3_48","Vehicle_P4_48","E2+R5020_P4_48","R5020_P4_48","Vehicle_P5_48","E2+R5020_P5_48","R5020_P5_48","Vehicle_P6_48","E2+R5020_P6_48","R5020_P6_48"],["Negative","Negative","Negative","Negative","Negative","Negative","Positive","Positive","Positive","Positive","Positive","Positive","Positive","Positive","Positive"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>patient<\/th>\n      <th>treatment<\/th>\n      <th>sample<\/th>\n      <th>PR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level2">
<h2 id="compute-weighted-single-sample-logfcs-sslogfcs">Compute weighted single-sample logFCs (ssLogFCs)<a class="anchor" aria-label="anchor" href="#compute-weighted-single-sample-logfcs-sslogfcs"></a>
</h2>
<p>It is expected that the logCPM matrix will be filtered to remove
undetectable genes and normalised to correct for library sizes or other
systematic artefacts, such as gene length or GC contents, prior to
applying this method. Filtration and normalisation has been performed on
the example dataset.</p>
<p>Before single-sample logFCs (ssLogFCs) can be computed, row names of
the logCPM matrix need to be converted to <code>entrez ID</code>. This
is because all the pathway topology information retrieved will be in
<code>entrez ID</code>. The conversion can be achieved through
bioconductor packages <code>AnnotationHub</code> and
<code>ensembldb</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##       Vehicle_N2_48 E2+R5020_N2_48 R5020_N2_48 Vehicle_N3_48 E2+R5020_N3_48</span>
<span class="co">## 7105       4.081669       5.109147    4.695391      4.673548       5.217774</span>
<span class="co">## 57147      5.211258       3.844185    4.343678      3.601125       3.427482</span>
<span class="co">## 55732      2.111865       1.366069    2.471200      1.975589       1.713835</span>
<span class="co">## 2268       6.930587       4.341791    4.388130      5.309135       5.066294</span>
<span class="co">## 90529      1.725384       2.997631    3.035200      3.161472       3.320000</span>
<span class="co">## 22875      4.099909       3.635736    3.845961      4.701901       4.886946</span>
<span class="co">##       R5020_N3_48 Vehicle_P4_48 E2+R5020_P4_48 R5020_P4_48 Vehicle_P5_48</span>
<span class="co">## 7105     4.296947      4.667555       4.741936    4.621193      5.484910</span>
<span class="co">## 57147    4.277160      5.128271       4.486209    4.624768      3.549639</span>
<span class="co">## 55732    2.222948      2.437893       3.147235    3.327889      2.368816</span>
<span class="co">## 2268     3.775364      2.912324       3.911222    4.258518      3.029545</span>
<span class="co">## 90529    2.719585      4.578478       3.624418    3.660976      3.477681</span>
<span class="co">## 22875    4.624176      5.453882       4.616053    5.200638      4.080996</span>
<span class="co">##       E2+R5020_P5_48 R5020_P5_48 Vehicle_P6_48 E2+R5020_P6_48 R5020_P6_48</span>
<span class="co">## 7105        5.678579    5.118092      5.636142       5.434636    5.753488</span>
<span class="co">## 57147       4.113434    3.802290      3.815933       3.965867    3.914067</span>
<span class="co">## 55732       2.438389    2.594323      1.028753       1.493969    2.365653</span>
<span class="co">## 2268        1.530079    3.044646      2.904204       3.796048    3.336565</span>
<span class="co">## 90529       3.307986    3.073607      3.577813       3.450174    3.613297</span>
<span class="co">## 22875       3.696430    3.267462      4.436550       4.363616    4.402518</span></code></pre>
<p>To compute the ssLogFCs, samples must be in matching pairs. In the
example, treated samples are matched to the corresponding control sample
that were derived from the same patients. So the <code>factor</code>
parameter of the <code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code> functions needs to be set
to be “patient”. The function also requires the control treatment level
to be specified, which was “Vehicle” in this case.</p>
<p><code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code> requires both the logCPM matrix and
sample metadata as input. The column names of the logCPM matrix should
be sample name, matching to a column called <code>sample</code> in the
metadata. The metadata must also contain a treatment column, and a
column corresponding to the <code>factor</code> parameter (i.e.. patient
in this case).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#compute weighted single sample logFCs</span>
<span class="va">weightedFC</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/weight_ss_fc.html">weight_ss_fc</a></span><span class="op">(</span><span class="va">logCPM_example</span>, metadata <span class="op">=</span> <span class="va">metadata_example</span>,
factor <span class="op">=</span> <span class="st">"patient"</span>, control <span class="op">=</span> <span class="st">"Vehicle"</span><span class="op">)</span></code></pre></div>
<p>The <code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code> function firstly computes raw
ssLogFCs for each gene by subtracting logCPM values of control sample
from the logCPM values of treated samples for each patient.</p>
<p>It has been demonstrated previously that in RNA-seq data, lowly
expressed genes turn to have larger variance, which is also demonstrated
by the plots below. To reduce the impact of this artefact,
<code>weight_ss_fc</code> also weight each ssLogFCs by estimating the
relationship between the variance in ssLogFCs and mean logCPM, and
defining the gene-wise weight to be the inverse of the predicted
variance of that gene’s mean logCPM value.</p>
<p><img src="sSNAPPY_files/figure-html/lowess-1.png" width="768"></p>
<p>The output of the <code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code> function is a list with
two element, where one is the weighted ssLogFCs matrix and the other is
a vector of gene-wise weights.</p>
</div>
<div class="section level2">
<h2 id="retrieve-pathway-topologies-in-required-format">Retrieve pathway topologies in required format<a class="anchor" aria-label="anchor" href="#retrieve-pathway-topologies-in-required-format"></a>
</h2>
<p><em>sSNAPPY</em> adopts the pathway perturbation scoring algorithm
proposed in <em><a href="https://bioconductor.org/packages/3.14/SPIA" class="external-link">SPIA</a></em>, which
makes use of gene-set topologies and gene-gene interaction to propagate
pathway genes’ logFCs down the topologies to compute pathway
perturbation scores, where signs of scores reflect the potential
directions of changes.</p>
<p>Therefore, pathway topology information need to be firstly retrieved
from your chosen database and converted to weight adjacency matrices,
the format required to apply the scoring algorithm. This step is
achieved through a chain of functions that are part of the <em><a href="https://bioconductor.org/packages/3.14/grapghite" class="external-link">grapghite</a></em>
and has been nested into one simple function in this package:
<code><a href="../reference/retrieve_topology.html">retrieve_topology()</a></code>. Databases that are currently supported
are:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">graphite</span><span class="op">)</span>
<span class="fu">graphite</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graphite/man/pathwayDatabases.html" class="external-link">pathwayDatabases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">species</span> <span class="op">==</span>  <span class="st">"hsapiens"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">pander</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pander/man/pander.html" class="external-link">pander</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<table style="width:36%;" class="table">
<colgroup>
<col width="15%">
<col width="20%">
</colgroup>
<thead><tr class="header">
<th align="center">species</th>
<th align="center">database</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">kegg</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">panther</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">pathbank</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">pharmgkb</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">reactome</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">smpdb</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">wikipathways</td>
</tr>
</tbody>
</table>
<p>The retrieved topology information will be a list where each element
is a pathway. We recommend you to save it as a file so this step only
needs to be performed once for each database.</p>
<p>This vignette chose <em>KEGG</em> pathways as an example. To reduce
the computation time in the following steps, only half of the randomly
sampled <em>KEGG</em> pathways were kept.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gsTopology</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/retrieve_topology.html">retrieve_topology</a></span><span class="op">(</span>database <span class="op">=</span> <span class="st">"kegg"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsTopology</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "Glycolysis / Gluconeogenesis"            </span>
<span class="co">## [2] "Citrate cycle (TCA cycle)"               </span>
<span class="co">## [3] "Pentose phosphate pathway"               </span>
<span class="co">## [4] "Pentose and glucuronate interconversions"</span>
<span class="co">## [5] "Fructose and mannose metabolism"         </span>
<span class="co">## [6] "Galactose metabolism"</span></code></pre>
<p>If only selected pathways are of interest, it’s possible to only
retrieve the topologies of those pathways by specifying the pathway
names.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gsTopology_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/retrieve_topology.html">retrieve_topology</a></span><span class="op">(</span>
  database <span class="op">=</span> <span class="st">"kegg"</span>,
  pathwayName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>
    <span class="st">"Glycolysis / Gluconeogenesis"</span>, 
    <span class="st">"Citrate cycle (TCA cycle)"</span>,
    <span class="st">"Pentose phosphate pathway"</span>
    <span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsTopology_sub</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "Glycolysis / Gluconeogenesis" "Citrate cycle (TCA cycle)"   </span>
<span class="co">## [3] "Pentose phosphate pathway"</span></code></pre>
</div>
<div class="section level2">
<h2 id="score-single-sample-pathway-perturbation">Score single sample pathway perturbation<a class="anchor" aria-label="anchor" href="#score-single-sample-pathway-perturbation"></a>
</h2>
<p>Once the expression matrix, sample metadata and pathway topologies
are all ready, single sample pathway perturbation scores (PS) can be
computed using function <code><a href="../reference/compute_perturbation_score.html">compute_perturbation_score()</a></code>, which
returns a data.frame containing the test perturbation scores for each
sample each pathway.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ssPertScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_perturbation_score.html">compute_perturbation_score</a></span><span class="op">(</span><span class="va">weightedFC</span><span class="op">$</span><span class="va">logFC</span>, <span class="va">gsTopology</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ssPertScore</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##           sample           tA                                   gs_name</span>
<span class="co">## 1 E2+R5020_N2_48  0.006751152 EGFR tyrosine kinase inhibitor resistance</span>
<span class="co">## 2    R5020_N2_48  0.006744556 EGFR tyrosine kinase inhibitor resistance</span>
<span class="co">## 3 E2+R5020_N3_48  0.002683974 EGFR tyrosine kinase inhibitor resistance</span>
<span class="co">## 4    R5020_N3_48 -0.003974106 EGFR tyrosine kinase inhibitor resistance</span>
<span class="co">## 5 E2+R5020_P4_48  0.004130870 EGFR tyrosine kinase inhibitor resistance</span>
<span class="co">## 6    R5020_P4_48  0.015279018 EGFR tyrosine kinase inhibitor resistance</span></code></pre>
</div>
<div class="section level2">
<h2 id="generate-null-distributions-of-perturbation-scores">Generate null distributions of perturbation scores<a class="anchor" aria-label="anchor" href="#generate-null-distributions-of-perturbation-scores"></a>
</h2>
<p>To derive the empirical p-values for each single sample PS or
normalize the raw scores for comparing overall treatment effects, null
distributions of scores for each pathway is generated through a
sample-label permutation approach.</p>
<p>For each round of permutation, sample labels are randomly shuffled to
derive the permuted ssLogFCs, which are then used to score pathway
perturbation. We recommend to perform a minimum of 1000 rounds of
permutation, which means at least 8 samples are required. The
<code><a href="../reference/generate_permuted_scores.html">generate_permuted_scores()</a></code> function does not require sample
metadata but the number of treatments in the study design, including the
control treatment, need to be specified. In this example data, the
number of treatment was 3.</p>
<p>Output of the <code><a href="../reference/generate_permuted_scores.html">generate_permuted_scores()</a></code> function is a
list where each element is a vector of permuted perturbation scores for
a pathway.</p>
<p>The permutation step relies on the parallel computing feature
provided by <em><a href="https://bioconductor.org/packages/3.14/BiocParallel" class="external-link">BiocParallel</a></em>.
User can choose to customize the parallel back-end or stick with the
default one returned by <code><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">BiocParallel::bpparam()</a></code>. Depending
on the size of the data, this step can take some time to complete. If
the sample size is large, we recommend users to consider performing this
step on a HPC.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">permutedScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_permuted_scores.html">generate_permuted_scores</a></span><span class="op">(</span>
  expreMatrix  <span class="op">=</span> <span class="va">logCPM_example</span>, 
  numOfTreat <span class="op">=</span> <span class="fl">3</span>, NB <span class="op">=</span> <span class="fl">1000</span>, 
  gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, 
  weight <span class="op">=</span> <span class="va">weightedFC</span><span class="op">$</span><span class="va">weight</span>
<span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="test-significances">Test Significances<a class="anchor" aria-label="anchor" href="#test-significances"></a>
</h2>
<div class="section level3">
<h3 id="of-individal-score">Of individal score<a class="anchor" aria-label="anchor" href="#of-individal-score"></a>
</h3>
<p>After the empirical null distributions are generated, the median and
mad will be calculated for each pathway to convert the test
single-sample perturbation scores derived from the
<code><a href="../reference/compute_perturbation_score.html">compute_perturbation_score()</a></code> to robust z-scores: <span class="math display">\[ (Score - Median)/MAD\]</span> Two-sided p-values
associated with each robust z-scores are also computed and will be
corrected for multiple-testing using a user-define approach. The default
is <code>fdr</code>.</p>
<p>The <code>normalise_by_permu</code> function requires the test
perturbation score and permuted perturbation scores as input.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">normalisedScores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalise_by_permu.html">normalise_by_permu</a></span><span class="op">(</span><span class="va">permutedScore</span>, <span class="va">ssPertScore</span><span class="op">)</span></code></pre></div>
<p>Since the permutation step takes a long time to run and the output is
too large to be included as part of the package, results of the
<code>normalise_by_permu</code> step has been pre-computed and can be
loaded with:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"normalisedScores.rda"</span>, package <span class="op">=</span> <span class="st">"sSNAPPY"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The pathways that were significant perturbed within individual
samples are:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">adjPvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_at</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/vars.html" class="external-link">vars</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"gs_name"</span><span class="op">)</span><span class="op">)</span>, <span class="va">as.factor</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_if</a></span><span class="op">(</span><span class="va">is.numeric</span>, <span class="va">sprintf</span>, fmt <span class="op">=</span> <span class="st">'%#.4f'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Direction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">robustZ</span> <span class="op">&lt;</span> <span class="fl">0</span>, <span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span>
        <span class="va">sample</span>, <span class="va">patient</span>, Treatment <span class="op">=</span> <span class="va">treatment</span>,
        `Perturbation Score` <span class="op">=</span> <span class="va">robustZ</span>, <span class="va">Direction</span>,
        `Gene-set name` <span class="op">=</span> <span class="va">gs_name</span>, 
        `P-value` <span class="op">=</span> <span class="va">pvalue</span>, 
        FDR <span class="op">=</span> <span class="va">adjPvalue</span>
    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span>
        filter <span class="op">=</span> <span class="st">"top"</span>, 
        options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
            columnDefs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="st">"Direction"</span>, visible <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
        <span class="op">)</span>
    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/formatCurrency.html" class="external-link">formatStyle</a></span><span class="op">(</span>
        <span class="st">'Perturbation Score'</span>, <span class="st">'Direction'</span>,
        backgroundColor <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DT/man/styleInterval.html" class="external-link">styleEqual</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'lightblue'</span>, <span class="st">'indianred'</span><span class="op">)</span><span class="op">)</span>
    <span class="op">)</span></code></pre></div>
<div id="htmlwidget-5b17678310cd7d3ad443" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5b17678310cd7d3ad443">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020_N2_48&quot;,&quot;R5020_N2_48&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;N2&quot;,&quot;N3&quot;,&quot;P4&quot;,&quot;P5&quot;,&quot;P6&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Vehicle&quot;,&quot;E2&quot;,&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Amoebiasis&quot;,&quot;Growth hormone synthesis, secretion and action&quot;,&quot;PI3K-Akt signaling pathway&quot;,&quot;TGF-beta signaling pathway&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8"],["E2+R5020_N2_48","E2+R5020_N2_48","E2+R5020_N2_48","E2+R5020_N2_48","R5020_N2_48","R5020_N2_48","R5020_N2_48","R5020_N2_48"],["N2","N2","N2","N2","N2","N2","N2","N2"],["E2+R5020","E2+R5020","E2+R5020","E2+R5020","R5020","R5020","R5020","R5020"],["3.4728","-9.3778","3.7926","3.6961","3.6920","-10.4028","4.1352","3.7838"],["Activation","Inhibited","Activation","Activation","Activation","Inhibited","Activation","Activation"],["PI3K-Akt signaling pathway","TGF-beta signaling pathway","Growth hormone synthesis, secretion and action","Amoebiasis","PI3K-Akt signaling pathway","TGF-beta signaling pathway","Growth hormone synthesis, secretion and action","Amoebiasis"],["0.0005","0.0000","0.0001","0.0002","0.0002","0.0000","0.0000","0.0002"],["0.0278","0.0000","0.0158","0.0158","0.0120","0.0000","0.0038","0.0111"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>patient<\/th>\n      <th>Treatment<\/th>\n      <th>Perturbation Score<\/th>\n      <th>Direction<\/th>\n      <th>Gene-set name<\/th>\n      <th>P-value<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":5,"visible":false},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[5]; $(this.api().cell(row, 4).node()).css({'background-color':value == \"Inhibited\" ? \"lightblue\" : value == \"Activation\" ? \"indianred\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script><div class="section level4">
<h4 id="visualisation">Visualisation<a class="anchor" aria-label="anchor" href="#visualisation"></a>
</h4>
<p>We can use the <code>plot_gs_network</code> function to visualise the
significantly perturbed biological pathways as networks, where edges
between gene-sets reflect how much overlap those two gene-sets share.
The function can take <code>normalise_by_permu</code>’s output, or a
subset of it as its direct input.</p>
<p>Nodes in the network plots could be coloured by the predicted
direction of perturbation (i.e.. sign of robust z-score):</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pl</span> <span class="op">&lt;-</span> <span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">adjPvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>
    <span class="va">plot_gs_network</span>, 
    <span class="co"># layout = "dh",</span>
    gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, 
    colorBy <span class="op">=</span> <span class="st">"robustZ"</span>
  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span>
        <span class="va">x</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>
        panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, 
        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>
        <span class="op">)</span> <span class="op">}</span><span class="op">)</span>
<span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span>
    plotlist <span class="op">=</span> <span class="va">pl</span>, 
    nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="sSNAPPY_files/figure-html/sigGS_nt_zscore-1.png" width="960"></p>
<p>Or p-values:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pl</span> <span class="op">&lt;-</span> <span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">adjPvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>
        <span class="va">plot_gs_network</span>, 
        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, 
        colorBy <span class="op">=</span> <span class="st">"pvalue"</span>, 
        color_lg_title <span class="op">=</span> <span class="st">"P-value"</span>
    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span>
        <span class="va">x</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>
        panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, 
        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>
        <span class="op">)</span> <span class="op">}</span><span class="op">)</span>
<span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span>plotlist <span class="op">=</span> <span class="va">pl</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<p><img src="sSNAPPY_files/figure-html/sigGS_nt_pvalue-1.png" width="960"></p>
<p>The function allows you to customize the layout, colour, edge
transparency and other aesthetics of the graph. More information can be
found in the help page (<code><a href="../reference/plot_gs_network.html">?plot_gs_network</a></code>). Output of the
graph is a <code>ggplot</code> object and the theme of it can be changed
just as other <code>ggplot</code> figures.</p>
</div>
</div>
<div class="section level3">
<h3 id="of-overall-treatment-effect">Of overall treatment effect<a class="anchor" aria-label="anchor" href="#of-overall-treatment-effect"></a>
</h3>
<p>Normalised perturbation scores can also be used to model mean
treatment effects. An advantage of this method is that it has great
flexibility that allows you to incorporate other cofactors or covariate
in your modelling.</p>
<p>For example, in the example dataset, samples were collected from
patients with different progesterone receptor (PR) status and we can
include it as a cofactor to offset its confounding effect.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op">&lt;-</span> <span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="co">#.["Estrogen signaling pathway"] %&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">robustZ</span> <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">treatment</span> <span class="op">+</span> <span class="va">PR</span>, data <span class="op">=</span> <span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">summary</span><span class="op">)</span>
<span class="va">treat_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>, 
  <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span>
    <span class="va">fit</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">coefficients</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
      <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
      <span class="va">.</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>,<span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
      <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Estimate</span>, pvalue <span class="op">=</span> <span class="va">`Pr(&gt;|t|)`</span> <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
      <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"Treatment"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>
        gs_name <span class="op">=</span> <span class="va">x</span>, 
        FDR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">pvalue</span>, <span class="st">"fdr"</span><span class="op">)</span>, 
        Treatment <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html" class="external-link">str_remove_all</a></span><span class="op">(</span><span class="va">Treatment</span>, <span class="st">"treatment"</span><span class="op">)</span>
      <span class="op">)</span> 
  <span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="op">)</span> </code></pre></div>
<p>The pathways that were on average perturbed due to each treatment
were:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_at</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/vars.html" class="external-link">vars</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Treatment"</span>, <span class="st">"gs_name"</span><span class="op">)</span><span class="op">)</span>, <span class="va">as.factor</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_if</a></span><span class="op">(</span><span class="va">is.numeric</span>, <span class="va">sprintf</span>, fmt <span class="op">=</span> <span class="st">'%#.4f'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Direction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">Estimate</span> <span class="op">&lt;</span> <span class="fl">0</span>, <span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span>
        <span class="va">Treatment</span>, `Perturbation Score` <span class="op">=</span> <span class="va">Estimate</span>, <span class="va">Direction</span>,
        `Gene-set name` <span class="op">=</span> <span class="va">gs_name</span>, 
        `P-value` <span class="op">=</span> <span class="va">pvalue</span>, 
        <span class="va">FDR</span>
    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span>
        filter <span class="op">=</span> <span class="st">"top"</span>, 
        options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
            columnDefs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="st">"Direction"</span>, visible <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
        <span class="op">)</span>
    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/formatCurrency.html" class="external-link">formatStyle</a></span><span class="op">(</span>
        <span class="st">'Perturbation Score'</span>, <span class="st">'Direction'</span>,
        backgroundColor <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DT/man/styleInterval.html" class="external-link">styleEqual</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'lightblue'</span>, <span class="st">'indianred'</span><span class="op">)</span><span class="op">)</span>
    <span class="op">)</span></code></pre></div>
<div id="htmlwidget-3acf93d30a1c3e9d386f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3acf93d30a1c3e9d386f">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Aldosterone synthesis and secretion&quot;,&quot;Cellular senescence&quot;,&quot;Cholinergic synapse&quot;,&quot;ECM-receptor interaction&quot;,&quot;Estrogen signaling pathway&quot;,&quot;Focal adhesion&quot;,&quot;Human papillomavirus infection&quot;,&quot;Malaria&quot;,&quot;Melanoma&quot;,&quot;Mineral absorption&quot;,&quot;Neutrophil extracellular trap formation&quot;,&quot;Ovarian steroidogenesis&quot;,&quot;PI3K-Akt signaling pathway&quot;,&quot;Prostate cancer&quot;,&quot;Ras signaling pathway&quot;,&quot;Renin secretion&quot;,&quot;Small cell lung cancer&quot;,&quot;Staphylococcus aureus infection&quot;,&quot;Tight junction&quot;,&quot;Toxoplasmosis&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35"],["E2+R5020","R5020","E2+R5020","E2+R5020","R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","E2+R5020","R5020"],["0.9674","1.4237","-0.8339","-1.1795","-0.7286","1.4260","1.4672","1.6127","1.6185","1.9071","1.2505","1.6861","0.8677","1.0799","1.1781","1.4798","2.0158","2.0295","-0.6745","-0.6729","1.1831","1.0498","2.1298","2.1809","1.2865","1.4512","1.7780","1.5990","-1.4120","1.0909","0.5081","0.4781","-1.1317","-1.1117","-1.0020"],["Activation","Activation","Inhibited","Inhibited","Inhibited","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Inhibited","Inhibited","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Inhibited","Activation","Activation","Activation","Inhibited","Inhibited","Inhibited"],["Aldosterone synthesis and secretion","Aldosterone synthesis and secretion","Cellular senescence","Cholinergic synapse","Cholinergic synapse","ECM-receptor interaction","Estrogen signaling pathway","Estrogen signaling pathway","Focal adhesion","Focal adhesion","Human papillomavirus infection","Human papillomavirus infection","Malaria","Malaria","Melanoma","Melanoma","Mineral absorption","Mineral absorption","Neutrophil extracellular trap formation","Neutrophil extracellular trap formation","Ovarian steroidogenesis","Ovarian steroidogenesis","PI3K-Akt signaling pathway","PI3K-Akt signaling pathway","Prostate cancer","Prostate cancer","Ras signaling pathway","Ras signaling pathway","Renin secretion","Small cell lung cancer","Staphylococcus aureus infection","Staphylococcus aureus infection","Tight junction","Toxoplasmosis","Toxoplasmosis"],["0.0160","0.0024","0.0163","0.0013","0.0151","0.0142","0.0015","0.0009","0.0264","0.0130","0.0219","0.0055","0.0343","0.0138","0.0083","0.0026","0.0018","0.0017","0.0347","0.0350","0.0256","0.0407","0.0341","0.0311","0.0302","0.0184","0.0336","0.0496","0.0117","0.0154","0.0385","0.0479","0.0052","0.0123","0.0196"],["0.0160","0.0047","0.0327","0.0026","0.0151","0.0285","0.0015","0.0015","0.0264","0.0261","0.0219","0.0110","0.0343","0.0276","0.0083","0.0052","0.0018","0.0018","0.0350","0.0350","0.0407","0.0407","0.0341","0.0341","0.0302","0.0302","0.0496","0.0496","0.0234","0.0308","0.0479","0.0479","0.0104","0.0196","0.0196"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Treatment<\/th>\n      <th>Perturbation Score<\/th>\n      <th>Direction<\/th>\n      <th>Gene-set name<\/th>\n      <th>P-value<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":3,"visible":false},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[3]; $(this.api().cell(row, 2).node()).css({'background-color':value == \"Inhibited\" ? \"lightblue\" : value == \"Activation\" ? \"indianred\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script><p>Results from this analysis indicate that the estrogen signalling
pathway was significantly activated among both E2 and E2+R5020 treated
samples, which makes sense biologically.</p>
<div class="section level4">
<h4 id="visualisation-1">Visualisation<a class="anchor" aria-label="anchor" href="#visualisation-1"></a>
</h4>
<p>Results of <code>lm</code> can also be visualised using the
<code><a href="../reference/plot_gs_network.html">plot_gs_network()</a></code> function. We just need to change the name
of the <code>Estimate</code> column to <code>robustZ</code> to colour
the networks by the predicted directionality.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> 
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>robustZ <span class="op">=</span> <span class="va">Estimate</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="../reference/plot_gs_network.html">plot_gs_network</a></span><span class="op">(</span>
       
        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, 
        colorBy <span class="op">=</span> <span class="st">"robustZ"</span>
    <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>
        panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, 
        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>
    <span class="op">)</span></code></pre></div>
<p><img src="sSNAPPY_files/figure-html/unnamed-chunk-3-1.png" width="1152"></p>
<p>By default, <code><a href="../reference/plot_gs_network.html">plot_gs_network()</a></code> function does not include
nodes that are not connected to any other nodes, which could be turnt of
by setting the <code>plotIsolated</code> to TURE.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Sales G, Calura E, Cavalieri D, Romualdi C (2012). “graphite - a
Bioconductor package to convert pathway topology to gene network.” BMC
Bioinformatics. <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-13-20" class="external-link uri">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-13-20</a>.</li>
<li>Tarca, Adi Laurentiu et al. (2009). “A novel signaling pathway
impact analysis.” Bioinformatics vol. 25,1 : 75-82. <a href="doi:10.1093/bioinformatics/btn577" class="uri">doi:10.1093/bioinformatics/btn577</a>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.3 (2022-03-10)</span>
<span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">## Running under: macOS Big Sur/Monterey 10.16</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] org.Hs.eg.db_3.14.0  AnnotationDbi_1.56.2 IRanges_2.28.0      </span>
<span class="co">##  [4] S4Vectors_0.32.4     Biobase_2.54.0       BiocGenerics_0.40.0 </span>
<span class="co">##  [7] graphite_1.40.0      DT_0.22              cowplot_1.1.1       </span>
<span class="co">## [10] magrittr_2.0.3       forcats_0.5.1        stringr_1.4.0       </span>
<span class="co">## [13] dplyr_1.0.8          purrr_0.3.4          readr_2.1.2         </span>
<span class="co">## [16] tidyr_1.2.0          tibble_3.1.6         ggplot2_3.3.5       </span>
<span class="co">## [19] tidyverse_1.3.1      sSNAPPY_0.99.6       BiocStyle_2.22.0    </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] readxl_1.4.0                backports_1.4.1            </span>
<span class="co">##   [3] systemfonts_1.0.4           plyr_1.8.7                 </span>
<span class="co">##   [5] igraph_1.3.0                splines_4.1.3              </span>
<span class="co">##   [7] BiocParallel_1.28.3         crosstalk_1.2.0            </span>
<span class="co">##   [9] GenomeInfoDb_1.30.1         digest_0.6.29              </span>
<span class="co">##  [11] htmltools_0.5.2             viridis_0.6.2              </span>
<span class="co">##  [13] fansi_1.0.3                 checkmate_2.0.0            </span>
<span class="co">##  [15] memoise_2.0.1               tzdb_0.3.0                 </span>
<span class="co">##  [17] limma_3.50.3                Biostrings_2.62.0          </span>
<span class="co">##  [19] graphlayouts_0.8.0          modelr_0.1.8               </span>
<span class="co">##  [21] matrixStats_0.61.0          pkgdown_2.0.2              </span>
<span class="co">##  [23] colorspace_2.0-3            blob_1.2.3                 </span>
<span class="co">##  [25] rvest_1.0.2                 rappdirs_0.3.3             </span>
<span class="co">##  [27] ggrepel_0.9.1               textshaping_0.3.6          </span>
<span class="co">##  [29] haven_2.5.0                 xfun_0.30                  </span>
<span class="co">##  [31] crayon_1.5.1                RCurl_1.98-1.6             </span>
<span class="co">##  [33] jsonlite_1.8.0              graph_1.72.0               </span>
<span class="co">##  [35] glue_1.6.2                  polyclip_1.10-0            </span>
<span class="co">##  [37] gtable_0.3.0                zlibbioc_1.40.0            </span>
<span class="co">##  [39] XVector_0.34.0              DelayedArray_0.20.0        </span>
<span class="co">##  [41] scales_1.2.0                DBI_1.1.2                  </span>
<span class="co">##  [43] edgeR_3.36.0                Rcpp_1.0.8.3               </span>
<span class="co">##  [45] viridisLite_0.4.0           bit_4.0.4                  </span>
<span class="co">##  [47] htmlwidgets_1.5.4           httr_1.4.2                 </span>
<span class="co">##  [49] ellipsis_0.3.2              pkgconfig_2.0.3            </span>
<span class="co">##  [51] farver_2.1.0                sass_0.4.1                 </span>
<span class="co">##  [53] dbplyr_2.1.1                locfit_1.5-9.5             </span>
<span class="co">##  [55] utf8_1.2.2                  tidyselect_1.1.2           </span>
<span class="co">##  [57] labeling_0.4.2              rlang_1.0.2                </span>
<span class="co">##  [59] reshape2_1.4.4              munsell_0.5.0              </span>
<span class="co">##  [61] cellranger_1.1.0            tools_4.1.3                </span>
<span class="co">##  [63] cachem_1.0.6                cli_3.2.0                  </span>
<span class="co">##  [65] generics_0.1.2              RSQLite_2.2.12             </span>
<span class="co">##  [67] broom_0.8.0                 evaluate_0.15              </span>
<span class="co">##  [69] fastmap_1.1.0               yaml_2.3.5                 </span>
<span class="co">##  [71] ragg_1.2.2                  knitr_1.38                 </span>
<span class="co">##  [73] bit64_4.0.5                 fs_1.5.2                   </span>
<span class="co">##  [75] tidygraph_1.2.1             pander_0.6.5               </span>
<span class="co">##  [77] KEGGREST_1.34.0             ggraph_2.0.5               </span>
<span class="co">##  [79] nlme_3.1-157                xml2_1.3.3                 </span>
<span class="co">##  [81] compiler_4.1.3              rstudioapi_0.13            </span>
<span class="co">##  [83] png_0.1-7                   reprex_2.0.1               </span>
<span class="co">##  [85] tweenr_1.0.2                bslib_0.3.1                </span>
<span class="co">##  [87] stringi_1.7.6               highr_0.9                  </span>
<span class="co">##  [89] desc_1.4.1                  lattice_0.20-45            </span>
<span class="co">##  [91] Matrix_1.4-1                vctrs_0.4.1                </span>
<span class="co">##  [93] pillar_1.7.0                lifecycle_1.0.1            </span>
<span class="co">##  [95] BiocManager_1.30.16         jquerylib_0.1.4            </span>
<span class="co">##  [97] bitops_1.0-7                GenomicRanges_1.46.1       </span>
<span class="co">##  [99] R6_2.5.1                    bookdown_0.26              </span>
<span class="co">## [101] gridExtra_2.3               MASS_7.3-56                </span>
<span class="co">## [103] assertthat_0.2.1            SummarizedExperiment_1.24.0</span>
<span class="co">## [105] rprojroot_2.0.3             withr_2.5.0                </span>
<span class="co">## [107] GenomeInfoDbData_1.2.7      mgcv_1.8-40                </span>
<span class="co">## [109] parallel_4.1.3              hms_1.1.1                  </span>
<span class="co">## [111] grid_4.1.3                  rmarkdown_2.13             </span>
<span class="co">## [113] MatrixGenerics_1.6.0        ggforce_0.3.3              </span>
<span class="co">## [115] lubridate_1.8.0</span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://github.com/Wenjun-Liu" class="external-link">Wenjun
Liu</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
