<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>sSNAPPY: Singel Sample directioNAl Pathway Perturbation analYsis • sSNAPPY</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="sSNAPPY: Singel Sample directioNAl Pathway Perturbation analYsis">
<meta property="og:description" content="sSNAPPY">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sSNAPPY</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.3.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/sSNAPPY.html">Get Started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/Wenjun-Liu/sSNAPPY" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="sSNAPPY_files/htmlwidgets-1.6.1/htmlwidgets.js"></script><link href="sSNAPPY_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="sSNAPPY_files/datatables-binding-0.27/datatables.js"></script><link href="sSNAPPY_files/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="sSNAPPY_files/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="sSNAPPY_files/dt-core-1.12.1/js/jquery.dataTables.min.js"></script><link href="sSNAPPY_files/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="sSNAPPY_files/nouislider-7.0.10/jquery.nouislider.min.js"></script><link href="sSNAPPY_files/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="sSNAPPY_files/selectize-0.12.0/selectize.min.js"></script><link href="sSNAPPY_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="sSNAPPY_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>sSNAPPY: Singel Sample directioNAl Pathway
Perturbation analYsis</h1>
                        <h4 data-toc-skip class="author">Wenjun Nora
Liu</h4>
            <address class="author_afil">
      Dame Roma Mitchell Cancer Research Laboratories, Adelaide Medical
School, University of
Adelaide<br><a class="author_email" href="mailto:#"></a><a href="mailto:wenjun.liu@adelaide.edu.au" class="email">wenjun.liu@adelaide.edu.au</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/Wenjun-Liu/sSNAPPY/blob/HEAD/vignettes/sSNAPPY.Rmd" class="external-link"><code>vignettes/sSNAPPY.Rmd</code></a></small>
      <div class="hidden name"><code>sSNAPPY.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette demonstrates how to use the package
<code>sSNAPPY</code> to compute directional single sample pathway
perturbation scores by incorporating pathway topologies, utilize sample
permutation to test the significance of individual scores and compare
average pathway activities across treatments.</p>
<p>The package also provides a function to visualise overlap between
pathway genes contained in perturbed biological pathways as network
plots.</p>
</div>
<div class="section level2">
<h2 id="to-get-ready">To get ready<a class="anchor" aria-label="anchor" href="#to-get-ready"></a>
</h2>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>The package <code>sSNAPPY</code> can be installed using the package
<code>BiocManager</code></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"BiocManager"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="co"># Other packages required in this vignette</span></span>
<span><span class="va">pkg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tidyverse"</span>, <span class="st">"magrittr"</span>, <span class="st">"ggplot2"</span>, <span class="st">"cowplot"</span>, <span class="st">"DT"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="va">pkg</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"sSNAPPY"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"htmltools"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wenjun-liu.github.io/sSNAPPY/">sSNAPPY</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/DT" class="external-link">DT</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/htmltools" class="external-link">htmltools</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-data">Load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h3>
<p>The example dataset used for this tutorial can be loaded with
<code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code> as shown below. It’s a subset of data retrieved from
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4928895/" class="external-link">Singhal
H et al. 2016</a>, where ER-positive primary breast cancer tumour
tissues collected from 12 patients were split into tissue fragments of
equal sizes for different treatments.</p>
<p>For this tutorial, we are only looking at the RNA-seq data from
samples that were treated with vehicle, R5020(progesterone) OR
E2(estrogen) + R5020 for 48 hrs. They were from 5 different patients,
giving rise to 15 samples in total. A more detailed description of the
dataset can be assessed through the help page
(<code><a href="../reference/logCPM_example.html">?logCPM_example</a></code> and <code><a href="../reference/metadata_example.html">?metadata_example</a></code>).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">)</span></span>
<span><span class="co"># check if samples included in the logCPM matrix and metadata dataframe are identical</span></span>
<span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setequal</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span>, <span class="va">metadata_example</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View sample metadata</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="va">metadata_example</span>,  filter <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-b3715ca449cbef139039" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-b3715ca449cbef139039">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;N2&quot;,&quot;N3&quot;,&quot;P4&quot;,&quot;P5&quot;,&quot;P6&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Vehicle&quot;,&quot;E2&quot;,&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020_N2_48&quot;,&quot;E2+R5020_N3_48&quot;,&quot;E2+R5020_P4_48&quot;,&quot;E2+R5020_P5_48&quot;,&quot;E2+R5020_P6_48&quot;,&quot;R5020_N2_48&quot;,&quot;R5020_N3_48&quot;,&quot;R5020_P4_48&quot;,&quot;R5020_P5_48&quot;,&quot;R5020_P6_48&quot;,&quot;Vehicle_N2_48&quot;,&quot;Vehicle_N3_48&quot;,&quot;Vehicle_P4_48&quot;,&quot;Vehicle_P5_48&quot;,&quot;Vehicle_P6_48&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Negative&quot;,&quot;Positive&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["N2","N2","N2","N3","N3","N3","P4","P4","P4","P5","P5","P5","P6","P6","P6"],["Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020"],["Vehicle_N2_48","E2+R5020_N2_48","R5020_N2_48","Vehicle_N3_48","E2+R5020_N3_48","R5020_N3_48","Vehicle_P4_48","E2+R5020_P4_48","R5020_P4_48","Vehicle_P5_48","E2+R5020_P5_48","R5020_P5_48","Vehicle_P6_48","E2+R5020_P6_48","R5020_P6_48"],["Negative","Negative","Negative","Negative","Negative","Negative","Positive","Positive","Positive","Positive","Positive","Positive","Positive","Positive","Positive"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>patient<\/th>\n      <th>treatment<\/th>\n      <th>sample<\/th>\n      <th>PR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="section level2">
<h2 id="ssnappy-workflow">
<code>sSNAPPY</code> workflow<a class="anchor" aria-label="anchor" href="#ssnappy-workflow"></a>
</h2>
<div class="section level3">
<h3 id="compute-weighted-single-sample-logfcs-sslogfcs">Compute weighted single-sample logFCs (ssLogFCs)<a class="anchor" aria-label="anchor" href="#compute-weighted-single-sample-logfcs-sslogfcs"></a>
</h3>
<p>It is expected that the logCPM matrix will be filtered to remove
undetectable genes and normalised to correct for library sizes or other
systematic artefacts, such as gene length or GC contents, prior to
applying the <code>sSNAPPY</code> workflow. Filtration and normalisation
have been performed on the example dataset.</p>
<p>Before single-sample logFCs (ssLogFCs) can be computed, row names of
the logCPM matrix need to be converted to <code>entrez ID</code>. This
is because all the pathway topology information retrieved will be in
<code>entrez ID</code>. The conversion can be achieved through
bioconductor packages <code>AnnotationHub</code> and
<code>ensembldb</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span></span></code></pre></div>
<p>To compute the ssLogFCs, samples must be in matching pairs. In our
example data, treated samples were matched to the corresponding control
samples derived from the same patients. Therefore the
<code>factor</code> parameter of the <code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code>
functions will be set to be “patient”. The function also requires the
control treatment level to be specified, which was “Vehicle” in this
case.</p>
<p><code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code> requires both the logCPM matrix and
sample metadata as input. The column names of the logCPM matrix should
be sample names, matching a column called <code>sample</code> in the
metadata. The metadata must also contain a treatment column, and a
column corresponding to the <code>factor</code> parameter (i.e.patient
in this case).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#compute weighted single sample logFCs</span></span>
<span><span class="va">weightedFC</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/weight_ss_fc.html">weight_ss_fc</a></span><span class="op">(</span><span class="va">logCPM_example</span>, metadata <span class="op">=</span> <span class="va">metadata_example</span>,</span>
<span>factor <span class="op">=</span> <span class="st">"patient"</span>, control <span class="op">=</span> <span class="st">"Vehicle"</span><span class="op">)</span></span></code></pre></div>
<p>The <code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code> function firstly computes raw
ssLogFCs for each gene by subtracting logCPM values of the control
sample from the logCPM values of treated samples for each patient.</p>
<p>It has been demonstrated previously that in RNA-seq data, lowly
expressed genes turn to have a larger variance, which is also
demonstrated by the plots below. To reduce the impact of this artefact,
<code>weight_ss_fc</code> also weights each ssLogFCs by estimating the
relationship between the variance in ssLogFCs and mean logCPM, and
defining the gene-wise weight to be the inverse of the predicted
variance of that gene’s mean logCPM value.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perSample_FC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">$</span><span class="va">patient</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">temp</span> <span class="op">&lt;-</span> <span class="va">logCPM_example</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span>,<span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span>, <span class="va">x</span><span class="op">)</span><span class="op">]</span> </span>
<span>    <span class="va">ratio</span> <span class="op">&lt;-</span> <span class="va">temp</span><span class="op">[</span>, <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>, <span class="st">"Vehicle"</span>, negate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span> <span class="op">-</span> <span class="va">temp</span><span class="op">[</span>, <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>, <span class="st">"Vehicle"</span><span class="op">)</span><span class="op">]</span> </span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">cbind</span>,<span class="va">.</span><span class="op">)</span></span>
<span><span class="va">aveCPM</span> <span class="op">&lt;-</span> <span class="va">logCPM_example</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span>,<span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/enframe.html" class="external-link">enframe</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"gene_id"</span>, </span>
<span>            value <span class="op">=</span> <span class="st">"aveCPM"</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">perSample_FC</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"gene_id"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="op">-</span><span class="st">"gene_id"</span>,</span>
<span>                 names_to <span class="op">=</span> <span class="st">"name"</span>,</span>
<span>                 values_to <span class="op">=</span> <span class="st">"logFC"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">aveCPM</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">aveCPM</span>, <span class="va">logFC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"sslogFC"</span>, </span>
<span>         x <span class="op">=</span> <span class="st">"Average logCPM"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">perSample_FC</span><span class="op">)</span>,</span>
<span>    variance <span class="op">=</span> <span class="va">perSample_FC</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fl">1</span>,<span class="va">var</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">aveCPM</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">aveCPM</span>, <span class="va">variance</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html" class="external-link">geom_smooth</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"loess"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"Variance in ssLogFCs"</span>, </span>
<span>         x <span class="op">=</span> <span class="st">"Average logCPM"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span><span class="op">)</span></span></code></pre></div>
<p><img src="sSNAPPY_files/figure-html/lowess-1.png" width="768"></p>
<p>The output of the <code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code> function is a list with
two element, where one is the weighted ssLogFCs matrix and the other is
a vector of gene-wise weights.</p>
</div>
<div class="section level3">
<h3 id="retrieve-pathway-topologies-in-the-required-format">Retrieve pathway topologies in the required format<a class="anchor" aria-label="anchor" href="#retrieve-pathway-topologies-in-the-required-format"></a>
</h3>
<p><em>sSNAPPY</em> adopts the pathway perturbation scoring algorithm
proposed in <em><a href="https://bioconductor.org/packages/3.16/SPIA" class="external-link">SPIA</a></em>, which
makes use of gene-set topologies and gene-gene interaction to propagate
pathway genes’ logFCs down the topologies to compute pathway
perturbation scores, where signs of scores reflect the potential
directions of changes.</p>
<p>Therefore, pathway topology information needs to be firstly retrieved
from your chosen database and converted to weight adjacency matrices,
the format required to apply the scoring algorithm.</p>
<p>This step is achieved through a chain of functions that are part of
the <em><a href="https://bioconductor.org/packages/3.16/grapghite" class="external-link">grapghite</a></em>,
which have been nested into one simple function in this package called
<code><a href="../reference/retrieve_topology.html">retrieve_topology()</a></code>. Databases that are currently supported
are:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sales-lab/graphite" class="external-link">graphite</a></span><span class="op">)</span></span>
<span><span class="fu">graphite</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graphite/man/pathwayDatabases.html" class="external-link">pathwayDatabases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">species</span> <span class="op">==</span>  <span class="st">"hsapiens"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">pander</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pander/man/pander.html" class="external-link">pander</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table style="width:36%;" class="table">
<colgroup>
<col width="15%">
<col width="20%">
</colgroup>
<thead><tr class="header">
<th align="center">species</th>
<th align="center">database</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">kegg</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">panther</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">pathbank</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">pharmgkb</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">reactome</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">smpdb</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">wikipathways</td>
</tr>
</tbody>
</table>
<p>The retrieved topology information will be a list where each element
corresponds a pathway. It’s recommended to save the list as a file so
this step only needs to be performed once for each database.</p>
<p>This vignette chose <em>KEGG</em> pathways as an example.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsTopology</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/retrieve_topology.html">retrieve_topology</a></span><span class="op">(</span>database <span class="op">=</span> <span class="st">"kegg"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsTopology</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Glycolysis / Gluconeogenesis"            </span></span>
<span><span class="co">## [2] "Citrate cycle (TCA cycle)"               </span></span>
<span><span class="co">## [3] "Pentose phosphate pathway"               </span></span>
<span><span class="co">## [4] "Pentose and glucuronate interconversions"</span></span>
<span><span class="co">## [5] "Fructose and mannose metabolism"         </span></span>
<span><span class="co">## [6] "Galactose metabolism"</span></span></code></pre>
<p>If only selected pathways are of interest to your research, it’s
possible to only retrieve the topologies of those pathways by specifying
the pathway names.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsTopology_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/retrieve_topology.html">retrieve_topology</a></span><span class="op">(</span></span>
<span>  database <span class="op">=</span> <span class="st">"kegg"</span>,</span>
<span>  pathwayName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"Glycolysis / Gluconeogenesis"</span>, </span>
<span>    <span class="st">"Citrate cycle (TCA cycle)"</span>,</span>
<span>    <span class="st">"Pentose phosphate pathway"</span></span>
<span>    <span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsTopology_sub</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Glycolysis / Gluconeogenesis" "Citrate cycle (TCA cycle)"   </span></span>
<span><span class="co">## [3] "Pentose phosphate pathway"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="score-single-sample-pathway-perturbation">Score single sample pathway perturbation<a class="anchor" aria-label="anchor" href="#score-single-sample-pathway-perturbation"></a>
</h3>
<p>Once the expression matrix, sample metadata and pathway topologies
are all ready, gene-wise single-sample perturbation scores can be
computed:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genePertScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/raw_gene_pert.html">raw_gene_pert</a></span><span class="op">(</span><span class="va">weightedFC</span><span class="op">$</span><span class="va">logFC</span>, <span class="va">gsTopology</span><span class="op">)</span></span></code></pre></div>
<p>and summed to derive pathway perturbation scores for each pathway in
each treated samples.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathwayPertScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pathway_pert.html">pathway_pert</a></span><span class="op">(</span><span class="va">genePertScore</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pathwayPertScore</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##           sample           tA                                   gs_name</span></span>
<span><span class="co">## 1 E2+R5020_N2_48  0.006751152 EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 2    R5020_N2_48  0.006744556 EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 3 E2+R5020_N3_48  0.002683974 EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 4    R5020_N3_48 -0.003974106 EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 5 E2+R5020_P4_48  0.004130870 EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 6    R5020_P4_48  0.015279018 EGFR tyrosine kinase inhibitor resistance</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="generate-null-distributions-of-perturbation-scores">Generate null distributions of perturbation scores<a class="anchor" aria-label="anchor" href="#generate-null-distributions-of-perturbation-scores"></a>
</h3>
<p>To derive the empirical p-values for each single-sample perturbation
scores and normalize the raw scores for comparing overall treatment
effects, null distributions of scores for each pathway are generated
through a sample-label permutation approach.</p>
<p>For each round of permutation, sample labels are randomly shuffled to
derive the permuted ssLogFCs, which are then used to score pathway
perturbation. We recommend performing a minimum of 1000 rounds of
permutation, which means at least 8 samples are required.</p>
<p>The <code><a href="../reference/generate_permuted_scores.html">generate_permuted_scores()</a></code> function does not require
sample metadata but the number of treatments in the study design,
including the control treatment, need to be specified. In this example
data, the number of treatmentS was 3 (Vehicle, E2, and E2+R5020).</p>
<p>The output of the <code><a href="../reference/generate_permuted_scores.html">generate_permuted_scores()</a></code> function is
a list where each element is a vector of permuted perturbation scores
for a pathway.</p>
<p>The permutation step relies on the parallel computing feature
provided by <em><a href="https://bioconductor.org/packages/3.16/BiocParallel" class="external-link">BiocParallel</a></em>.
You can choose to customize the parallel back-end or stick with the
default one returned by <code><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">BiocParallel::bpparam()</a></code>. Depending
on the size of the data, this step can take some time to complete. If
the sample size is large, we recommend users consider performing this
step on an HPC.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">permutedScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_permuted_scores.html">generate_permuted_scores</a></span><span class="op">(</span></span>
<span>  expreMatrix  <span class="op">=</span> <span class="va">logCPM_example</span>, </span>
<span>  numOfTreat <span class="op">=</span> <span class="fl">3</span>, NB <span class="op">=</span> <span class="fl">1000</span>, </span>
<span>  gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>  weight <span class="op">=</span> <span class="va">weightedFC</span><span class="op">$</span><span class="va">weight</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="test-significant-perturbation-on">Test significant perturbation on<a class="anchor" aria-label="anchor" href="#test-significant-perturbation-on"></a>
</h3>
<div class="section level4">
<h4 id="single-sample-level">single-sample level<a class="anchor" aria-label="anchor" href="#single-sample-level"></a>
</h4>
<p>After the empirical null distributions are generated, the median and
mad of each distribution will be calculated for each pathway to convert
the test single-sample perturbation scores derived from the
<code>compute_perturbation_score()</code> function to robust z-scores:
<span class="math display">\[(Score - Median)/MAD\]</span></p>
<p>Two-sided p-values associated with each robust z-scores are also
computed and will be corrected for multiple-testing using a user-defined
approach. The default is <code>fdr</code>.</p>
<p>The <code><a href="../reference/normalise_by_permu.html">normalise_by_permu()</a></code> function requires the test
perturbation scores and permuted perturbation scores as input.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normalisedScores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalise_by_permu.html">normalise_by_permu</a></span><span class="op">(</span><span class="va">permutedScore</span>, <span class="va">ssPertScore</span><span class="op">)</span></span></code></pre></div>
<p>Since the permutation step takes a long time to run and the output is
too large to be included as part of the package, the results of the
<code>normalise_by_permu</code> step has been pre-computed and can be
loaded with:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"normalisedScores.rda"</span>, package <span class="op">=</span> <span class="st">"sSNAPPY"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Pathways that were significant perturbed within individual samples
are:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">adjPvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_at</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/vars.html" class="external-link">vars</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"gs_name"</span><span class="op">)</span><span class="op">)</span>, <span class="va">as.factor</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_if</a></span><span class="op">(</span><span class="va">is.numeric</span>, <span class="va">sprintf</span>, fmt <span class="op">=</span> <span class="st">'%#.4f'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Direction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">robustZ</span> <span class="op">&lt;</span> <span class="fl">0</span>, <span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>        <span class="va">sample</span>, <span class="va">patient</span>, Treatment <span class="op">=</span> <span class="va">treatment</span>,</span>
<span>        `Perturbation Score` <span class="op">=</span> <span class="va">robustZ</span>, <span class="va">Direction</span>,</span>
<span>        `Gene-set name` <span class="op">=</span> <span class="va">gs_name</span>, </span>
<span>        `P-value` <span class="op">=</span> <span class="va">pvalue</span>, </span>
<span>        FDR <span class="op">=</span> <span class="va">adjPvalue</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span></span>
<span>        filter <span class="op">=</span> <span class="st">"top"</span>, </span>
<span>        options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>            columnDefs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="st">"Direction"</span>, visible <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span>        <span class="op">)</span>, </span>
<span>        caption <span class="op">=</span> <span class="fu">htmltools</span><span class="fu">::</span><span class="va"><a href="https://rstudio.github.io/htmltools/reference/builder.html" class="external-link">tags</a></span><span class="op">$</span><span class="fu">caption</span><span class="op">(</span></span>
<span>                  <span class="fu">htmltools</span><span class="fu">::</span><span class="fu"><a href="https://rstudio.github.io/htmltools/reference/builder.html" class="external-link">em</a></span><span class="op">(</span></span>
<span>                      <span class="st">"Pathways that were significant perturbed within individual samples."</span><span class="op">)</span></span>
<span>              <span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/formatCurrency.html" class="external-link">formatStyle</a></span><span class="op">(</span></span>
<span>        <span class="st">'Perturbation Score'</span>, <span class="st">'Direction'</span>,</span>
<span>        color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DT/man/styleInterval.html" class="external-link">styleEqual</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-aabb62a9301552364260" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-aabb62a9301552364260">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020_N2_48&quot;,&quot;R5020_N2_48&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;N2&quot;,&quot;N3&quot;,&quot;P4&quot;,&quot;P5&quot;,&quot;P6&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Vehicle&quot;,&quot;E2&quot;,&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Amoebiasis&quot;,&quot;Growth hormone synthesis, secretion and action&quot;,&quot;PI3K-Akt signaling pathway&quot;,&quot;TGF-beta signaling pathway&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","caption":"<caption>\n  <em>Pathways that were significant perturbed within individual samples.<\/em>\n<\/caption>","data":[["1","2","3","4","5","6","7","8"],["E2+R5020_N2_48","E2+R5020_N2_48","E2+R5020_N2_48","E2+R5020_N2_48","R5020_N2_48","R5020_N2_48","R5020_N2_48","R5020_N2_48"],["N2","N2","N2","N2","N2","N2","N2","N2"],["E2+R5020","E2+R5020","E2+R5020","E2+R5020","R5020","R5020","R5020","R5020"],["3.4728","-9.3778","3.7926","3.6961","3.6920","-10.4028","4.1352","3.7838"],["Activation","Inhibited","Activation","Activation","Activation","Inhibited","Activation","Activation"],["PI3K-Akt signaling pathway","TGF-beta signaling pathway","Growth hormone synthesis, secretion and action","Amoebiasis","PI3K-Akt signaling pathway","TGF-beta signaling pathway","Growth hormone synthesis, secretion and action","Amoebiasis"],["0.0005","0.0000","0.0001","0.0002","0.0002","0.0000","0.0000","0.0002"],["0.0278","0.0000","0.0158","0.0158","0.0120","0.0000","0.0038","0.0111"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>patient<\/th>\n      <th>Treatment<\/th>\n      <th>Perturbation Score<\/th>\n      <th>Direction<\/th>\n      <th>Gene-set name<\/th>\n      <th>P-value<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":5,"visible":false},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[5]; $(this.api().cell(row, 4).node()).css({'color':value == \"Inhibited\" ? \"blue\" : value == \"Activation\" ? \"red\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script><div class="section level5">
<h5 id="visualise-overlap-between-gene-sets-as-networks">Visualise overlap between gene-sets as networks<a class="anchor" aria-label="anchor" href="#visualise-overlap-between-gene-sets-as-networks"></a>
</h5>
<p>To visualise significantly perturbed biological pathways as a
network, where edges between gene-sets reflect how much overlap those
two gene-sets share, we can use the <code>plot_gs_network</code>
function. The function can take <code><a href="../reference/normalise_by_permu.html">normalise_by_permu()</a></code>’s
output, or a subset of it as its direct input.</p>
<p>Nodes in the network plots could be colored by the predicted
direction of perturbation (i.e.. sign of robust z-score):</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pl</span> <span class="op">&lt;-</span> <span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">adjPvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>    <span class="va">plot_gs_network</span>, </span>
<span>    <span class="co"># layout = "dh",</span></span>
<span>    gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>    colorBy <span class="op">=</span> <span class="st">"robustZ"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="va">x</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>        <span class="op">)</span> <span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span></span>
<span>    plotlist <span class="op">=</span> <span class="va">pl</span>, </span>
<span>    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pl</span><span class="op">)</span>,</span>
<span>    label_size <span class="op">=</span> <span class="fl">8</span>,</span>
<span>    nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/sigGS_nt_zscore-1.png" alt="*Pathways significantly perturbed in individual samples, where gene-sets were colored by pathways' directions of changes.*" width="960"><p class="caption">
<em>Pathways significantly perturbed in individual samples, where
gene-sets were colored by pathways’ directions of changes.</em>
</p>
</div>
<p>Or p-values:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pl</span> <span class="op">&lt;-</span> <span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">adjPvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>        <span class="va">plot_gs_network</span>, </span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorBy <span class="op">=</span> <span class="st">"pvalue"</span>, </span>
<span>        color_lg_title <span class="op">=</span> <span class="st">"P-value"</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="va">x</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>        <span class="op">)</span> <span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span></span>
<span>    plotlist <span class="op">=</span> <span class="va">pl</span>, </span>
<span>    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pl</span><span class="op">)</span>,</span>
<span>    label_size <span class="op">=</span> <span class="fl">8</span>,</span>
<span>    nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/sigGS_nt_pvalue-1.png" alt="*Pathways significantly perturbed in individual samples, where gene-sets were colored by pathways' p-values*" width="960"><p class="caption">
<em>Pathways significantly perturbed in individual samples, where
gene-sets were colored by pathways’ p-values</em>
</p>
</div>
<p>The function allows you to customize the layout, colour, edge
transparency and other aesthetics of the graph. More information can be
found on the help page (<code><a href="../reference/plot_gs_network.html">?plot_gs_network</a></code>). The output of
the graph is a <code>ggplot</code> object and the theme of it can be
changed just as any other <code>ggplot</code> figures.</p>
</div>
</div>
<div class="section level4">
<h4 id="treatment-level">treatment-level<a class="anchor" aria-label="anchor" href="#treatment-level"></a>
</h4>
<p>In addition to testing pathway perturbations at single-sample level,
normalised perturbation scores can also be used to model mean treatment
effects within a group, such as within each treatment group. An
advantage of this method is that it has a high level of flexibility that
allows us to incorporate confounding factors as co-factors or
co-variates to offset their effects.</p>
<p>For example, in the example data-set, samples were collected from
patients with different progesterone receptor (PR) statuses. Knowing
that PR status would affect how tumour tissues responded to estrogen
and/or progesterone treatments, we can include it as a cofactor to
offset its confounding effect.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co">#.["Estrogen signaling pathway"] %&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">robustZ</span> <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">treatment</span> <span class="op">+</span> <span class="va">PR</span>, data <span class="op">=</span> <span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">summary</span><span class="op">)</span></span>
<span><span class="va">treat_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>, </span>
<span>  <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">fit</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">coefficients</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="va">.</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>,<span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Estimate</span>, pvalue <span class="op">=</span> <span class="va">`Pr(&gt;|t|)`</span> <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"Treatment"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>        gs_name <span class="op">=</span> <span class="va">x</span>, </span>
<span>        FDR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">pvalue</span>, <span class="st">"fdr"</span><span class="op">)</span>, </span>
<span>        Treatment <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html" class="external-link">str_remove_all</a></span><span class="op">(</span><span class="va">Treatment</span>, <span class="st">"treatment"</span><span class="op">)</span></span>
<span>      <span class="op">)</span> </span>
<span>  <span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<p>Results from the linear modelling revealed pathways that were on
average perturbed due to each treatment:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_at</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/vars.html" class="external-link">vars</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Treatment"</span>, <span class="st">"gs_name"</span><span class="op">)</span><span class="op">)</span>, <span class="va">as.factor</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_if</a></span><span class="op">(</span><span class="va">is.numeric</span>, <span class="va">sprintf</span>, fmt <span class="op">=</span> <span class="st">'%#.4f'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Direction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">Estimate</span> <span class="op">&lt;</span> <span class="fl">0</span>, <span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>        <span class="va">Treatment</span>, `Perturbation Score` <span class="op">=</span> <span class="va">Estimate</span>, <span class="va">Direction</span>,</span>
<span>        `Gene-set name` <span class="op">=</span> <span class="va">gs_name</span>, </span>
<span>        `P-value` <span class="op">=</span> <span class="va">pvalue</span>, </span>
<span>        <span class="va">FDR</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span></span>
<span>        filter <span class="op">=</span> <span class="st">"top"</span>, </span>
<span>        options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>            columnDefs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="st">"Direction"</span>, visible <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span>        <span class="op">)</span>, </span>
<span>        caption <span class="op">=</span> <span class="fu">htmltools</span><span class="fu">::</span><span class="va"><a href="https://rstudio.github.io/htmltools/reference/builder.html" class="external-link">tags</a></span><span class="op">$</span><span class="fu">caption</span><span class="op">(</span></span>
<span>                  <span class="fu">htmltools</span><span class="fu">::</span><span class="fu"><a href="https://rstudio.github.io/htmltools/reference/builder.html" class="external-link">em</a></span><span class="op">(</span></span>
<span>                      <span class="st">"Pathways that were significant perturbed within each treatment group."</span><span class="op">)</span></span>
<span>              <span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/formatCurrency.html" class="external-link">formatStyle</a></span><span class="op">(</span></span>
<span>        <span class="st">'Perturbation Score'</span>, <span class="st">'Direction'</span>,</span>
<span>        color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DT/man/styleInterval.html" class="external-link">styleEqual</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-5e2b419453e4a7567bcb" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-5e2b419453e4a7567bcb">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Aldosterone synthesis and secretion&quot;,&quot;Cellular senescence&quot;,&quot;Cholinergic synapse&quot;,&quot;ECM-receptor interaction&quot;,&quot;Estrogen signaling pathway&quot;,&quot;Focal adhesion&quot;,&quot;Human papillomavirus infection&quot;,&quot;Malaria&quot;,&quot;Melanoma&quot;,&quot;Mineral absorption&quot;,&quot;Neutrophil extracellular trap formation&quot;,&quot;Ovarian steroidogenesis&quot;,&quot;PI3K-Akt signaling pathway&quot;,&quot;Prostate cancer&quot;,&quot;Ras signaling pathway&quot;,&quot;Renin secretion&quot;,&quot;Small cell lung cancer&quot;,&quot;Staphylococcus aureus infection&quot;,&quot;Tight junction&quot;,&quot;Toxoplasmosis&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","caption":"<caption>\n  <em>Pathways that were significant perturbed within each treatment group.<\/em>\n<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35"],["E2+R5020","R5020","E2+R5020","E2+R5020","R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","E2+R5020","R5020"],["0.9674","1.4237","-0.8339","-1.1795","-0.7286","1.4260","1.4672","1.6127","1.6185","1.9071","1.2505","1.6861","0.8677","1.0799","1.1781","1.4798","2.0158","2.0295","-0.6745","-0.6729","1.1831","1.0498","2.1298","2.1809","1.2865","1.4512","1.7780","1.5990","-1.4120","1.0909","0.5081","0.4781","-1.1317","-1.1117","-1.0020"],["Activation","Activation","Inhibited","Inhibited","Inhibited","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Inhibited","Inhibited","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Inhibited","Activation","Activation","Activation","Inhibited","Inhibited","Inhibited"],["Aldosterone synthesis and secretion","Aldosterone synthesis and secretion","Cellular senescence","Cholinergic synapse","Cholinergic synapse","ECM-receptor interaction","Estrogen signaling pathway","Estrogen signaling pathway","Focal adhesion","Focal adhesion","Human papillomavirus infection","Human papillomavirus infection","Malaria","Malaria","Melanoma","Melanoma","Mineral absorption","Mineral absorption","Neutrophil extracellular trap formation","Neutrophil extracellular trap formation","Ovarian steroidogenesis","Ovarian steroidogenesis","PI3K-Akt signaling pathway","PI3K-Akt signaling pathway","Prostate cancer","Prostate cancer","Ras signaling pathway","Ras signaling pathway","Renin secretion","Small cell lung cancer","Staphylococcus aureus infection","Staphylococcus aureus infection","Tight junction","Toxoplasmosis","Toxoplasmosis"],["0.0160","0.0024","0.0163","0.0013","0.0151","0.0142","0.0015","0.0009","0.0264","0.0130","0.0219","0.0055","0.0343","0.0138","0.0083","0.0026","0.0018","0.0017","0.0347","0.0350","0.0256","0.0407","0.0341","0.0311","0.0302","0.0184","0.0336","0.0496","0.0117","0.0154","0.0385","0.0479","0.0052","0.0123","0.0196"],["0.0160","0.0047","0.0327","0.0026","0.0151","0.0285","0.0015","0.0015","0.0264","0.0261","0.0219","0.0110","0.0343","0.0276","0.0083","0.0052","0.0018","0.0018","0.0350","0.0350","0.0407","0.0407","0.0341","0.0341","0.0302","0.0302","0.0496","0.0496","0.0234","0.0308","0.0479","0.0479","0.0104","0.0196","0.0196"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Treatment<\/th>\n      <th>Perturbation Score<\/th>\n      <th>Direction<\/th>\n      <th>Gene-set name<\/th>\n      <th>P-value<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":3,"visible":false},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[3]; $(this.api().cell(row, 2).node()).css({'color':value == \"Inhibited\" ? \"blue\" : value == \"Activation\" ? \"red\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script><p>It is not surprising to see that the estrogen signalling pathway was
significantly activated under both R5020 and E2+R5020 treatments.</p>
<div class="section level5">
<h5 id="visualise-genes-contributions-to-pathway-perturbation">Visualise genes’ contributions to pathway perturbation<a class="anchor" aria-label="anchor" href="#visualise-genes-contributions-to-pathway-perturbation"></a>
</h5>
<p>If there’s a specific pathway that we would like to dig deeper into
and explore which pathway genes played a key role in its perturbation,
for example, activation of the “Ovarian steroidogenesis”, we can plot
the gene-level perturbation scores of all genes contained in that
pathway as a heatmap and annotate each column (ie. each sample) by the
direction of pathway perturbation in that sample or any other sample
metadata using the <code><a href="../reference/plot_gene_contribution.html">plot_gene_contribution()</a></code> function.</p>
<p>From the heatmap below that we can see that the activation of this
pathway was driven by a few dominating genes.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_gene_contribution.html">plot_gene_contribution</a></span><span class="op">(</span></span>
<span>    genePertScore <span class="op">=</span> <span class="va">genePertScore</span>, gsToPlot <span class="op">=</span> <span class="st">"Ovarian steroidogenesis"</span>, metadata <span class="op">=</span> <span class="va">metadata_example</span>, </span>
<span>    annotation_attribute <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pathwayPertScore"</span>, <span class="st">"treatment"</span><span class="op">)</span>, pathwayPertScore <span class="op">=</span> <span class="va">pathwayPertScore</span>, </span>
<span>    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.0001</span>, <span class="fl">0.0001</span>, length.out <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>,</span>
<span>    show_rownames <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-5-1.png" alt="*Gene-level perturbation scores of all genes in the Ovarian steroidogenesis gene-set. The activation of pathway was driven by a few dominating genes.*" width="768"><p class="caption">
<em>Gene-level perturbation scores of all genes in the Ovarian
steroidogenesis gene-set. The activation of pathway was driven by a few
dominating genes.</em>
</p>
</div>
<p>By default, genes’ entrez IDs are used and plotted as row names,
which may not be very informative. So the row names could be overwritten
by providing a <code>data.frame</code> mapping entrez IDs to other
identifiers through the <code>mapRownameTo</code> parameter.</p>
<p>Mapping between different gene identifiers could be achieved through
the <code>mapIDs()</code> function from the Bioconductor package <a href="https://bioconductor.org/packages/release/bioc/html/AnnotationDbi.html" class="external-link"><code>AnnotationDbi</code></a>.
But to reduce the compiling time of this vignette, mappings between
entrez IDs and gene names have been provided as a
<code>data.frame</code> called <code>entrez2name</code>.</p>
<p>Note that if the mapping information was provided and the mapping was
successful for some genes but not the others, only genes that have been
mapped successfully will be plotted.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"entrez2name.rda"</span>, package <span class="op">=</span> <span class="st">"sSNAPPY"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_gene_contribution.html">plot_gene_contribution</a></span><span class="op">(</span></span>
<span>    genePertScore <span class="op">=</span> <span class="va">genePertScore</span>, gsToPlot <span class="op">=</span> <span class="st">"Ovarian steroidogenesis"</span>, metadata <span class="op">=</span> <span class="va">metadata_example</span>, </span>
<span>    annotation_attribute <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pathwayPertScore"</span>, <span class="st">"treatment"</span><span class="op">)</span>, pathwayPertScore <span class="op">=</span> <span class="va">pathwayPertScore</span>,</span>
<span>    breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="op">-</span><span class="fl">0.0001</span>, <span class="fl">0.0001</span>, length.out <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>, </span>
<span>    fontsize_row <span class="op">=</span> <span class="fl">6</span>,</span>
<span>    mapEntrezID <span class="op">=</span> <span class="va">entrez2name</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-6-1.png" alt="*Gene-level perturbation scores of all genes in the Ovarian steroidogenesis gene-set.Genes playing the most important roles in pathway activation was INS, CGA, GNAS.*" width="768"><p class="caption">
<em>Gene-level perturbation scores of all genes in the Ovarian
steroidogenesis gene-set.Genes playing the most important roles in
pathway activation was INS, CGA, GNAS.</em>
</p>
</div>
</div>
<div class="section level5">
<h5 id="visualise-overlap-between-gene-sets-as-networks-1">Visualise overlap between gene-sets as networks<a class="anchor" aria-label="anchor" href="#visualise-overlap-between-gene-sets-as-networks-1"></a>
</h5>
<p>Results of group-level perturbation can also be visualised using the
<code><a href="../reference/plot_gs_network.html">plot_gs_network()</a></code> function. We just need to change the name
of the <code>Estimate</code> column we had in the <code>treat_sig</code>
object to <code>robustZ</code> to colour the networks by the predicted
directions of changes.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>robustZ <span class="op">=</span> <span class="va">Estimate</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/plot_gs_network.html">plot_gs_network</a></span><span class="op">(</span></span>
<span>       </span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorBy <span class="op">=</span> <span class="st">"robustZ"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-7-1.png" alt="*Pathways significantly perturbed by the E2+R5020 combintation treatment, where colors of nodes reflect pathways' directions of changes.*" width="960"><p class="caption">
<em>Pathways significantly perturbed by the E2+R5020 combintation
treatment, where colors of nodes reflect pathways’ directions of
changes.</em>
</p>
</div>
<p>By default, <code><a href="../reference/plot_gs_network.html">plot_gs_network()</a></code> function does not include
nodes that are not connected to any other nodes, which could be turnt
off by setting the <code>plotIsolated</code> parameter to TURE.</p>
</div>
<div class="section level5">
<h5 id="visualise-community-structure-in-the-gene-set-network">Visualise community structure in the gene-set network<a class="anchor" aria-label="anchor" href="#visualise-community-structure-in-the-gene-set-network"></a>
</h5>
<p>When a large number of pathways were perturbed, it is hard to answer
the question “What key biological processes were perturbed?” solely by
looking at all the pathway names. To solve that, we can use the
<code><a href="../reference/plot_community.html">plot_community()</a></code> function to apply a community detection
algorithm to the network we had above, and annotate each community by
the biological process that most pathways assigned to that community
belong to.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>robustZ <span class="op">=</span> <span class="va">Estimate</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/plot_community.html">plot_community</a></span><span class="op">(</span></span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorBy <span class="op">=</span> <span class="st">"community"</span>, </span>
<span>        color_lg_title <span class="op">=</span> <span class="st">"Community"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-8-1.png" alt="*Pathways significantly perturbed by the E2+R5020 combintation treatment, annotated by the main biological processes each pathways belong to.*" width="1152"><p class="caption">
<em>Pathways significantly perturbed by the E2+R5020 combintation
treatment, annotated by the main biological processes each pathways
belong to.</em>
</p>
</div>
<p><code>sSNAPPY</code> was built in with categorisations of
<em>KEGG</em> pathways. But to annotate pathways retrieved from other
databases, customised annotation <code>data.frame</code> could be
provided through the <code>gsAnnotation</code> parameter.</p>
</div>
<div class="section level5">
<h5 id="visualise-genes-included-in-perturbed-pathways-networks">Visualise genes included in perturbed pathways networks<a class="anchor" aria-label="anchor" href="#visualise-genes-included-in-perturbed-pathways-networks"></a>
</h5>
<p>If we want to not only know if two pathways are connected but also
the genes connecting those pathways, we can use the
<code><a href="../reference/plot_gs2gene.html">plot_gs2gene()</a></code> function instead:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>robustZ <span class="op">=</span> <span class="va">Estimate</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/plot_gs2gene.html">plot_gs2gene</a></span><span class="op">(</span></span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorGS_By <span class="op">=</span> <span class="st">"robustZ"</span>, </span>
<span>        label_Gene <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        GeneNode_size <span class="op">=</span> <span class="fl">1</span>, </span>
<span>        edgeAlpha <span class="op">=</span> <span class="fl">0.1</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-9-1.png" alt="*Pathways significantly perturbed by the E2+R5020 combintation treatment and all genes contained in those pathways.*" width="1152"><p class="caption">
<em>Pathways significantly perturbed by the E2+R5020 combintation
treatment and all genes contained in those pathways.</em>
</p>
</div>
<p>However, since there are a large number of genes in each pathway, the
plot above was quite messy and it was unrealistic to plot all genes’
names. So it is recommend to filter pathway genes by their perturbation
scores or logFCs.</p>
<p>For example, we can rank genes by the absolute values of their mean
single-sample logFCs and only focus on genes that were ranked in the top
200 of the list.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meanFC</span> <span class="op">&lt;-</span> <span class="va">weightedFC</span><span class="op">$</span><span class="va">logFC</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">.</span><span class="op">[</span>, <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, <span class="st">"E2"</span>, negate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">mean</span> <span class="op">)</span></span>
<span><span class="va">top200_gene</span> <span class="op">&lt;-</span> <span class="va">meanFC</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span>decreasing <span class="op">=</span> <span class="cn">TRUE</span>, <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">.</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">200</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">top200_FC</span> <span class="op">&lt;-</span> <span class="va">meanFC</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">.</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">top200_gene</span><span class="op">]</span></span></code></pre></div>
<p>When we provide genes’ logFCs as a named vector through the
<code>geneFC</code> parameter, only pathway genes with logFCs provided
will be plotted and gene nodes will be colored by genes’ directions of
changes. The names of the logFC vector must be genes’ entrez IDs in the
format of “ENTREZID:XXXX”, as pathway topology matrices retrieved
through <code><a href="../reference/retrieve_topology.html">retrieve_topology()</a></code> always use entrez IDs as
identifiers.</p>
<p>However, it is not going to be informative to label genes with their
entrez IDs. So the same as the <code><a href="../reference/plot_gene_contribution.html">plot_gene_contribution()</a></code>
function, we can provide a <code>data.frame</code> to match genes’
entrez IDs to our chosen identifiers through the
<code>mapEntrezID</code> parameter in the <code><a href="../reference/plot_gs2gene.html">plot_gs2gene()</a></code>
function too.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>robustZ <span class="op">=</span> <span class="va">Estimate</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/plot_gs2gene.html">plot_gs2gene</a></span><span class="op">(</span></span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorGS_By <span class="op">=</span> <span class="st">"robustZ"</span>, </span>
<span>        mapEntrezID <span class="op">=</span> <span class="va">entrez2name</span>, </span>
<span>        geneFC <span class="op">=</span> <span class="va">top200_FC</span>, </span>
<span>        edgeAlpha <span class="op">=</span> <span class="fl">0.3</span>, </span>
<span>        GsName_size <span class="op">=</span> <span class="fl">4</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-10-1.png" alt="*Pathways significantly perturbed by the E2+R5020 combintation treatment, and pathway genes with top 200 magnitudes of changes among all E2+R5020-treated samples. Both pathways and genes were colored by their directions of changes.*" width="960"><p class="caption">
<em>Pathways significantly perturbed by the E2+R5020 combintation
treatment, and pathway genes with top 200 magnitudes of changes among
all E2+R5020-treated samples. Both pathways and genes were colored by
their directions of changes.</em>
</p>
</div>
<p>We can also filters genes’ by their contributions to a pathway
perturbations.To help with that, <code>sSNAPPYT</code> provides an
option to rank genes’ perturbation scores within each sample for each
pathway.</p>
<p>If in a given pathway, both positive and negative gene-wise
perturbation scores exist, positive and negative scores are ranked
separately, where the larger a positive rank, the more the gene
contributed to the pathway’s activation, and the smaller a negative
rank, the more the gene contributed to the pathways’ inhibition.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneRank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rank_gene_pert.html">rank_gene_pert</a></span><span class="op">(</span><span class="va">genePertScore</span>, <span class="va">gsTopology</span><span class="op">)</span></span></code></pre></div>
<p>Depending on the biological question, it could be interesting to plot
only the pathway genes with the same directions of changes as the
pathway they belonged to, and ignoring those genes that were
antagonizing the pathway perturbation.</p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Sales G, Calura E, Cavalieri D, Romualdi C (2012). “graphite - a
Bioconductor package to convert pathway topology to gene network.” BMC
Bioinformatics. <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-13-20" class="external-link uri">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-13-20</a>.</li>
<li>Tarca, Adi Laurentiu et al. (2009). “A novel signaling pathway
impact analysis.” Bioinformatics vol. 25,1 : 75-82. <a href="doi:10.1093/bioinformatics/btn577" class="uri">doi:10.1093/bioinformatics/btn577</a>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.2 (2022-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] graphite_1.44.0  htmltools_0.5.4  DT_0.27          cowplot_1.1.1   </span></span>
<span><span class="co">##  [5] magrittr_2.0.3   forcats_1.0.0    stringr_1.5.0    dplyr_1.1.0     </span></span>
<span><span class="co">##  [9] purrr_1.0.1      readr_2.1.3      tidyr_1.3.0      tibble_3.1.8    </span></span>
<span><span class="co">## [13] ggplot2_3.4.0    tidyverse_1.3.2  sSNAPPY_1.3.3    BiocStyle_2.26.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] readxl_1.4.1                backports_1.4.1            </span></span>
<span><span class="co">##   [3] systemfonts_1.0.4           plyr_1.8.8                 </span></span>
<span><span class="co">##   [5] igraph_1.3.5                splines_4.2.2              </span></span>
<span><span class="co">##   [7] BiocParallel_1.32.5         crosstalk_1.2.0            </span></span>
<span><span class="co">##   [9] GenomeInfoDb_1.34.7         digest_0.6.31              </span></span>
<span><span class="co">##  [11] viridis_0.6.2               fansi_1.0.4                </span></span>
<span><span class="co">##  [13] memoise_2.0.1               googlesheets4_1.0.1        </span></span>
<span><span class="co">##  [15] tzdb_0.3.0                  limma_3.54.1               </span></span>
<span><span class="co">##  [17] Biostrings_2.66.0           graphlayouts_0.8.4         </span></span>
<span><span class="co">##  [19] modelr_0.1.10               matrixStats_0.63.0         </span></span>
<span><span class="co">##  [21] timechange_0.2.0            pkgdown_2.0.7              </span></span>
<span><span class="co">##  [23] colorspace_2.1-0            blob_1.2.3                 </span></span>
<span><span class="co">##  [25] rvest_1.0.3                 rappdirs_0.3.3             </span></span>
<span><span class="co">##  [27] ggrepel_0.9.2               textshaping_0.3.6          </span></span>
<span><span class="co">##  [29] haven_2.5.1                 xfun_0.36                  </span></span>
<span><span class="co">##  [31] crayon_1.5.2                RCurl_1.98-1.10            </span></span>
<span><span class="co">##  [33] jsonlite_1.8.4              graph_1.76.0               </span></span>
<span><span class="co">##  [35] glue_1.6.2                  polyclip_1.10-4            </span></span>
<span><span class="co">##  [37] gtable_0.3.1                gargle_1.2.1               </span></span>
<span><span class="co">##  [39] zlibbioc_1.44.0             XVector_0.38.0             </span></span>
<span><span class="co">##  [41] DelayedArray_0.24.0         BiocGenerics_0.44.0        </span></span>
<span><span class="co">##  [43] scales_1.2.1                pheatmap_1.0.12            </span></span>
<span><span class="co">##  [45] DBI_1.1.3                   edgeR_3.40.2               </span></span>
<span><span class="co">##  [47] Rcpp_1.0.10                 viridisLite_0.4.1          </span></span>
<span><span class="co">##  [49] bit_4.0.5                   stats4_4.2.2               </span></span>
<span><span class="co">##  [51] htmlwidgets_1.6.1           httr_1.4.4                 </span></span>
<span><span class="co">##  [53] RColorBrewer_1.1-3          ellipsis_0.3.2             </span></span>
<span><span class="co">##  [55] pkgconfig_2.0.3             farver_2.1.1               </span></span>
<span><span class="co">##  [57] sass_0.4.5                  dbplyr_2.3.0               </span></span>
<span><span class="co">##  [59] locfit_1.5-9.7              utf8_1.2.2                 </span></span>
<span><span class="co">##  [61] labeling_0.4.2              tidyselect_1.2.0           </span></span>
<span><span class="co">##  [63] rlang_1.0.6                 reshape2_1.4.4             </span></span>
<span><span class="co">##  [65] AnnotationDbi_1.60.0        munsell_0.5.0              </span></span>
<span><span class="co">##  [67] cellranger_1.1.0            tools_4.2.2                </span></span>
<span><span class="co">##  [69] cachem_1.0.6                cli_3.6.0                  </span></span>
<span><span class="co">##  [71] generics_0.1.3              RSQLite_2.2.20             </span></span>
<span><span class="co">##  [73] broom_1.0.3                 evaluate_0.20              </span></span>
<span><span class="co">##  [75] fastmap_1.1.0               yaml_2.3.7                 </span></span>
<span><span class="co">##  [77] ragg_1.2.5                  org.Hs.eg.db_3.16.0        </span></span>
<span><span class="co">##  [79] knitr_1.42                  bit64_4.0.5                </span></span>
<span><span class="co">##  [81] fs_1.6.0                    tidygraph_1.2.2            </span></span>
<span><span class="co">##  [83] pander_0.6.5                KEGGREST_1.38.0            </span></span>
<span><span class="co">##  [85] ggraph_2.1.0                nlme_3.1-160               </span></span>
<span><span class="co">##  [87] xml2_1.3.3                  compiler_4.2.2             </span></span>
<span><span class="co">##  [89] curl_5.0.0                  png_0.1-8                  </span></span>
<span><span class="co">##  [91] reprex_2.0.2                tweenr_2.0.2               </span></span>
<span><span class="co">##  [93] bslib_0.4.2                 stringi_1.7.12             </span></span>
<span><span class="co">##  [95] highr_0.10                  desc_1.4.2                 </span></span>
<span><span class="co">##  [97] lattice_0.20-45             Matrix_1.5-1               </span></span>
<span><span class="co">##  [99] vctrs_0.5.2                 pillar_1.8.1               </span></span>
<span><span class="co">## [101] lifecycle_1.0.3             BiocManager_1.30.19        </span></span>
<span><span class="co">## [103] jquerylib_0.1.4             bitops_1.0-7               </span></span>
<span><span class="co">## [105] GenomicRanges_1.50.2        R6_2.5.1                   </span></span>
<span><span class="co">## [107] bookdown_0.32               gridExtra_2.3              </span></span>
<span><span class="co">## [109] IRanges_2.32.0              codetools_0.2-18           </span></span>
<span><span class="co">## [111] MASS_7.3-58.1               assertthat_0.2.1           </span></span>
<span><span class="co">## [113] SummarizedExperiment_1.28.0 rprojroot_2.0.3            </span></span>
<span><span class="co">## [115] withr_2.5.0                 S4Vectors_0.36.1           </span></span>
<span><span class="co">## [117] GenomeInfoDbData_1.2.9      mgcv_1.8-41                </span></span>
<span><span class="co">## [119] parallel_4.2.2              hms_1.1.2                  </span></span>
<span><span class="co">## [121] grid_4.2.2                  rmarkdown_2.20             </span></span>
<span><span class="co">## [123] MatrixGenerics_1.10.0       googledrive_2.0.0          </span></span>
<span><span class="co">## [125] ggnewscale_0.4.8            ggforce_0.4.1              </span></span>
<span><span class="co">## [127] Biobase_2.58.0              lubridate_1.9.1</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://github.com/Wenjun-Liu" class="external-link">Wenjun Liu</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
