<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>sSNAPPY: Singel Sample directioNAl Pathway Perturbation analYsis • sSNAPPY</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="sSNAPPY: Singel Sample directioNAl Pathway Perturbation analYsis">
<meta property="og:description" content="sSNAPPY">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sSNAPPY</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.3.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/sSNAPPY.html">Get Started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/Wenjun-Liu/sSNAPPY" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="sSNAPPY_files/htmlwidgets-1.6.2/htmlwidgets.js"></script><link href="sSNAPPY_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="sSNAPPY_files/datatables-binding-0.30/datatables.js"></script><link href="sSNAPPY_files/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="sSNAPPY_files/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="sSNAPPY_files/dt-core-1.13.4/js/jquery.dataTables.min.js"></script><link href="sSNAPPY_files/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="sSNAPPY_files/nouislider-7.0.10/jquery.nouislider.min.js"></script><link href="sSNAPPY_files/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="sSNAPPY_files/selectize-0.12.0/selectize.min.js"></script><link href="sSNAPPY_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="sSNAPPY_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>sSNAPPY: Singel Sample directioNAl Pathway
Perturbation analYsis</h1>
                        <h4 data-toc-skip class="author">Wenjun Nora
Liu</h4>
            <address class="author_afil">
      Dame Roma Mitchell Cancer Research Laboratories, Adelaide Medical
School, University of
Adelaide<br><a class="author_email" href="mailto:#"></a><a href="mailto:wenjun.liu@adelaide.edu.au" class="email">wenjun.liu@adelaide.edu.au</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/Wenjun-Liu/sSNAPPY/blob/HEAD/vignettes/sSNAPPY.Rmd" class="external-link"><code>vignettes/sSNAPPY.Rmd</code></a></small>
      <div class="hidden name"><code>sSNAPPY.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette demonstrates how to use the package
<code>sSNAPPY</code> to compute directional single sample pathway
perturbation scores by incorporating pathway topologies, utilize sample
permutation to test the significance of individual scores and compare
average pathway activities across treatments.</p>
<p>The package also provides a function to visualise overlap between
pathway genes contained in perturbed biological pathways as network
plots.</p>
</div>
<div class="section level2">
<h2 id="to-get-ready">To get ready<a class="anchor" aria-label="anchor" href="#to-get-ready"></a>
</h2>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>The package <code>sSNAPPY</code> can be installed using the package
<code>BiocManager</code></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"BiocManager"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="co"># Other packages required in this vignette</span></span>
<span><span class="va">pkg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tidyverse"</span>, <span class="st">"magrittr"</span>, <span class="st">"ggplot2"</span>, <span class="st">"cowplot"</span>, <span class="st">"DT"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="va">pkg</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"sSNAPPY"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"htmltools"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wenjun-liu.github.io/sSNAPPY/">sSNAPPY</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/DT" class="external-link">DT</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/htmltools" class="external-link">htmltools</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-data">Load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h3>
<p>The example dataset used for this tutorial can be loaded with
<code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code> as shown below. It’s a subset of data retrieved from
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4928895/" class="external-link">Singhal
H et al. 2016</a>, where ER-positive primary breast cancer tumour
tissues collected from 12 patients were split into tissue fragments of
equal sizes for different treatments.</p>
<p>For this tutorial, we are only looking at the RNA-seq data from
samples that were treated with vehicle, R5020(progesterone) OR
E2(estrogen) + R5020 for 48 hrs. They were from 5 different patients,
giving rise to 15 samples in total. A more detailed description of the
dataset can be assessed through the help page
(<code><a href="../reference/logCPM_example.html">?logCPM_example</a></code> and <code><a href="../reference/metadata_example.html">?metadata_example</a></code>).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">)</span></span>
<span><span class="co"># check if samples included in the logCPM matrix and metadata dataframe are identical</span></span>
<span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setequal</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span>, <span class="va">metadata_example</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View sample metadata</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="va">metadata_example</span>,  filter <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-389b744c268236b6293e" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-389b744c268236b6293e">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;N2&quot;,&quot;N3&quot;,&quot;P4&quot;,&quot;P5&quot;,&quot;P6&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Vehicle&quot;,&quot;E2&quot;,&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020_N2_48&quot;,&quot;E2+R5020_N3_48&quot;,&quot;E2+R5020_P4_48&quot;,&quot;E2+R5020_P5_48&quot;,&quot;E2+R5020_P6_48&quot;,&quot;R5020_N2_48&quot;,&quot;R5020_N3_48&quot;,&quot;R5020_P4_48&quot;,&quot;R5020_P5_48&quot;,&quot;R5020_P6_48&quot;,&quot;Vehicle_N2_48&quot;,&quot;Vehicle_N3_48&quot;,&quot;Vehicle_P4_48&quot;,&quot;Vehicle_P5_48&quot;,&quot;Vehicle_P6_48&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Negative&quot;,&quot;Positive&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["N2","N2","N2","N3","N3","N3","P4","P4","P4","P5","P5","P5","P6","P6","P6"],["Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020"],["Vehicle_N2_48","E2+R5020_N2_48","R5020_N2_48","Vehicle_N3_48","E2+R5020_N3_48","R5020_N3_48","Vehicle_P4_48","E2+R5020_P4_48","R5020_P4_48","Vehicle_P5_48","E2+R5020_P5_48","R5020_P5_48","Vehicle_P6_48","E2+R5020_P6_48","R5020_P6_48"],["Negative","Negative","Negative","Negative","Negative","Negative","Positive","Positive","Positive","Positive","Positive","Positive","Positive","Positive","Positive"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>patient<\/th>\n      <th>treatment<\/th>\n      <th>sample<\/th>\n      <th>PR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="section level2">
<h2 id="ssnappy-workflow">
<code>sSNAPPY</code> workflow<a class="anchor" aria-label="anchor" href="#ssnappy-workflow"></a>
</h2>
<div class="section level3">
<h3 id="compute-weighted-single-sample-logfcs-sslogfcs">Compute weighted single-sample logFCs (ssLogFCs)<a class="anchor" aria-label="anchor" href="#compute-weighted-single-sample-logfcs-sslogfcs"></a>
</h3>
<p>It is expected that the logCPM matrix will be filtered to remove
undetectable genes and normalised to correct for library sizes or other
systematic artefacts, such as gene length or GC contents, prior to
applying the <code>sSNAPPY</code> workflow. Filtration and normalisation
have been performed on the example dataset.</p>
<p>Before single-sample logFCs (ssLogFCs) can be computed, row names of
the logCPM matrix need to be converted to <code>entrez ID</code>. This
is because all the pathway topology information retrieved will be in
<code>entrez ID</code>. The conversion can be achieved through
bioconductor packages <code>AnnotationHub</code> and
<code>ensembldb</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span></span></code></pre></div>
<p>To compute the ssLogFCs, samples must be in matching pairs. In our
example data, treated samples were matched to the corresponding control
samples derived from the same patients. Therefore the
<code>groupBy</code> parameter of the <code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code>
functions will be set to be “patient”.</p>
<p><code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code> requires both the logCPM matrix and
sample metadata as input. The column names of the logCPM matrix should
be sample names, matching a column in the metadata. Name of the column
with sample name will be provided as the <code>sampleColumn</code>
parameter. The function also requires the name of the metadata column
containing treatment information to be specified. The column with
treatment information must be a factor with the control treatment set to
be the reference level.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#compute weighted single sample logFCs</span></span>
<span><span class="va">weightedFC</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/weight_ss_fc.html">weight_ss_fc</a></span><span class="op">(</span><span class="va">logCPM_example</span>, metadata <span class="op">=</span> <span class="va">metadata_example</span>,</span>
<span>                           groupBy  <span class="op">=</span> <span class="st">"patient"</span>, sampleColumn <span class="op">=</span> <span class="st">"sample"</span>, </span>
<span>                           treatColumn <span class="op">=</span> <span class="st">"treatment"</span><span class="op">)</span></span></code></pre></div>
<p>The <code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code> function firstly computes raw
ssLogFCs for each gene by subtracting logCPM values of the control
sample from the logCPM values of treated samples for each patient.</p>
<p>It has been demonstrated previously that in RNA-seq data, lowly
expressed genes turn to have a larger variance, which is also
demonstrated by the plots below. To reduce the impact of this artefact,
<code>weight_ss_fc</code> also weights each ssLogFCs by estimating the
relationship between the variance in ssLogFCs and mean logCPM, and
defining the gene-wise weight to be the inverse of the predicted
variance of that gene’s mean logCPM value.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">perSample_FC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">$</span><span class="va">patient</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">temp</span> <span class="op">&lt;-</span> <span class="va">logCPM_example</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span>,<span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span>, <span class="va">x</span><span class="op">)</span><span class="op">]</span> </span>
<span>    <span class="va">ratio</span> <span class="op">&lt;-</span> <span class="va">temp</span><span class="op">[</span>, <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>, <span class="st">"Vehicle"</span>, negate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span> <span class="op">-</span> <span class="va">temp</span><span class="op">[</span>, <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">temp</span><span class="op">)</span>, <span class="st">"Vehicle"</span><span class="op">)</span><span class="op">]</span> </span>
<span><span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">cbind</span>,<span class="va">.</span><span class="op">)</span></span>
<span><span class="va">aveCPM</span> <span class="op">&lt;-</span> <span class="va">logCPM_example</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span>,<span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/enframe.html" class="external-link">enframe</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"gene_id"</span>, </span>
<span>            value <span class="op">=</span> <span class="st">"aveCPM"</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">perSample_FC</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"gene_id"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="op">-</span><span class="st">"gene_id"</span>,</span>
<span>                 names_to <span class="op">=</span> <span class="st">"name"</span>,</span>
<span>                 values_to <span class="op">=</span> <span class="st">"logFC"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">aveCPM</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">aveCPM</span>, <span class="va">logFC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"sslogFC"</span>, </span>
<span>         x <span class="op">=</span> <span class="st">"Average logCPM"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">perSample_FC</span><span class="op">)</span>,</span>
<span>    variance <span class="op">=</span> <span class="va">perSample_FC</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fl">1</span>,<span class="va">var</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">aveCPM</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">aveCPM</span>, <span class="va">variance</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html" class="external-link">geom_smooth</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"loess"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>y <span class="op">=</span> <span class="st">"Variance in ssLogFCs"</span>, </span>
<span>         x <span class="op">=</span> <span class="st">"Average logCPM"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span><span class="op">)</span></span></code></pre></div>
<p><img src="sSNAPPY_files/figure-html/lowess-1.png" width="768"></p>
<p>The output of the <code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code> function is a list with
two element, where one is the weighted ssLogFCs matrix and the other is
a vector of gene-wise weights.</p>
</div>
<div class="section level3">
<h3 id="retrieve-pathway-topologies-in-the-required-format">Retrieve pathway topologies in the required format<a class="anchor" aria-label="anchor" href="#retrieve-pathway-topologies-in-the-required-format"></a>
</h3>
<p><em>sSNAPPY</em> adopts the pathway perturbation scoring algorithm
proposed in <em><a href="https://bioconductor.org/packages/3.18/SPIA" class="external-link">SPIA</a></em>, which
makes use of gene-set topologies and gene-gene interaction to propagate
pathway genes’ logFCs down the topologies to compute pathway
perturbation scores, where signs of scores reflect the potential
directions of changes.</p>
<p>Therefore, pathway topology information needs to be firstly retrieved
from your chosen database and converted to weight adjacency matrices,
the format required to apply the scoring algorithm.</p>
<p>This step is achieved through a chain of functions that are part of
the <em><a href="https://bioconductor.org/packages/3.18/grapghite" class="external-link">grapghite</a></em>,
which have been nested into one simple function in this package called
<code><a href="../reference/retrieve_topology.html">retrieve_topology()</a></code>. The <code>retrieve_topology</code>
function now supports large lists of species and databases. Databases
that are currently supported for human are:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sales-lab/graphite" class="external-link">graphite</a></span><span class="op">)</span></span>
<span><span class="fu">graphite</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graphite/man/pathwayDatabases.html" class="external-link">pathwayDatabases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">species</span> <span class="op">==</span>  <span class="st">"hsapiens"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">pander</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pander/man/pander.html" class="external-link">pander</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table style="width:36%;" class="table">
<colgroup>
<col width="15%">
<col width="20%">
</colgroup>
<thead><tr class="header">
<th align="center">species</th>
<th align="center">database</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">kegg</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">panther</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">pathbank</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">pharmgkb</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">reactome</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">smpdb</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">wikipathways</td>
</tr>
</tbody>
</table>
<p>The retrieved topology information will be a list where each element
corresponds a pathway. It’s recommended to save the list as a file so
this step only needs to be performed once for each database.</p>
<p>This vignette chose <em>KEGG</em> pathways in humanh as an
example.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsTopology</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/retrieve_topology.html">retrieve_topology</a></span><span class="op">(</span>database <span class="op">=</span> <span class="st">"kegg"</span>, species <span class="op">=</span> <span class="st">"hsapiens"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsTopology</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "kegg.Glycolysis / Gluconeogenesis"            </span></span>
<span><span class="co">## [2] "kegg.Citrate cycle (TCA cycle)"               </span></span>
<span><span class="co">## [3] "kegg.Pentose phosphate pathway"               </span></span>
<span><span class="co">## [4] "kegg.Pentose and glucuronate interconversions"</span></span>
<span><span class="co">## [5] "kegg.Fructose and mannose metabolism"         </span></span>
<span><span class="co">## [6] "kegg.Galactose metabolism"</span></span></code></pre>
<p>If only selected biological processes are of interest to your
research, it’s possible to only retrieve the topologies of those
pathways by specifying keywords. For example, to retrieve all
metabolism-related <em>KEGG</em> pathways:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsTopology_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/retrieve_topology.html">retrieve_topology</a></span><span class="op">(</span></span>
<span>  database <span class="op">=</span> <span class="st">"kegg"</span>,</span>
<span>  species <span class="op">=</span> <span class="st">"hsapiens"</span>, </span>
<span>  keyword <span class="op">=</span> <span class="st">"metabolism"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsTopology_sub</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "kegg.Fructose and mannose metabolism"             </span></span>
<span><span class="co">##  [2] "kegg.Galactose metabolism"                        </span></span>
<span><span class="co">##  [3] "kegg.Ascorbate and aldarate metabolism"           </span></span>
<span><span class="co">##  [4] "kegg.Purine metabolism"                           </span></span>
<span><span class="co">##  [5] "kegg.Caffeine metabolism"                         </span></span>
<span><span class="co">##  [6] "kegg.Pyrimidine metabolism"                       </span></span>
<span><span class="co">##  [7] "kegg.Alanine, aspartate and glutamate metabolism" </span></span>
<span><span class="co">##  [8] "kegg.Glycine, serine and threonine metabolism"    </span></span>
<span><span class="co">##  [9] "kegg.Cysteine and methionine metabolism"          </span></span>
<span><span class="co">## [10] "kegg.Arginine and proline metabolism"             </span></span>
<span><span class="co">## [11] "kegg.Histidine metabolism"                        </span></span>
<span><span class="co">## [12] "kegg.Tyrosine metabolism"                         </span></span>
<span><span class="co">## [13] "kegg.Phenylalanine metabolism"                    </span></span>
<span><span class="co">## [14] "kegg.Tryptophan metabolism"                       </span></span>
<span><span class="co">## [15] "kegg.beta-Alanine metabolism"                     </span></span>
<span><span class="co">## [16] "kegg.Taurine and hypotaurine metabolism"          </span></span>
<span><span class="co">## [17] "kegg.Phosphonate and phosphinate metabolism"      </span></span>
<span><span class="co">## [18] "kegg.Selenocompound metabolism"                   </span></span>
<span><span class="co">## [19] "kegg.Glutathione metabolism"                      </span></span>
<span><span class="co">## [20] "kegg.Starch and sucrose metabolism"               </span></span>
<span><span class="co">## [21] "kegg.Amino sugar and nucleotide sugar metabolism" </span></span>
<span><span class="co">## [22] "kegg.Glycerolipid metabolism"                     </span></span>
<span><span class="co">## [23] "kegg.Inositol phosphate metabolism"               </span></span>
<span><span class="co">## [24] "kegg.Glycerophospholipid metabolism"              </span></span>
<span><span class="co">## [25] "kegg.Ether lipid metabolism"                      </span></span>
<span><span class="co">## [26] "kegg.Arachidonic acid metabolism"                 </span></span>
<span><span class="co">## [27] "kegg.Linoleic acid metabolism"                    </span></span>
<span><span class="co">## [28] "kegg.alpha-Linolenic acid metabolism"             </span></span>
<span><span class="co">## [29] "kegg.Sphingolipid metabolism"                     </span></span>
<span><span class="co">## [30] "kegg.Pyruvate metabolism"                         </span></span>
<span><span class="co">## [31] "kegg.Glyoxylate and dicarboxylate metabolism"     </span></span>
<span><span class="co">## [32] "kegg.Propanoate metabolism"                       </span></span>
<span><span class="co">## [33] "kegg.Butanoate metabolism"                        </span></span>
<span><span class="co">## [34] "kegg.Thiamine metabolism"                         </span></span>
<span><span class="co">## [35] "kegg.Riboflavin metabolism"                       </span></span>
<span><span class="co">## [36] "kegg.Vitamin B6 metabolism"                       </span></span>
<span><span class="co">## [37] "kegg.Nicotinate and nicotinamide metabolism"      </span></span>
<span><span class="co">## [38] "kegg.Lipoic acid metabolism"                      </span></span>
<span><span class="co">## [39] "kegg.Retinol metabolism"                          </span></span>
<span><span class="co">## [40] "kegg.Porphyrin metabolism"                        </span></span>
<span><span class="co">## [41] "kegg.Nitrogen metabolism"                         </span></span>
<span><span class="co">## [42] "kegg.Sulfur metabolism"                           </span></span>
<span><span class="co">## [43] "kegg.Metabolism of xenobiotics by cytochrome P450"</span></span>
<span><span class="co">## [44] "kegg.Drug metabolism - cytochrome P450"           </span></span>
<span><span class="co">## [45] "kegg.Drug metabolism - other enzymes"             </span></span>
<span><span class="co">## [46] "kegg.Carbon metabolism"                           </span></span>
<span><span class="co">## [47] "kegg.2-Oxocarboxylic acid metabolism"             </span></span>
<span><span class="co">## [48] "kegg.Fatty acid metabolism"                       </span></span>
<span><span class="co">## [49] "kegg.Nucleotide metabolism"                       </span></span>
<span><span class="co">## [50] "kegg.Cholesterol metabolism"                      </span></span>
<span><span class="co">## [51] "kegg.Central carbon metabolism in cancer"         </span></span>
<span><span class="co">## [52] "kegg.Choline metabolism in cancer"</span></span></code></pre>
<p>It is also possible to provide multiple databases’ names and/or
multiple keywords.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsTopology_mult</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/retrieve_topology.html">retrieve_topology</a></span><span class="op">(</span></span>
<span>  database <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"kegg"</span>, <span class="st">"reactome"</span><span class="op">)</span>,</span>
<span>  species <span class="op">=</span> <span class="st">"hsapiens"</span>, </span>
<span>  keyword <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"metabolism"</span>, <span class="st">"estrogen"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsTopology_mult</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">##   [1] "kegg.Fructose and mannose metabolism"                                           </span></span>
<span><span class="co">##   [2] "kegg.Galactose metabolism"                                                      </span></span>
<span><span class="co">##   [3] "kegg.Ascorbate and aldarate metabolism"                                         </span></span>
<span><span class="co">##   [4] "kegg.Purine metabolism"                                                         </span></span>
<span><span class="co">##   [5] "kegg.Caffeine metabolism"                                                       </span></span>
<span><span class="co">##   [6] "kegg.Pyrimidine metabolism"                                                     </span></span>
<span><span class="co">##   [7] "kegg.Alanine, aspartate and glutamate metabolism"                               </span></span>
<span><span class="co">##   [8] "kegg.Glycine, serine and threonine metabolism"                                  </span></span>
<span><span class="co">##   [9] "kegg.Cysteine and methionine metabolism"                                        </span></span>
<span><span class="co">##  [10] "kegg.Arginine and proline metabolism"                                           </span></span>
<span><span class="co">##  [11] "kegg.Histidine metabolism"                                                      </span></span>
<span><span class="co">##  [12] "kegg.Tyrosine metabolism"                                                       </span></span>
<span><span class="co">##  [13] "kegg.Phenylalanine metabolism"                                                  </span></span>
<span><span class="co">##  [14] "kegg.Tryptophan metabolism"                                                     </span></span>
<span><span class="co">##  [15] "kegg.beta-Alanine metabolism"                                                   </span></span>
<span><span class="co">##  [16] "kegg.Taurine and hypotaurine metabolism"                                        </span></span>
<span><span class="co">##  [17] "kegg.Phosphonate and phosphinate metabolism"                                    </span></span>
<span><span class="co">##  [18] "kegg.Selenocompound metabolism"                                                 </span></span>
<span><span class="co">##  [19] "kegg.Glutathione metabolism"                                                    </span></span>
<span><span class="co">##  [20] "kegg.Starch and sucrose metabolism"                                             </span></span>
<span><span class="co">##  [21] "kegg.Amino sugar and nucleotide sugar metabolism"                               </span></span>
<span><span class="co">##  [22] "kegg.Glycerolipid metabolism"                                                   </span></span>
<span><span class="co">##  [23] "kegg.Inositol phosphate metabolism"                                             </span></span>
<span><span class="co">##  [24] "kegg.Glycerophospholipid metabolism"                                            </span></span>
<span><span class="co">##  [25] "kegg.Ether lipid metabolism"                                                    </span></span>
<span><span class="co">##  [26] "kegg.Arachidonic acid metabolism"                                               </span></span>
<span><span class="co">##  [27] "kegg.Linoleic acid metabolism"                                                  </span></span>
<span><span class="co">##  [28] "kegg.alpha-Linolenic acid metabolism"                                           </span></span>
<span><span class="co">##  [29] "kegg.Sphingolipid metabolism"                                                   </span></span>
<span><span class="co">##  [30] "kegg.Pyruvate metabolism"                                                       </span></span>
<span><span class="co">##  [31] "kegg.Glyoxylate and dicarboxylate metabolism"                                   </span></span>
<span><span class="co">##  [32] "kegg.Propanoate metabolism"                                                     </span></span>
<span><span class="co">##  [33] "kegg.Butanoate metabolism"                                                      </span></span>
<span><span class="co">##  [34] "kegg.Thiamine metabolism"                                                       </span></span>
<span><span class="co">##  [35] "kegg.Riboflavin metabolism"                                                     </span></span>
<span><span class="co">##  [36] "kegg.Vitamin B6 metabolism"                                                     </span></span>
<span><span class="co">##  [37] "kegg.Nicotinate and nicotinamide metabolism"                                    </span></span>
<span><span class="co">##  [38] "kegg.Lipoic acid metabolism"                                                    </span></span>
<span><span class="co">##  [39] "kegg.Retinol metabolism"                                                        </span></span>
<span><span class="co">##  [40] "kegg.Porphyrin metabolism"                                                      </span></span>
<span><span class="co">##  [41] "kegg.Nitrogen metabolism"                                                       </span></span>
<span><span class="co">##  [42] "kegg.Sulfur metabolism"                                                         </span></span>
<span><span class="co">##  [43] "kegg.Metabolism of xenobiotics by cytochrome P450"                              </span></span>
<span><span class="co">##  [44] "kegg.Drug metabolism - cytochrome P450"                                         </span></span>
<span><span class="co">##  [45] "kegg.Drug metabolism - other enzymes"                                           </span></span>
<span><span class="co">##  [46] "kegg.Carbon metabolism"                                                         </span></span>
<span><span class="co">##  [47] "kegg.2-Oxocarboxylic acid metabolism"                                           </span></span>
<span><span class="co">##  [48] "kegg.Fatty acid metabolism"                                                     </span></span>
<span><span class="co">##  [49] "kegg.Nucleotide metabolism"                                                     </span></span>
<span><span class="co">##  [50] "kegg.Estrogen signaling pathway"                                                </span></span>
<span><span class="co">##  [51] "kegg.Cholesterol metabolism"                                                    </span></span>
<span><span class="co">##  [52] "kegg.Central carbon metabolism in cancer"                                       </span></span>
<span><span class="co">##  [53] "kegg.Choline metabolism in cancer"                                              </span></span>
<span><span class="co">##  [54] "reactome.Metabolism"                                                            </span></span>
<span><span class="co">##  [55] "reactome.Inositol phosphate metabolism"                                         </span></span>
<span><span class="co">##  [56] "reactome.PI Metabolism"                                                         </span></span>
<span><span class="co">##  [57] "reactome.Phospholipid metabolism"                                               </span></span>
<span><span class="co">##  [58] "reactome.Nucleotide metabolism"                                                 </span></span>
<span><span class="co">##  [59] "reactome.Sulfur amino acid metabolism"                                          </span></span>
<span><span class="co">##  [60] "reactome.Glycosaminoglycan metabolism"                                          </span></span>
<span><span class="co">##  [61] "reactome.Integration of energy metabolism"                                      </span></span>
<span><span class="co">##  [62] "reactome.Keratan sulfate/keratin metabolism"                                    </span></span>
<span><span class="co">##  [63] "reactome.Heparan sulfate/heparin (HS-GAG) metabolism"                           </span></span>
<span><span class="co">##  [64] "reactome.Glycosphingolipid metabolism"                                          </span></span>
<span><span class="co">##  [65] "reactome.Lipoprotein metabolism"                                                </span></span>
<span><span class="co">##  [66] "reactome.Chondroitin sulfate/dermatan sulfate metabolism"                       </span></span>
<span><span class="co">##  [67] "reactome.Porphyrin metabolism"                                                  </span></span>
<span><span class="co">##  [68] "reactome.Estrogen biosynthesis"                                                 </span></span>
<span><span class="co">##  [69] "reactome.Bile acid and bile salt metabolism"                                    </span></span>
<span><span class="co">##  [70] "reactome.Non-coding RNA Metabolism"                                             </span></span>
<span><span class="co">##  [71] "reactome.Metabolism of steroid hormones"                                        </span></span>
<span><span class="co">##  [72] "reactome.Cobalamin (Cbl, vitamin B12) transport and metabolism"                 </span></span>
<span><span class="co">##  [73] "reactome.Metabolism of folate and pterines"                                     </span></span>
<span><span class="co">##  [74] "reactome.Biotin transport and metabolism"                                       </span></span>
<span><span class="co">##  [75] "reactome.Vitamin D (calciferol) metabolism"                                     </span></span>
<span><span class="co">##  [76] "reactome.Nicotinate metabolism"                                                 </span></span>
<span><span class="co">##  [77] "reactome.Vitamin C (ascorbate) metabolism"                                      </span></span>
<span><span class="co">##  [78] "reactome.Vitamin B2 (riboflavin) metabolism"                                    </span></span>
<span><span class="co">##  [79] "reactome.Metabolism of water-soluble vitamins and cofactors"                    </span></span>
<span><span class="co">##  [80] "reactome.Metabolism of vitamins and cofactors"                                  </span></span>
<span><span class="co">##  [81] "reactome.Vitamin B5 (pantothenate) metabolism"                                  </span></span>
<span><span class="co">##  [82] "reactome.Carnitine metabolism"                                                  </span></span>
<span><span class="co">##  [83] "reactome.Metabolism of nitric oxide: NOS3 activation and regulation"            </span></span>
<span><span class="co">##  [84] "reactome.Metabolism of Angiotensinogen to Angiotensins"                         </span></span>
<span><span class="co">##  [85] "reactome.alpha-linolenic (omega3) and linoleic (omega6) acid metabolism"        </span></span>
<span><span class="co">##  [86] "reactome.Linoleic acid (LA) metabolism"                                         </span></span>
<span><span class="co">##  [87] "reactome.alpha-linolenic acid (ALA) metabolism"                                 </span></span>
<span><span class="co">##  [88] "reactome.Metabolism of amine-derived hormones"                                  </span></span>
<span><span class="co">##  [89] "reactome.Arachidonic acid metabolism"                                           </span></span>
<span><span class="co">##  [90] "reactome.Hyaluronan metabolism"                                                 </span></span>
<span><span class="co">##  [91] "reactome.Metabolism of ingested SeMet, Sec, MeSec into H2Se"                    </span></span>
<span><span class="co">##  [92] "reactome.Selenoamino acid metabolism"                                           </span></span>
<span><span class="co">##  [93] "reactome.Peptide hormone metabolism"                                            </span></span>
<span><span class="co">##  [94] "reactome.Defects in vitamin and cofactor metabolism"                            </span></span>
<span><span class="co">##  [95] "reactome.Defects in biotin (Btn) metabolism"                                    </span></span>
<span><span class="co">##  [96] "reactome.Metabolism of polyamines"                                              </span></span>
<span><span class="co">##  [97] "reactome.Diseases associated with glycosaminoglycan metabolism"                 </span></span>
<span><span class="co">##  [98] "reactome.Glyoxylate metabolism and glycine degradation"                         </span></span>
<span><span class="co">##  [99] "reactome.Peroxisomal lipid metabolism"                                          </span></span>
<span><span class="co">## [100] "reactome.Metabolism of proteins"                                                </span></span>
<span><span class="co">## [101] "reactome.Regulation of lipid metabolism by PPARalpha"                           </span></span>
<span><span class="co">## [102] "reactome.Sialic acid metabolism"                                                </span></span>
<span><span class="co">## [103] "reactome.Sphingolipid metabolism"                                               </span></span>
<span><span class="co">## [104] "reactome.Metabolism of lipids"                                                  </span></span>
<span><span class="co">## [105] "reactome.Fructose metabolism"                                                   </span></span>
<span><span class="co">## [106] "reactome.Diseases of carbohydrate metabolism"                                   </span></span>
<span><span class="co">## [107] "reactome.Diseases of metabolism"                                                </span></span>
<span><span class="co">## [108] "reactome.Surfactant metabolism"                                                 </span></span>
<span><span class="co">## [109] "reactome.Metabolism of fat-soluble vitamins"                                    </span></span>
<span><span class="co">## [110] "reactome.Pyruvate metabolism"                                                   </span></span>
<span><span class="co">## [111] "reactome.Glucose metabolism"                                                    </span></span>
<span><span class="co">## [112] "reactome.Creatine metabolism"                                                   </span></span>
<span><span class="co">## [113] "reactome.Amino acid and derivative metabolism"                                  </span></span>
<span><span class="co">## [114] "reactome.Carbohydrate metabolism"                                               </span></span>
<span><span class="co">## [115] "reactome.Pyruvate metabolism and Citric Acid (TCA) cycle"                       </span></span>
<span><span class="co">## [116] "reactome.Ketone body metabolism"                                                </span></span>
<span><span class="co">## [117] "reactome.RUNX1 regulates estrogen receptor mediated transcription"              </span></span>
<span><span class="co">## [118] "reactome.Metabolism of RNA"                                                     </span></span>
<span><span class="co">## [119] "reactome.Metabolism of steroids"                                                </span></span>
<span><span class="co">## [120] "reactome.Phenylalanine and tyrosine metabolism"                                 </span></span>
<span><span class="co">## [121] "reactome.Aspartate and asparagine metabolism"                                   </span></span>
<span><span class="co">## [122] "reactome.Phenylalanine metabolism"                                              </span></span>
<span><span class="co">## [123] "reactome.Glutamate and glutamine metabolism"                                    </span></span>
<span><span class="co">## [124] "reactome.Fatty acid metabolism"                                                 </span></span>
<span><span class="co">## [125] "reactome.Metabolism of cofactors"                                               </span></span>
<span><span class="co">## [126] "reactome.Triglyceride metabolism"                                               </span></span>
<span><span class="co">## [127] "reactome.Glycogen metabolism"                                                   </span></span>
<span><span class="co">## [128] "reactome.Extra-nuclear estrogen signaling"                                      </span></span>
<span><span class="co">## [129] "reactome.Estrogen-dependent gene expression"                                    </span></span>
<span><span class="co">## [130] "reactome.Regulation of glycolysis by fructose 2,6-bisphosphate metabolism"      </span></span>
<span><span class="co">## [131] "reactome.Estrogen-stimulated signaling through PRKCZ"                           </span></span>
<span><span class="co">## [132] "reactome.Estrogen-dependent nuclear events downstream of ESR-membrane signaling"</span></span>
<span><span class="co">## [133] "reactome.Retinoid metabolism and transport"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="score-single-sample-pathway-perturbation">Score single sample pathway perturbation<a class="anchor" aria-label="anchor" href="#score-single-sample-pathway-perturbation"></a>
</h3>
<p>Once the expression matrix, sample metadata and pathway topologies
are all ready, gene-wise single-sample perturbation scores can be
computed:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genePertScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/raw_gene_pert.html">raw_gene_pert</a></span><span class="op">(</span><span class="va">weightedFC</span><span class="op">$</span><span class="va">logFC</span>, <span class="va">gsTopology</span><span class="op">)</span></span></code></pre></div>
<p>and summed to derive pathway perturbation scores for each pathway in
each treated samples.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathwayPertScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pathway_pert.html">pathway_pert</a></span><span class="op">(</span><span class="va">genePertScore</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pathwayPertScore</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##           sample        score                                        gs_name</span></span>
<span><span class="co">## 1 E2+R5020_N2_48  0.006717358 kegg.EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 2    R5020_N2_48  0.006718738 kegg.EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 3 E2+R5020_N3_48  0.002785146 kegg.EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 4    R5020_N3_48 -0.004030015 kegg.EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 5 E2+R5020_P4_48  0.004087122 kegg.EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 6    R5020_P4_48  0.015609886 kegg.EGFR tyrosine kinase inhibitor resistance</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="generate-null-distributions-of-perturbation-scores">Generate null distributions of perturbation scores<a class="anchor" aria-label="anchor" href="#generate-null-distributions-of-perturbation-scores"></a>
</h3>
<p>To derive the empirical p-values for each single-sample perturbation
scores and normalize the raw scores for comparing overall treatment
effects, null distributions of scores for each pathway are generated
through a sample-label permutation approach.</p>
<p>For each round of permutation, sample labels are randomly shuffled to
derive the permuted ssLogFCs, which are then used to score pathway
perturbation. We recommend performing a minimum of 1000 rounds of
permutation, which means at least 8 samples are required.</p>
<p>The <code><a href="../reference/generate_permuted_scores.html">generate_permuted_scores()</a></code> function does not require
sample metadata but the number of treatments in the study design,
including the control treatment, need to be specified. In this example
data, the number of treatmentS was 3 (Vehicle, E2, and E2+R5020).</p>
<p>The output of the <code><a href="../reference/generate_permuted_scores.html">generate_permuted_scores()</a></code> function is
a list where each element is a vector of permuted perturbation scores
for a pathway.</p>
<p>The permutation step relies on the parallel computing feature
provided by <em><a href="https://bioconductor.org/packages/3.18/BiocParallel" class="external-link">BiocParallel</a></em>.
You can choose to customize the parallel back-end or stick with the
default one returned by <code><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">BiocParallel::bpparam()</a></code>. Depending
on the size of the data, this step can take some time to complete. If
the sample size is large, we recommend users consider performing this
step on an HPC.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">permutedScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_permuted_scores.html">generate_permuted_scores</a></span><span class="op">(</span></span>
<span>  expreMatrix  <span class="op">=</span> <span class="va">logCPM_example</span>, </span>
<span>  numOfTreat <span class="op">=</span> <span class="fl">3</span>, NB <span class="op">=</span> <span class="fl">1000</span>, </span>
<span>  gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>  weight <span class="op">=</span> <span class="va">weightedFC</span><span class="op">$</span><span class="va">weight</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="test-significant-perturbation-on">Test significant perturbation on<a class="anchor" aria-label="anchor" href="#test-significant-perturbation-on"></a>
</h3>
<div class="section level4">
<h4 id="single-sample-level">single-sample level<a class="anchor" aria-label="anchor" href="#single-sample-level"></a>
</h4>
<p>After the empirical null distributions are generated, the median and
mad of each distribution will be calculated for each pathway to convert
the test single-sample perturbation scores derived from the
<code>compute_perturbation_score()</code> function to robust z-scores:
<span class="math display">\[(Score - Median)/MAD\]</span></p>
<p>Two-sided p-values associated with each robust z-scores are also
computed and will be corrected for multiple-testing using a user-defined
approach. The default is <code>fdr</code>.</p>
<p>The <code><a href="../reference/normalise_by_permu.html">normalise_by_permu()</a></code> function requires the test
perturbation scores and permuted perturbation scores as input. Users can
choose to sort the output by p-values, gene-set names or sample
names.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normalisedScores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalise_by_permu.html">normalise_by_permu</a></span><span class="op">(</span><span class="va">permutedScore</span>, <span class="va">pathwayPertScore</span>, sortBy <span class="op">=</span> <span class="st">"pvalue"</span><span class="op">)</span></span></code></pre></div>
<p>Since the permutation step takes a long time to run and the output is
too large to be included as part of the package, the results of the
<code>normalise_by_permu</code> step has been pre-computed and can be
loaded with:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"normalisedScores.rda"</span>, package <span class="op">=</span> <span class="st">"sSNAPPY"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Pathways that were significant perturbed within individual samples
are:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">adjPvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_at</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/vars.html" class="external-link">vars</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sample"</span>, <span class="st">"gs_name"</span><span class="op">)</span><span class="op">)</span>, <span class="va">as.factor</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_if</a></span><span class="op">(</span><span class="va">is.numeric</span>, <span class="va">sprintf</span>, fmt <span class="op">=</span> <span class="st">'%#.4f'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Direction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">robustZ</span> <span class="op">&lt;</span> <span class="fl">0</span>, <span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>        <span class="va">sample</span>, <span class="va">patient</span>, Treatment <span class="op">=</span> <span class="va">treatment</span>,</span>
<span>        `Perturbation Score` <span class="op">=</span> <span class="va">robustZ</span>, <span class="va">Direction</span>,</span>
<span>        `Gene-set name` <span class="op">=</span> <span class="va">gs_name</span>, </span>
<span>        `P-value` <span class="op">=</span> <span class="va">pvalue</span>, </span>
<span>        FDR <span class="op">=</span> <span class="va">adjPvalue</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span></span>
<span>        filter <span class="op">=</span> <span class="st">"top"</span>, </span>
<span>        options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>            columnDefs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="st">"Direction"</span>, visible <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span>        <span class="op">)</span>, </span>
<span>        caption <span class="op">=</span> <span class="fu">htmltools</span><span class="fu">::</span><span class="va"><a href="https://rstudio.github.io/htmltools/reference/builder.html" class="external-link">tags</a></span><span class="op">$</span><span class="fu">caption</span><span class="op">(</span></span>
<span>                  <span class="fu">htmltools</span><span class="fu">::</span><span class="fu"><a href="https://rstudio.github.io/htmltools/reference/builder.html" class="external-link">em</a></span><span class="op">(</span></span>
<span>                      <span class="st">"Pathways that were significant perturbed within individual samples."</span><span class="op">)</span></span>
<span>              <span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/formatCurrency.html" class="external-link">formatStyle</a></span><span class="op">(</span></span>
<span>        <span class="st">'Perturbation Score'</span>, <span class="st">'Direction'</span>,</span>
<span>        color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DT/man/styleInterval.html" class="external-link">styleEqual</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-4579ec248708214c07a2" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4579ec248708214c07a2">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020_N2_48&quot;,&quot;E2+R5020_N3_48&quot;,&quot;E2+R5020_P4_48&quot;,&quot;E2+R5020_P5_48&quot;,&quot;E2+R5020_P6_48&quot;,&quot;R5020_N2_48&quot;,&quot;R5020_N3_48&quot;,&quot;R5020_P4_48&quot;,&quot;R5020_P5_48&quot;,&quot;R5020_P6_48&quot;,&quot;Vehicle_N2_48&quot;,&quot;Vehicle_N3_48&quot;,&quot;Vehicle_P4_48&quot;,&quot;Vehicle_P5_48&quot;,&quot;Vehicle_P6_48&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;N2&quot;,&quot;N3&quot;,&quot;P4&quot;,&quot;P5&quot;,&quot;P6&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Vehicle&quot;,&quot;E2&quot;,&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;kegg.Acute myeloid leukemia&quot;,&quot;kegg.Adherens junction&quot;,&quot;kegg.Adipocytokine signaling pathway&quot;,&quot;kegg.Adrenergic signaling in cardiomyocytes&quot;,&quot;kegg.African trypanosomiasis&quot;,&quot;kegg.AGE-RAGE signaling pathway in diabetic complications&quot;,&quot;kegg.Alcoholic liver disease&quot;,&quot;kegg.Alcoholism&quot;,&quot;kegg.Aldosterone synthesis and secretion&quot;,&quot;kegg.Aldosterone-regulated sodium reabsorption&quot;,&quot;kegg.Allograft rejection&quot;,&quot;kegg.Alzheimer disease&quot;,&quot;kegg.Amoebiasis&quot;,&quot;kegg.Amphetamine addiction&quot;,&quot;kegg.AMPK signaling pathway&quot;,&quot;kegg.Amyotrophic lateral sclerosis&quot;,&quot;kegg.Antifolate resistance&quot;,&quot;kegg.Antigen processing and presentation&quot;,&quot;kegg.Apelin signaling pathway&quot;,&quot;kegg.Apoptosis&quot;,&quot;kegg.Apoptosis - multiple species&quot;,&quot;kegg.Arrhythmogenic right ventricular cardiomyopathy&quot;,&quot;kegg.Autoimmune thyroid disease&quot;,&quot;kegg.Autophagy - animal&quot;,&quot;kegg.Autophagy - other&quot;,&quot;kegg.Axon guidance&quot;,&quot;kegg.B cell receptor signaling pathway&quot;,&quot;kegg.Bacterial invasion of epithelial cells&quot;,&quot;kegg.Basal cell carcinoma&quot;,&quot;kegg.Bile secretion&quot;,&quot;kegg.Bladder cancer&quot;,&quot;kegg.Breast cancer&quot;,&quot;kegg.C-type lectin receptor signaling pathway&quot;,&quot;kegg.Calcium signaling pathway&quot;,&quot;kegg.cAMP signaling pathway&quot;,&quot;kegg.Carbohydrate digestion and absorption&quot;,&quot;kegg.Cell adhesion molecules&quot;,&quot;kegg.Cell cycle&quot;,&quot;kegg.Cellular senescence&quot;,&quot;kegg.Central carbon metabolism in cancer&quot;,&quot;kegg.cGMP-PKG signaling pathway&quot;,&quot;kegg.Chagas disease&quot;,&quot;kegg.Chemical carcinogenesis - reactive oxygen species&quot;,&quot;kegg.Chemical carcinogenesis - receptor activation&quot;,&quot;kegg.Chemokine signaling pathway&quot;,&quot;kegg.Cholesterol metabolism&quot;,&quot;kegg.Choline metabolism in cancer&quot;,&quot;kegg.Cholinergic synapse&quot;,&quot;kegg.Chronic myeloid leukemia&quot;,&quot;kegg.Circadian entrainment&quot;,&quot;kegg.Cocaine addiction&quot;,&quot;kegg.Colorectal cancer&quot;,&quot;kegg.Complement and coagulation cascades&quot;,&quot;kegg.Coronavirus disease - COVID-19&quot;,&quot;kegg.Cortisol synthesis and secretion&quot;,&quot;kegg.Cushing syndrome&quot;,&quot;kegg.Cytokine-cytokine receptor interaction&quot;,&quot;kegg.Cytosolic DNA-sensing pathway&quot;,&quot;kegg.Diabetic cardiomyopathy&quot;,&quot;kegg.Dilated cardiomyopathy&quot;,&quot;kegg.Dopaminergic synapse&quot;,&quot;kegg.ECM-receptor interaction&quot;,&quot;kegg.EGFR tyrosine kinase inhibitor resistance&quot;,&quot;kegg.Endocrine and other factor-regulated calcium reabsorption&quot;,&quot;kegg.Endocrine resistance&quot;,&quot;kegg.Endometrial cancer&quot;,&quot;kegg.Epithelial cell signaling in Helicobacter pylori infection&quot;,&quot;kegg.Epstein-Barr virus infection&quot;,&quot;kegg.ErbB signaling pathway&quot;,&quot;kegg.Estrogen signaling pathway&quot;,&quot;kegg.Fanconi anemia pathway&quot;,&quot;kegg.Fc epsilon RI signaling pathway&quot;,&quot;kegg.Fc gamma R-mediated phagocytosis&quot;,&quot;kegg.Fluid shear stress and atherosclerosis&quot;,&quot;kegg.Focal adhesion&quot;,&quot;kegg.FoxO signaling pathway&quot;,&quot;kegg.GABAergic synapse&quot;,&quot;kegg.Gap junction&quot;,&quot;kegg.Gastric acid secretion&quot;,&quot;kegg.Gastric cancer&quot;,&quot;kegg.Glioma&quot;,&quot;kegg.Glucagon signaling pathway&quot;,&quot;kegg.Glutamatergic synapse&quot;,&quot;kegg.GnRH signaling pathway&quot;,&quot;kegg.Graft-versus-host disease&quot;,&quot;kegg.Growth hormone synthesis, secretion and action&quot;,&quot;kegg.Hedgehog signaling pathway&quot;,&quot;kegg.Hepatitis B&quot;,&quot;kegg.Hepatitis C&quot;,&quot;kegg.Hepatocellular carcinoma&quot;,&quot;kegg.Herpes simplex virus 1 infection&quot;,&quot;kegg.HIF-1 signaling pathway&quot;,&quot;kegg.Hippo signaling pathway&quot;,&quot;kegg.Hippo signaling pathway - multiple species&quot;,&quot;kegg.Human cytomegalovirus infection&quot;,&quot;kegg.Human immunodeficiency virus 1 infection&quot;,&quot;kegg.Human papillomavirus infection&quot;,&quot;kegg.Human T-cell leukemia virus 1 infection&quot;,&quot;kegg.Huntington disease&quot;,&quot;kegg.IL-17 signaling pathway&quot;,&quot;kegg.Inflammatory bowel disease&quot;,&quot;kegg.Inflammatory mediator regulation of TRP channels&quot;,&quot;kegg.Influenza A&quot;,&quot;kegg.Insulin resistance&quot;,&quot;kegg.Insulin secretion&quot;,&quot;kegg.Insulin signaling pathway&quot;,&quot;kegg.Intestinal immune network for IgA production&quot;,&quot;kegg.JAK-STAT signaling pathway&quot;,&quot;kegg.Kaposi sarcoma-associated herpesvirus infection&quot;,&quot;kegg.Legionellosis&quot;,&quot;kegg.Leishmaniasis&quot;,&quot;kegg.Leukocyte transendothelial migration&quot;,&quot;kegg.Lipid and atherosclerosis&quot;,&quot;kegg.Long-term depression&quot;,&quot;kegg.Long-term potentiation&quot;,&quot;kegg.Longevity regulating pathway&quot;,&quot;kegg.Longevity regulating pathway - multiple species&quot;,&quot;kegg.Malaria&quot;,&quot;kegg.MAPK signaling pathway&quot;,&quot;kegg.Maturity onset diabetes of the young&quot;,&quot;kegg.Measles&quot;,&quot;kegg.Melanogenesis&quot;,&quot;kegg.Melanoma&quot;,&quot;kegg.MicroRNAs in cancer&quot;,&quot;kegg.Mineral absorption&quot;,&quot;kegg.Mitophagy - animal&quot;,&quot;kegg.Morphine addiction&quot;,&quot;kegg.mRNA surveillance pathway&quot;,&quot;kegg.mTOR signaling pathway&quot;,&quot;kegg.Natural killer cell mediated cytotoxicity&quot;,&quot;kegg.Necroptosis&quot;,&quot;kegg.Neuroactive ligand-receptor interaction&quot;,&quot;kegg.Neurotrophin signaling pathway&quot;,&quot;kegg.Neutrophil extracellular trap formation&quot;,&quot;kegg.NF-kappa B signaling pathway&quot;,&quot;kegg.NOD-like receptor signaling pathway&quot;,&quot;kegg.Non-alcoholic fatty liver disease&quot;,&quot;kegg.Non-small cell lung cancer&quot;,&quot;kegg.Notch signaling pathway&quot;,&quot;kegg.Olfactory transduction&quot;,&quot;kegg.Oocyte meiosis&quot;,&quot;kegg.Osteoclast differentiation&quot;,&quot;kegg.Ovarian steroidogenesis&quot;,&quot;kegg.Oxytocin signaling pathway&quot;,&quot;kegg.p53 signaling pathway&quot;,&quot;kegg.Pancreatic cancer&quot;,&quot;kegg.Pancreatic secretion&quot;,&quot;kegg.Parathyroid hormone synthesis, secretion and action&quot;,&quot;kegg.Parkinson disease&quot;,&quot;kegg.Pathogenic Escherichia coli infection&quot;,&quot;kegg.Pathways in cancer&quot;,&quot;kegg.Pathways of neurodegeneration - multiple diseases&quot;,&quot;kegg.PD-L1 expression and PD-1 checkpoint pathway in cancer&quot;,&quot;kegg.Pertussis&quot;,&quot;kegg.Phospholipase D signaling pathway&quot;,&quot;kegg.Phototransduction&quot;,&quot;kegg.PI3K-Akt signaling pathway&quot;,&quot;kegg.Platelet activation&quot;,&quot;kegg.Platinum drug resistance&quot;,&quot;kegg.PPAR signaling pathway&quot;,&quot;kegg.Prion disease&quot;,&quot;kegg.Progesterone-mediated oocyte maturation&quot;,&quot;kegg.Prolactin signaling pathway&quot;,&quot;kegg.Prostate cancer&quot;,&quot;kegg.Protein processing in endoplasmic reticulum&quot;,&quot;kegg.Proteoglycans in cancer&quot;,&quot;kegg.Rap1 signaling pathway&quot;,&quot;kegg.Ras signaling pathway&quot;,&quot;kegg.Regulation of actin cytoskeleton&quot;,&quot;kegg.Regulation of lipolysis in adipocytes&quot;,&quot;kegg.Relaxin signaling pathway&quot;,&quot;kegg.Renal cell carcinoma&quot;,&quot;kegg.Renin secretion&quot;,&quot;kegg.Renin-angiotensin system&quot;,&quot;kegg.Retrograde endocannabinoid signaling&quot;,&quot;kegg.Rheumatoid arthritis&quot;,&quot;kegg.RIG-I-like receptor signaling pathway&quot;,&quot;kegg.RNA degradation&quot;,&quot;kegg.Salivary secretion&quot;,&quot;kegg.Salmonella infection&quot;,&quot;kegg.Serotonergic synapse&quot;,&quot;kegg.Shigellosis&quot;,&quot;kegg.Signaling pathways regulating pluripotency of stem cells&quot;,&quot;kegg.Small cell lung cancer&quot;,&quot;kegg.SNARE interactions in vesicular transport&quot;,&quot;kegg.Sphingolipid signaling pathway&quot;,&quot;kegg.Spinocerebellar ataxia&quot;,&quot;kegg.Staphylococcus aureus infection&quot;,&quot;kegg.Sulfur relay system&quot;,&quot;kegg.Systemic lupus erythematosus&quot;,&quot;kegg.T cell receptor signaling pathway&quot;,&quot;kegg.Taste transduction&quot;,&quot;kegg.TGF-beta signaling pathway&quot;,&quot;kegg.Th1 and Th2 cell differentiation&quot;,&quot;kegg.Th17 cell differentiation&quot;,&quot;kegg.Thermogenesis&quot;,&quot;kegg.Thyroid cancer&quot;,&quot;kegg.Thyroid hormone signaling pathway&quot;,&quot;kegg.Thyroid hormone synthesis&quot;,&quot;kegg.Tight junction&quot;,&quot;kegg.TNF signaling pathway&quot;,&quot;kegg.Toll-like receptor signaling pathway&quot;,&quot;kegg.Toxoplasmosis&quot;,&quot;kegg.Transcriptional misregulation in cancer&quot;,&quot;kegg.Tuberculosis&quot;,&quot;kegg.Type II diabetes mellitus&quot;,&quot;kegg.Vascular smooth muscle contraction&quot;,&quot;kegg.Vasopressin-regulated water reabsorption&quot;,&quot;kegg.VEGF signaling pathway&quot;,&quot;kegg.Vibrio cholerae infection&quot;,&quot;kegg.Viral myocarditis&quot;,&quot;kegg.Viral protein interaction with cytokine and cytokine receptor&quot;,&quot;kegg.Wnt signaling pathway&quot;,&quot;kegg.Yersinia infection&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","caption":"<caption>\n  <em>Pathways that were significant perturbed within individual samples.<\/em>\n<\/caption>","data":[["1","2","3","4","5","6","7","8"],["R5020_N2_48","E2+R5020_N2_48","R5020_N2_48","R5020_N2_48","E2+R5020_N2_48","E2+R5020_N2_48","R5020_N2_48","E2+R5020_N2_48"],["N2","N2","N2","N2","N2","N2","N2","N2"],["R5020","E2+R5020","R5020","R5020","E2+R5020","E2+R5020","R5020","E2+R5020"],["-10.8154","-9.7261","4.0107","3.8374","3.6771","3.6317","3.4288","3.3398"],["Inhibited","Inhibited","Activation","Activation","Activation","Activation","Activation","Activation"],["kegg.TGF-beta signaling pathway","kegg.TGF-beta signaling pathway","kegg.Growth hormone synthesis, secretion and action","kegg.PI3K-Akt signaling pathway","kegg.Growth hormone synthesis, secretion and action","kegg.PI3K-Akt signaling pathway","kegg.Amoebiasis","kegg.Amoebiasis"],["0.0000","0.0000","0.0001","0.0001","0.0002","0.0003","0.0006","0.0008"],["0.0000","0.0000","0.0065","0.0089","0.0201","0.0201","0.0324","0.0449"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>patient<\/th>\n      <th>Treatment<\/th>\n      <th>Perturbation Score<\/th>\n      <th>Direction<\/th>\n      <th>Gene-set name<\/th>\n      <th>P-value<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":5,"visible":false},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[5]; $(this.api().cell(row, 4).node()).css({'color':value == \"Inhibited\" ? \"blue\" : value == \"Activation\" ? \"red\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script><div class="section level5">
<h5 id="visualise-overlap-between-gene-sets-as-networks">Visualise overlap between gene-sets as networks<a class="anchor" aria-label="anchor" href="#visualise-overlap-between-gene-sets-as-networks"></a>
</h5>
<p>To visualise significantly perturbed biological pathways as a
network, where edges between gene-sets reflect how much overlap those
two gene-sets share, we can use the <code>plot_gs_network</code>
function. The function can take <code><a href="../reference/normalise_by_permu.html">normalise_by_permu()</a></code>’s
output, or a subset of it as its direct input.</p>
<p>Nodes in the network plots could be colored by the predicted
direction of perturbation (i.e.. sign of robust z-score):</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pl</span> <span class="op">&lt;-</span> <span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">adjPvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/droplevels.html" class="external-link">droplevels</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span><span class="va">x</span>, </span>
<span>               status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span></span>
<span>                   <span class="va">robustZ</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">"Activated"</span>, <span class="st">"Inhibited"</span><span class="op">)</span><span class="op">)</span> </span>
<span>    <span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>        <span class="va">plot_gs_network</span>, </span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorBy <span class="op">=</span> <span class="st">"status"</span>,</span>
<span>        gsLegTitle <span class="op">=</span> <span class="st">"Direction of pathway Perturbation"</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="va">x</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>            panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>            panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>        <span class="op">)</span> <span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span></span>
<span>    plotlist <span class="op">=</span> <span class="va">pl</span>, </span>
<span>    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pl</span><span class="op">)</span>,</span>
<span>    label_size <span class="op">=</span> <span class="fl">8</span>,</span>
<span>    nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/sigGS_nt_zscore-1.png" alt="*Pathways significantly perturbed in individual samples, where gene-sets were colored by pathways' directions of changes.*" width="1344"><p class="caption">
<em>Pathways significantly perturbed in individual samples, where
gene-sets were colored by pathways’ directions of changes.</em>
</p>
</div>
<p>Or p-values:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pl</span> <span class="op">&lt;-</span> <span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">adjPvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/droplevels.html" class="external-link">droplevels</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>        <span class="va">plot_gs_network</span>, </span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorBy <span class="op">=</span> <span class="st">"pvalue"</span>, </span>
<span>        color_lg_title <span class="op">=</span> <span class="st">"P-value"</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>        <span class="va">x</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>        <span class="op">)</span> <span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html" class="external-link">plot_grid</a></span><span class="op">(</span></span>
<span>    plotlist <span class="op">=</span> <span class="va">pl</span>, </span>
<span>    labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pl</span><span class="op">)</span>,</span>
<span>    label_size <span class="op">=</span> <span class="fl">8</span>,</span>
<span>    nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/sigGS_nt_pvalue-1.png" alt="*Pathways significantly perturbed in individual samples, where gene-sets were colored by pathways' p-values*" width="960"><p class="caption">
<em>Pathways significantly perturbed in individual samples, where
gene-sets were colored by pathways’ p-values</em>
</p>
</div>
<p>The function allows you to customize the layout, colour, edge
transparency and other aesthetics of the graph. More information can be
found on the help page (<code><a href="../reference/plot_gs_network.html">?plot_gs_network</a></code>). The output of
the graph is a <code>ggplot</code> object and the theme of it can be
changed just as any other <code>ggplot</code> figures.</p>
</div>
</div>
<div class="section level4">
<h4 id="treatment-level">treatment-level<a class="anchor" aria-label="anchor" href="#treatment-level"></a>
</h4>
<p>In addition to testing pathway perturbations at single-sample level,
normalised perturbation scores can also be used to model mean treatment
effects within a group, such as within each treatment group. An
advantage of this method is that it has a high level of flexibility that
allows us to incorporate confounding factors as co-factors or
co-variates to offset their effects.</p>
<p>For example, in the example data-set, samples were collected from
patients with different progesterone receptor (PR) statuses. Knowing
that PR status would affect how tumour tissues responded to estrogen
and/or progesterone treatments, we can include it as a cofactor to
offset its confounding effect.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co">#.["Estrogen signaling pathway"] %&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">robustZ</span> <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">treatment</span> <span class="op">+</span> <span class="va">PR</span>, data <span class="op">=</span> <span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">summary</span><span class="op">)</span></span>
<span><span class="va">treat_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>, </span>
<span>  <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">fit</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">coefficients</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="va">.</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>,<span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Estimate</span>, pvalue <span class="op">=</span> <span class="va">`Pr(&gt;|t|)`</span> <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"Treatment"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>        gs_name <span class="op">=</span> <span class="va">x</span>, </span>
<span>        FDR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">pvalue</span>, <span class="st">"fdr"</span><span class="op">)</span>, </span>
<span>        Treatment <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html" class="external-link">str_remove_all</a></span><span class="op">(</span><span class="va">Treatment</span>, <span class="st">"treatment"</span><span class="op">)</span></span>
<span>      <span class="op">)</span> </span>
<span>  <span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<p>Results from the linear modelling revealed pathways that were on
average perturbed due to each treatment:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_at</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/vars.html" class="external-link">vars</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Treatment"</span>, <span class="st">"gs_name"</span><span class="op">)</span><span class="op">)</span>, <span class="va">as.factor</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_if</a></span><span class="op">(</span><span class="va">is.numeric</span>, <span class="va">sprintf</span>, fmt <span class="op">=</span> <span class="st">'%#.4f'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Direction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">Estimate</span> <span class="op">&lt;</span> <span class="fl">0</span>, <span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>        <span class="va">Treatment</span>, `Perturbation Score` <span class="op">=</span> <span class="va">Estimate</span>, <span class="va">Direction</span>,</span>
<span>        `Gene-set name` <span class="op">=</span> <span class="va">gs_name</span>, </span>
<span>        `P-value` <span class="op">=</span> <span class="va">pvalue</span>, </span>
<span>        <span class="va">FDR</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span></span>
<span>        filter <span class="op">=</span> <span class="st">"top"</span>, </span>
<span>        options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>            columnDefs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="st">"Direction"</span>, visible <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span>        <span class="op">)</span>, </span>
<span>        caption <span class="op">=</span> <span class="fu">htmltools</span><span class="fu">::</span><span class="va"><a href="https://rstudio.github.io/htmltools/reference/builder.html" class="external-link">tags</a></span><span class="op">$</span><span class="fu">caption</span><span class="op">(</span></span>
<span>                  <span class="fu">htmltools</span><span class="fu">::</span><span class="fu"><a href="https://rstudio.github.io/htmltools/reference/builder.html" class="external-link">em</a></span><span class="op">(</span></span>
<span>                      <span class="st">"Pathways that were significant perturbed within each treatment group."</span><span class="op">)</span></span>
<span>              <span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/formatCurrency.html" class="external-link">formatStyle</a></span><span class="op">(</span></span>
<span>        <span class="st">'Perturbation Score'</span>, <span class="st">'Direction'</span>,</span>
<span>        color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DT/man/styleInterval.html" class="external-link">styleEqual</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-5238a454670b7bac6f9d" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-5238a454670b7bac6f9d">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;kegg.Aldosterone synthesis and secretion&quot;,&quot;kegg.Antigen processing and presentation&quot;,&quot;kegg.Autophagy - other&quot;,&quot;kegg.Cellular senescence&quot;,&quot;kegg.Cholinergic synapse&quot;,&quot;kegg.ECM-receptor interaction&quot;,&quot;kegg.Estrogen signaling pathway&quot;,&quot;kegg.Focal adhesion&quot;,&quot;kegg.Human papillomavirus infection&quot;,&quot;kegg.Malaria&quot;,&quot;kegg.Melanoma&quot;,&quot;kegg.Mineral absorption&quot;,&quot;kegg.Neutrophil extracellular trap formation&quot;,&quot;kegg.Ovarian steroidogenesis&quot;,&quot;kegg.PI3K-Akt signaling pathway&quot;,&quot;kegg.Prostate cancer&quot;,&quot;kegg.Ras signaling pathway&quot;,&quot;kegg.Renin secretion&quot;,&quot;kegg.Small cell lung cancer&quot;,&quot;kegg.Staphylococcus aureus infection&quot;,&quot;kegg.Tight junction&quot;,&quot;kegg.Toxoplasmosis&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","caption":"<caption>\n  <em>Pathways that were significant perturbed within each treatment group.<\/em>\n<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39"],["E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","E2+R5020","R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","E2+R5020","R5020"],["0.9733","1.4371","-0.7297","-0.6828","0.7529","0.6684","-0.8443","-1.2284","-0.7919","1.4750","1.4804","1.6235","1.6262","1.9054","1.2115","1.6280","0.9518","1.1671","1.2218","1.5215","2.0280","2.0447","-0.7257","-0.7203","1.2237","1.0859","2.2375","2.2825","1.3148","1.4846","1.7706","1.5966","-1.3629","1.0960","0.5441","0.5132","-1.2165","-1.1793","-1.0709"],["Activation","Activation","Inhibited","Inhibited","Activation","Activation","Inhibited","Inhibited","Inhibited","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Inhibited","Inhibited","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Inhibited","Activation","Activation","Activation","Inhibited","Inhibited","Inhibited"],["kegg.Aldosterone synthesis and secretion","kegg.Aldosterone synthesis and secretion","kegg.Antigen processing and presentation","kegg.Antigen processing and presentation","kegg.Autophagy - other","kegg.Autophagy - other","kegg.Cellular senescence","kegg.Cholinergic synapse","kegg.Cholinergic synapse","kegg.ECM-receptor interaction","kegg.Estrogen signaling pathway","kegg.Estrogen signaling pathway","kegg.Focal adhesion","kegg.Focal adhesion","kegg.Human papillomavirus infection","kegg.Human papillomavirus infection","kegg.Malaria","kegg.Malaria","kegg.Melanoma","kegg.Melanoma","kegg.Mineral absorption","kegg.Mineral absorption","kegg.Neutrophil extracellular trap formation","kegg.Neutrophil extracellular trap formation","kegg.Ovarian steroidogenesis","kegg.Ovarian steroidogenesis","kegg.PI3K-Akt signaling pathway","kegg.PI3K-Akt signaling pathway","kegg.Prostate cancer","kegg.Prostate cancer","kegg.Ras signaling pathway","kegg.Ras signaling pathway","kegg.Renin secretion","kegg.Small cell lung cancer","kegg.Staphylococcus aureus infection","kegg.Staphylococcus aureus infection","kegg.Tight junction","kegg.Toxoplasmosis","kegg.Toxoplasmosis"],["0.0146","0.0021","0.0348","0.0445","0.0258","0.0407","0.0183","0.0009","0.0092","0.0125","0.0014","0.0008","0.0258","0.0131","0.0232","0.0060","0.0258","0.0106","0.0085","0.0027","0.0017","0.0017","0.0273","0.0281","0.0252","0.0400","0.0327","0.0302","0.0292","0.0177","0.0335","0.0490","0.0097","0.0170","0.0298","0.0373","0.0055","0.0132","0.0202"],["0.0146","0.0041","0.0445","0.0445","0.0407","0.0407","0.0366","0.0018","0.0092","0.0249","0.0014","0.0014","0.0258","0.0258","0.0232","0.0120","0.0258","0.0212","0.0085","0.0055","0.0017","0.0017","0.0281","0.0281","0.0400","0.0400","0.0327","0.0327","0.0292","0.0292","0.0490","0.0490","0.0194","0.0340","0.0373","0.0373","0.0111","0.0202","0.0202"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Treatment<\/th>\n      <th>Perturbation Score<\/th>\n      <th>Direction<\/th>\n      <th>Gene-set name<\/th>\n      <th>P-value<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":3,"visible":false},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[3]; $(this.api().cell(row, 2).node()).css({'color':value == \"Inhibited\" ? \"blue\" : value == \"Activation\" ? \"red\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script><p>It is not surprising to see that the estrogen signalling pathway was
significantly activated under both R5020 and E2+R5020 treatments.</p>
<div class="section level5">
<h5 id="visualise-genes-contributions-to-pathway-perturbation">Visualise genes’ contributions to pathway perturbation<a class="anchor" aria-label="anchor" href="#visualise-genes-contributions-to-pathway-perturbation"></a>
</h5>
<p>If there’s a specific pathway that we would like to dig deeper into
and explore which pathway genes played a key role in its perturbation,
for example, activation of the “Ovarian steroidogenesis”, we can plot
the gene-level perturbation scores of genes that are constantly highly
perturbed or highly variable in that pathway as a heatmap using the
<code><a href="../reference/plot_gene_contribution.html">plot_gene_contribution()</a></code> function.</p>
<p>From the heatmap below that we can see that the activation of this
pathway was strongly driven by two genes: ENTREZID:3630 and
ENTREZID:3479.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_gene_contribution.html">plot_gene_contribution</a></span><span class="op">(</span></span>
<span>    genePertMatr  <span class="op">=</span> <span class="va">genePertScore</span><span class="op">$</span><span class="va">`kegg.Ovarian steroidogenesis`</span>,</span>
<span>    <span class="co"># only plot genes with top 15 absolute mean gene-wise scores</span></span>
<span>    filterBy <span class="op">=</span> <span class="st">"mean"</span>, topGene <span class="op">=</span> <span class="fl">15</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-5-1.png" alt="*Gene-level perturbation scores of genes with top 15 absolute mean gene-wise perturbation scores in the Ovarian steroidogenesis gene-set.*" width="768"><p class="caption">
<em>Gene-level perturbation scores of genes with top 15 absolute mean
gene-wise perturbation scores in the Ovarian steroidogenesis
gene-set.</em>
</p>
</div>
<p>By default, genes’ entrez IDs are used and plotted as row names,
which may not be very informative. So the row names could be overwritten
by providing a <code>data.frame</code> mapping entrez IDs to other
identifiers through the <code>mapRownameTo</code> parameter.</p>
<p>Mapping between different gene identifiers could be achieved through
the <code>mapIDs()</code> function from the Bioconductor package <a href="https://bioconductor.org/packages/release/bioc/html/AnnotationDbi.html" class="external-link"><code>AnnotationDbi</code></a>.
But to reduce the compiling time of this vignette, mappings between
entrez IDs and gene names have been provided as a
<code>data.frame</code> called <code>entrez2name</code>.</p>
<p>Note that if the mapping information was provided and the mapping was
successful for some genes but not the others, only genes that have been
mapped successfully will be plotted.</p>
<p>We can also annotate each column (ie. each sample) by the direction
of pathway perturbation in that sample or any other sample metadata,
such as their treaatments. To annotate the columns, we need to create an
annotation <code>data.frame</code> with desired attributes.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotation_df</span> <span class="op">&lt;-</span> <span class="va">pathwayPertScore</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">gs_name</span> <span class="op">==</span> <span class="st">"kegg.Ovarian steroidogenesis"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>        `pathway-level` <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span></span>
<span>            <span class="va">score</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">"Activated"</span>, <span class="st">"Inhibited"</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sample</span>, <span class="va">`pathway-level`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>        <span class="va">.</span>,  <span class="va">metadata_example</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sample</span>, <span class="va">treatment</span><span class="op">)</span>, </span>
<span>        unmatched <span class="op">=</span> <span class="st">"drop"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>Colors of the annotation could be customised through
<code><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap::pheatmap()</a></code>’s <code>annotation_colors</code>
parameter.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"entrez2name.rda"</span>, package <span class="op">=</span> <span class="st">"sSNAPPY"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_gene_contribution.html">plot_gene_contribution</a></span><span class="op">(</span></span>
<span>    genePertMatr  <span class="op">=</span> <span class="va">genePertScore</span><span class="op">$</span><span class="va">`kegg.Ovarian steroidogenesis`</span>,</span>
<span>    annotation_df <span class="op">=</span> <span class="va">annotation_df</span>,</span>
<span>    filterBy <span class="op">=</span> <span class="st">"mean"</span>, topGene <span class="op">=</span> <span class="fl">15</span>,</span>
<span>    mapEntrezID <span class="op">=</span> <span class="va">entrez2name</span>,</span>
<span>    annotation_colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        treatment <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>R5020 <span class="op">=</span> <span class="st">"black"</span>, `E2+R5020` <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>,</span>
<span>        `pathway-level` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Activated <span class="op">=</span> <span class="st">"darkred"</span>, Inhibited <span class="op">=</span> <span class="st">"lightskyblue"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-6-1.png" alt="*Gene-level perturbation scores of genes with top 15 absolute mean gene-wise perturbation scores in the Ovarian steroidogenesis gene-set.Genes playing the most important roles in pathway activation was INS, CGA, GNAS.*" width="768"><p class="caption">
<em>Gene-level perturbation scores of genes with top 15 absolute mean
gene-wise perturbation scores in the Ovarian steroidogenesis
gene-set.Genes playing the most important roles in pathway activation
was INS, CGA, GNAS.</em>
</p>
</div>
</div>
<div class="section level5">
<h5 id="visualise-overlap-between-gene-sets-as-networks-1">Visualise overlap between gene-sets as networks<a class="anchor" aria-label="anchor" href="#visualise-overlap-between-gene-sets-as-networks-1"></a>
</h5>
<p>Results of group-level perturbation can also be visualised using the
<code><a href="../reference/plot_gs_network.html">plot_gs_network()</a></code> function.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">Estimate</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">"Activated"</span>, <span class="st">"Inhibited"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/plot_gs_network.html">plot_gs_network</a></span><span class="op">(</span></span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorBy <span class="op">=</span> <span class="st">"status"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span> </span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-7-1.png" alt="*Pathways significantly perturbed by the E2+R5020 combintation treatment, where colors of nodes reflect pathways' directions of changes.*" width="960"><p class="caption">
<em>Pathways significantly perturbed by the E2+R5020 combintation
treatment, where colors of nodes reflect pathways’ directions of
changes.</em>
</p>
</div>
<p>By default, <code><a href="../reference/plot_gs_network.html">plot_gs_network()</a></code> function does not include
nodes that are not connected to any other nodes, which could be turnt
off by setting the <code>plotIsolated</code> parameter to TURE.</p>
</div>
<div class="section level5">
<h5 id="visualise-community-structure-in-the-gene-set-network">Visualise community structure in the gene-set network<a class="anchor" aria-label="anchor" href="#visualise-community-structure-in-the-gene-set-network"></a>
</h5>
<p>When a large number of pathways were perturbed, it is hard to answer
the question “What key biological processes were perturbed?” solely by
looking at all the pathway names. To solve that, we can use the
<code><a href="../reference/plot_community.html">plot_community()</a></code> function to apply a community detection
algorithm to the network we had above, and annotate each community by
the biological process that most pathways assigned to that community
belong to.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/plot_community.html">plot_community</a></span><span class="op">(</span></span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorBy <span class="op">=</span> <span class="st">"community"</span>, </span>
<span>        color_lg_title <span class="op">=</span> <span class="st">"Community"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-8-1.png" alt="*Pathways significantly perturbed by the E2+R5020 combintation treatment, annotated by the main biological processes each pathways belong to.*" width="1536"><p class="caption">
<em>Pathways significantly perturbed by the E2+R5020 combintation
treatment, annotated by the main biological processes each pathways
belong to.</em>
</p>
</div>
<p><code>sSNAPPY</code> was built in with categorisations of
<em>KEGG</em> pathways. But to annotate pathways retrieved from other
databases, customised annotation <code>data.frame</code> could be
provided through the <code>gsAnnotation</code> parameter.</p>
</div>
<div class="section level5">
<h5 id="visualise-genes-included-in-perturbed-pathways-networks">Visualise genes included in perturbed pathways networks<a class="anchor" aria-label="anchor" href="#visualise-genes-included-in-perturbed-pathways-networks"></a>
</h5>
<p>If we want to not only know if two pathways are connected but also
the genes connecting those pathways, we can use the
<code><a href="../reference/plot_gs2gene.html">plot_gs2gene()</a></code> function instead:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/plot_gs2gene.html">plot_gs2gene</a></span><span class="op">(</span></span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorGsBy <span class="op">=</span> <span class="st">"Estimate"</span>, </span>
<span>        labelGene <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        geneNodeSize  <span class="op">=</span> <span class="fl">1</span>, </span>
<span>        edgeAlpha <span class="op">=</span> <span class="fl">0.1</span>, </span>
<span>        gsNameSize <span class="op">=</span> <span class="fl">2</span>,</span>
<span>        filterGeneBy <span class="op">=</span> <span class="fl">3</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradient2</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-9-1.png" alt="*Pathways significantly perturbed by the E2+R5020 combintation treatment and genes contained in at least 3 of those pathways.*" width="1152"><p class="caption">
<em>Pathways significantly perturbed by the E2+R5020 combintation
treatment and genes contained in at least 3 of those pathways.</em>
</p>
</div>
<p>However, since there are a large number of genes in each pathway, the
plot above was quite messy and it was unrealistic to plot all genes’
names. So it is recommend to filter pathway genes by their perturbation
scores or logFCs.</p>
<p>For example, we can rank genes by the absolute values of their mean
single-sample logFCs and only focus on genes that were ranked in the top
200 of the list.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meanFC</span> <span class="op">&lt;-</span> <span class="va">weightedFC</span><span class="op">$</span><span class="va">logFC</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">.</span><span class="op">[</span>, <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, <span class="st">"E2"</span>, negate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">mean</span> <span class="op">)</span></span>
<span><span class="va">top200_gene</span> <span class="op">&lt;-</span> <span class="va">meanFC</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span>decreasing <span class="op">=</span> <span class="cn">TRUE</span>, <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">.</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">200</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">top200_FC</span> <span class="op">&lt;-</span> <span class="va">meanFC</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">.</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">top200_gene</span><span class="op">]</span></span>
<span><span class="va">top200_FC</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">top200_FC</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">"Up-Regulated"</span>, <span class="st">"Down-Regulated"</span><span class="op">)</span></span></code></pre></div>
<p>When we provide genes’ logFCs as a named vector through the
<code>geneFC</code> parameter, only pathway genes with logFCs provided
will be plotted and gene nodes will be colored by genes’ directions of
changes. The names of the logFC vector must be genes’ entrez IDs in the
format of “ENTREZID:XXXX”, as pathway topology matrices retrieved
through <code><a href="../reference/retrieve_topology.html">retrieve_topology()</a></code> always use entrez IDs as
identifiers.</p>
<p>However, it is not going to be informative to label genes with their
entrez IDs. So the same as the <code><a href="../reference/plot_gene_contribution.html">plot_gene_contribution()</a></code>
function, we can provide a <code>data.frame</code> to match genes’
entrez IDs to our chosen identifiers through the
<code>mapEntrezID</code> parameter in the <code><a href="../reference/plot_gs2gene.html">plot_gs2gene()</a></code>
function too.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">Estimate</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">"Activated"</span>, <span class="st">"Inhibited"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/plot_gs2gene.html">plot_gs2gene</a></span><span class="op">(</span></span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorGsBy <span class="op">=</span> <span class="st">"status"</span>, </span>
<span>        geneFC <span class="op">=</span> <span class="va">top200_FC</span>, </span>
<span>        mapEntrezID <span class="op">=</span> <span class="va">entrez2name</span>, </span>
<span>        gsNameSize <span class="op">=</span> <span class="fl">3</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkred"</span>, <span class="st">"lightskyblue"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"blue"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-10-1.png" alt="*Pathways significantly perturbed by the E2+R5020 combintation treatment, and pathway genes with top 200 magnitudes of changes among all E2+R5020-treated samples. Both pathways and genes were colored by their directions of changes.*" width="960"><p class="caption">
<em>Pathways significantly perturbed by the E2+R5020 combintation
treatment, and pathway genes with top 200 magnitudes of changes among
all E2+R5020-treated samples. Both pathways and genes were colored by
their directions of changes.</em>
</p>
</div>
<p>We can also filters genes’ by their contributions to a pathway
perturbations.To help with that, <code>sSNAPPY</code> provides an option
to rank genes’ perturbation scores within each sample for each
pathway.</p>
<p>If in a given pathway, both positive and negative gene-wise
perturbation scores exist, positive and negative scores are ranked
separately, where the larger a positive rank, the more the gene
contributed to the pathway’s activation, and the smaller a negative
rank, the more the gene contributed to the pathways’ inhibition.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneRank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rank_gene_pert.html">rank_gene_pert</a></span><span class="op">(</span><span class="va">genePertScore</span>, <span class="va">gsTopology</span><span class="op">)</span></span></code></pre></div>
<p>Depending on the biological question, it could be interesting to plot
only the pathway genes with the same directions of changes as the
pathway they belonged to, and ignoring those genes that were
antagonizing the pathway perturbation.</p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Sales G, Calura E, Cavalieri D, Romualdi C (2012). “graphite - a
Bioconductor package to convert pathway topology to gene network.” BMC
Bioinformatics. <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-13-20" class="external-link uri">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-13-20</a>.</li>
<li>Tarca, Adi Laurentiu et al. (2009). “A novel signaling pathway
impact analysis.” Bioinformatics vol. 25,1 : 75-82. <a href="doi:10.1093/bioinformatics/btn577" class="uri">doi:10.1093/bioinformatics/btn577</a>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.2 (2023-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] graphite_1.46.0   htmltools_0.5.6.1 DT_0.30           cowplot_1.1.1    </span></span>
<span><span class="co">##  [5] magrittr_2.0.3    lubridate_1.9.3   forcats_1.0.0     stringr_1.5.0    </span></span>
<span><span class="co">##  [9] dplyr_1.1.3       purrr_1.0.2       readr_2.1.4       tidyr_1.3.0      </span></span>
<span><span class="co">## [13] tibble_3.2.1      tidyverse_2.0.0   sSNAPPY_1.3.5     ggplot2_3.4.4    </span></span>
<span><span class="co">## [17] BiocStyle_2.28.1 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RColorBrewer_1.1-3          jsonlite_1.8.7             </span></span>
<span><span class="co">##   [3] farver_2.1.1                rmarkdown_2.25             </span></span>
<span><span class="co">##   [5] fs_1.6.3                    zlibbioc_1.46.0            </span></span>
<span><span class="co">##   [7] ragg_1.2.6                  vctrs_0.6.4                </span></span>
<span><span class="co">##   [9] memoise_2.0.1               RCurl_1.98-1.12            </span></span>
<span><span class="co">##  [11] S4Arrays_1.0.6              curl_5.1.0                 </span></span>
<span><span class="co">##  [13] sass_0.4.7                  bslib_0.5.1                </span></span>
<span><span class="co">##  [15] htmlwidgets_1.6.2           desc_1.4.2                 </span></span>
<span><span class="co">##  [17] plyr_1.8.9                  cachem_1.0.8               </span></span>
<span><span class="co">##  [19] igraph_1.5.1                lifecycle_1.0.3            </span></span>
<span><span class="co">##  [21] pkgconfig_2.0.3             Matrix_1.6-1.1             </span></span>
<span><span class="co">##  [23] R6_2.5.1                    fastmap_1.1.1              </span></span>
<span><span class="co">##  [25] GenomeInfoDbData_1.2.10     MatrixGenerics_1.12.3      </span></span>
<span><span class="co">##  [27] digest_0.6.33               colorspace_2.1-0           </span></span>
<span><span class="co">##  [29] AnnotationDbi_1.62.2        S4Vectors_0.38.2           </span></span>
<span><span class="co">##  [31] rprojroot_2.0.3             textshaping_0.3.7          </span></span>
<span><span class="co">##  [33] crosstalk_1.2.0             GenomicRanges_1.52.1       </span></span>
<span><span class="co">##  [35] RSQLite_2.3.2               org.Hs.eg.db_3.17.0        </span></span>
<span><span class="co">##  [37] labeling_0.4.3              fansi_1.0.5                </span></span>
<span><span class="co">##  [39] timechange_0.2.0            httr_1.4.7                 </span></span>
<span><span class="co">##  [41] polyclip_1.10-6             abind_1.4-5                </span></span>
<span><span class="co">##  [43] mgcv_1.9-0                  compiler_4.3.2             </span></span>
<span><span class="co">##  [45] bit64_4.0.5                 withr_2.5.1                </span></span>
<span><span class="co">##  [47] pander_0.6.5                BiocParallel_1.34.2        </span></span>
<span><span class="co">##  [49] viridis_0.6.4               DBI_1.1.3                  </span></span>
<span><span class="co">##  [51] ggforce_0.4.1               MASS_7.3-60                </span></span>
<span><span class="co">##  [53] rappdirs_0.3.3              DelayedArray_0.26.7        </span></span>
<span><span class="co">##  [55] tools_4.3.2                 glue_1.6.2                 </span></span>
<span><span class="co">##  [57] nlme_3.1-163                grid_4.3.2                 </span></span>
<span><span class="co">##  [59] reshape2_1.4.4              generics_0.1.3             </span></span>
<span><span class="co">##  [61] gtable_0.3.4                tzdb_0.4.0                 </span></span>
<span><span class="co">##  [63] hms_1.1.3                   tidygraph_1.2.3            </span></span>
<span><span class="co">##  [65] utf8_1.2.4                  XVector_0.40.0             </span></span>
<span><span class="co">##  [67] BiocGenerics_0.46.0         ggrepel_0.9.4              </span></span>
<span><span class="co">##  [69] pillar_1.9.0                limma_3.56.2               </span></span>
<span><span class="co">##  [71] splines_4.3.2               tweenr_2.0.2               </span></span>
<span><span class="co">##  [73] lattice_0.21-9              bit_4.0.5                  </span></span>
<span><span class="co">##  [75] tidyselect_1.2.0            locfit_1.5-9.8             </span></span>
<span><span class="co">##  [77] Biostrings_2.68.1           knitr_1.44                 </span></span>
<span><span class="co">##  [79] gridExtra_2.3               bookdown_0.36              </span></span>
<span><span class="co">##  [81] IRanges_2.34.1              edgeR_3.42.4               </span></span>
<span><span class="co">##  [83] SummarizedExperiment_1.30.2 stats4_4.3.2               </span></span>
<span><span class="co">##  [85] xfun_0.40                   graphlayouts_1.0.1         </span></span>
<span><span class="co">##  [87] Biobase_2.60.0              matrixStats_1.0.0          </span></span>
<span><span class="co">##  [89] pheatmap_1.0.12             stringi_1.7.12             </span></span>
<span><span class="co">##  [91] yaml_2.3.7                  evaluate_0.22              </span></span>
<span><span class="co">##  [93] codetools_0.2-19            ggraph_2.1.0               </span></span>
<span><span class="co">##  [95] BiocManager_1.30.22         graph_1.78.0               </span></span>
<span><span class="co">##  [97] cli_3.6.1                   systemfonts_1.0.5          </span></span>
<span><span class="co">##  [99] munsell_0.5.0               jquerylib_0.1.4            </span></span>
<span><span class="co">## [101] Rcpp_1.0.11                 GenomeInfoDb_1.36.4        </span></span>
<span><span class="co">## [103] png_0.1-8                   parallel_4.3.2             </span></span>
<span><span class="co">## [105] ellipsis_0.3.2              pkgdown_2.0.7              </span></span>
<span><span class="co">## [107] blob_1.2.4                  bitops_1.0-7               </span></span>
<span><span class="co">## [109] viridisLite_0.4.2           scales_1.2.1               </span></span>
<span><span class="co">## [111] crayon_1.5.2                rlang_1.1.1                </span></span>
<span><span class="co">## [113] KEGGREST_1.40.1</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://github.com/Wenjun-Liu" class="external-link">Wenjun Liu</a>, Stephen Pederson.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
