<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>sSNAPPY: Singel Sample directioNAl Pathway Perturbation analYsis • sSNAPPY</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="sSNAPPY: Singel Sample directioNAl Pathway Perturbation analYsis">
<meta property="og:description" content="sSNAPPY">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">sSNAPPY</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/sSNAPPY.html">Get Started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/Wenjun-Liu/sSNAPPY" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="sSNAPPY_files/htmlwidgets-1.6.2/htmlwidgets.js"></script><link href="sSNAPPY_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="sSNAPPY_files/datatables-binding-0.28/datatables.js"></script><link href="sSNAPPY_files/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="sSNAPPY_files/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="sSNAPPY_files/dt-core-1.13.4/js/jquery.dataTables.min.js"></script><link href="sSNAPPY_files/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="sSNAPPY_files/nouislider-7.0.10/jquery.nouislider.min.js"></script><link href="sSNAPPY_files/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="sSNAPPY_files/selectize-0.12.0/selectize.min.js"></script><link href="sSNAPPY_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="sSNAPPY_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>sSNAPPY: Singel Sample directioNAl Pathway
Perturbation analYsis</h1>
                        <h4 data-toc-skip class="author">Wenjun Nora
Liu</h4>
            <address class="author_afil">
      Dame Roma Mitchell Cancer Research Laboratories, Adelaide Medical
School, University of
Adelaide<br><a class="author_email" href="mailto:#"></a><a href="mailto:wenjun.liu@adelaide.edu.au" class="email">wenjun.liu@adelaide.edu.au</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/Wenjun-Liu/sSNAPPY/blob/HEAD/vignettes/sSNAPPY.Rmd" class="external-link"><code>vignettes/sSNAPPY.Rmd</code></a></small>
      <div class="hidden name"><code>sSNAPPY.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette demonstrates how to use the package
<code>sSNAPPY</code> to compute directional single sample pathway
perturbation scores by incorporating pathway topologies, utilizing
sample permutation to test the significance of individual scores and
comparing average pathway activities across treatments.</p>
<p>The package also provides many powerful and easy-to-use visualisation
functions that helps visualising significantly perturbed pathways as
networks, detecting community structures in pathway networks, and
revealing pathway genes’ involvement in the perturbation.</p>
</div>
<div class="section level2">
<h2 id="to-get-ready">To get ready<a class="anchor" aria-label="anchor" href="#to-get-ready"></a>
</h2>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>The package <code>sSNAPPY</code> can be installed using the package
<code>BiocManager</code></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"BiocManager"</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="co"># Other packages required in this vignette</span></span>
<span><span class="va">pkg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tidyverse"</span>, <span class="st">"magrittr"</span>, <span class="st">"ggplot2"</span>, <span class="st">"cowplot"</span>, <span class="st">"DT"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="va">pkg</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"sSNAPPY"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"htmltools"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wenjun-liu.github.io/sSNAPPY/">sSNAPPY</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/" class="external-link">cowplot</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/DT" class="external-link">DT</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/htmltools" class="external-link">htmltools</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-data">Load data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h3>
<p>The example dataset used for this tutorial can be loaded with
<code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code> as shown below. It’s a subset of data retrieved from
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4928895/" class="external-link">Singhal
H et al. 2016</a>, where ER-positive primary breast cancer tumour
tissues collected from 12 patients were split into tissue fragments of
equal sizes for different treatments.</p>
<p>For this tutorial, we are only looking at the RNA-seq data from
samples that were treated with vehicle, R5020(progesterone) OR
E2(estrogen) + R5020 for 48 hrs. Tumour specimens were collected from 5
patients, giving rise to a total of 15 samples. A more detailed
description of the dataset can be assessed through the help page
(<code><a href="../reference/logCPM_example.html">?logCPM_example</a></code> and <code><a href="../reference/metadata_example.html">?metadata_example</a></code>).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">)</span></span>
<span><span class="co"># check if samples included in the logCPM matrix and metadata dataframe are identical</span></span>
<span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setequal</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span>, <span class="va">metadata_example</span><span class="op">$</span><span class="va">sample</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View sample metadata</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="va">metadata_example</span>,  filter <span class="op">=</span> <span class="st">"top"</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-58437ed608bdba60264e" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-58437ed608bdba60264e">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;N2&quot;,&quot;N3&quot;,&quot;P4&quot;,&quot;P5&quot;,&quot;P6&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Vehicle&quot;,&quot;E2&quot;,&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020_N2_48&quot;,&quot;E2+R5020_N3_48&quot;,&quot;E2+R5020_P4_48&quot;,&quot;E2+R5020_P5_48&quot;,&quot;E2+R5020_P6_48&quot;,&quot;R5020_N2_48&quot;,&quot;R5020_N3_48&quot;,&quot;R5020_P4_48&quot;,&quot;R5020_P5_48&quot;,&quot;R5020_P6_48&quot;,&quot;Vehicle_N2_48&quot;,&quot;Vehicle_N3_48&quot;,&quot;Vehicle_P4_48&quot;,&quot;Vehicle_P5_48&quot;,&quot;Vehicle_P6_48&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Negative&quot;,&quot;Positive&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["N2","N2","N2","N3","N3","N3","P4","P4","P4","P5","P5","P5","P6","P6","P6"],["Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020","Vehicle","E2+R5020","R5020"],["Vehicle_N2_48","E2+R5020_N2_48","R5020_N2_48","Vehicle_N3_48","E2+R5020_N3_48","R5020_N3_48","Vehicle_P4_48","E2+R5020_P4_48","R5020_P4_48","Vehicle_P5_48","E2+R5020_P5_48","R5020_P5_48","Vehicle_P6_48","E2+R5020_P6_48","R5020_P6_48"],["Negative","Negative","Negative","Negative","Negative","Negative","Positive","Positive","Positive","Positive","Positive","Positive","Positive","Positive","Positive"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>patient<\/th>\n      <th>treatment<\/th>\n      <th>sample<\/th>\n      <th>PR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="section level2">
<h2 id="ssnappy-workflow">
<code>sSNAPPY</code> workflow<a class="anchor" aria-label="anchor" href="#ssnappy-workflow"></a>
</h2>
<div class="section level3">
<h3 id="compute-weighted-single-sample-logfcs-sslogfcs">Compute weighted single-sample logFCs (ssLogFCs)<a class="anchor" aria-label="anchor" href="#compute-weighted-single-sample-logfcs-sslogfcs"></a>
</h3>
<p>It is expected that the logCPM matrix will be filtered to remove
undetectable genes and normalised to correct for library sizes or other
systematic artefacts, such as gene length or GC contents, prior to
applying the <code>sSNAPPY</code> workflow. Filtration and normalisation
have been performed on the example dataset.</p>
<p>Before single-sample logFCs (ssLogFCs) can be computed, row names of
the logCPM matrix need to be converted to <code>entrez ID</code>. This
is because all the pathway topology information retrieved will be in
<code>entrez ID</code>. The conversion can be achieved through
bioconductor packages <code>AnnotationHub</code> and
<code>ensembldb</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">logCPM_example</span><span class="op">)</span></span></code></pre></div>
<p>To compute the ssLogFCs, samples must be in matching pairs. In our
example data, treated samples were matched to the corresponding control
samples derived from the same patients. Therefore the
<code>groupBy</code> parameter of the <code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code>
functions will be set to be “patient”.</p>
<p><code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code> requires both the logCPM matrix and
sample metadata as input. The column names of the logCPM matrix should
be sample names, matching a column in the metadata. Name of the column
in the metadata with sample name will be provided as the
<code>sampleColumn</code> parameter. The function also requires the name
of the metadata column containing treatment information to be specified.
The column with treatment information must be a factor with the control
treatment set to be the reference level.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check that the baseline level of the treatment column is the control</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">$</span><span class="va">treatment</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Vehicle"</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#compute weighted single sample logFCs</span></span>
<span><span class="va">weightedFC</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/weight_ss_fc.html">weight_ss_fc</a></span><span class="op">(</span><span class="va">logCPM_example</span>, metadata <span class="op">=</span> <span class="va">metadata_example</span>,</span>
<span>                           groupBy  <span class="op">=</span> <span class="st">"patient"</span>, sampleColumn <span class="op">=</span> <span class="st">"sample"</span>, </span>
<span>                           treatColumn <span class="op">=</span> <span class="st">"treatment"</span><span class="op">)</span></span></code></pre></div>
<p>The <code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code> function firstly computes raw
ssLogFCs for each gene by subtracting logCPM values of the control
sample from the logCPM values of treated samples within each
patient.</p>
<p>It has been demonstrated previously that in RNA-seq data, lowly
expressed genes turn to have larger variances, which is also
demonstrated in the plot below. To reduce the impact of this artifact,
<code>weight_ss_fc</code> also weights each ssLogFCs by estimating the
mean-variance relationship of logCPM, and defining the gene-wise weight
to be the inverse of the predicted variance of that gene’s mean logCPM
value.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">logCPM_example</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>        sd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">.</span>, <span class="fl">1</span>, <span class="va">sd</span><span class="op">)</span>,</span>
<span>        mean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">.</span>, <span class="fl">1</span>, <span class="va">mean</span><span class="op">)</span></span>
<span>        <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">mean</span>, <span class="va">sd</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html" class="external-link">geom_smooth</a></span><span class="op">(</span></span>
<span>        method <span class="op">=</span> <span class="st">"loess"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>        x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="va">Gene</span><span class="op">-</span><span class="va">wise</span><span class="op">~</span><span class="va">average</span><span class="op">~</span><span class="va">expression</span><span class="op">~</span><span class="op">(</span><span class="fu">bar</span><span class="op">(</span><span class="va">mu</span><span class="op">[</span><span class="va">g</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="va">Gene</span><span class="op">-</span><span class="va">wise</span><span class="op">~</span><span class="va">standard</span><span class="op">~</span><span class="va">deviation</span><span class="op">~</span><span class="op">(</span><span class="va">sigma</span><span class="op">[</span><span class="va">g</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>        axis.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">14</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/lowess-1.png" alt="*Gene-wise standard deviations are plotted against the mean logCPM values with mean-variance trend modelled by a loess fit. Genes with low expression values tend to have a larger variance.*" width="576"><p class="caption">
<em>Gene-wise standard deviations are plotted against the mean logCPM
values with mean-variance trend modelled by a loess fit. Genes with low
expression values tend to have a larger variance.</em>
</p>
</div>
<p>The output of the <code><a href="../reference/weight_ss_fc.html">weight_ss_fc()</a></code> function is a list with
two element, where one is the weighted ssLogFCs matrix and the other is
a vector of gene-wise weights.</p>
</div>
<div class="section level3">
<h3 id="retrieve-pathway-topologies-in-the-required-format">Retrieve pathway topologies in the required format<a class="anchor" aria-label="anchor" href="#retrieve-pathway-topologies-in-the-required-format"></a>
</h3>
<p><em>sSNAPPY</em> adopts the pathway perturbation scoring algorithm
proposed in <em><a href="https://bioconductor.org/packages/3.17/SPIA" class="external-link">SPIA</a></em>, which
makes use of gene-set topologies and gene-gene interaction to propagate
pathway genes’ logFCs down the topologies to compute pathway
perturbation scores, where signs of scores reflect pathways’ potential
directions of changes.</p>
<p>Therefore, pathway topology information needs to be firstly retrieved
from a chosen database and converted to weight adjacency matrices, the
format required to apply the scoring algorithm.</p>
<p>This step is achieved through a chain of functions that are part of
the <em><a href="https://bioconductor.org/packages/3.17/grapghite" class="external-link">grapghite</a></em>,
which have been nested into one simple function in <code>sSNAPPY</code>
called <code><a href="../reference/retrieve_topology.html">retrieve_topology()</a></code>. The
<code>retrieve_topology</code> function now supports a large list of
species and databases. Databases that are currently supported for human
are:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sales-lab/graphite" class="external-link">graphite</a></span><span class="op">)</span></span>
<span><span class="fu">graphite</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graphite/man/pathwayDatabases.html" class="external-link">pathwayDatabases</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">species</span> <span class="op">==</span>  <span class="st">"hsapiens"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">pander</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pander/man/pander.html" class="external-link">pander</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table style="width:36%;" class="table">
<colgroup>
<col width="15%">
<col width="20%">
</colgroup>
<thead><tr class="header">
<th align="center">species</th>
<th align="center">database</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">kegg</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">panther</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">pathbank</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">pharmgkb</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">reactome</td>
</tr>
<tr class="even">
<td align="center">hsapiens</td>
<td align="center">smpdb</td>
</tr>
<tr class="odd">
<td align="center">hsapiens</td>
<td align="center">wikipathways</td>
</tr>
</tbody>
</table>
<p>The retrieved topology information will be a list where each element
corresponds a pathway. It’s recommended to save the list as a file so
this step only needs to be performed once for each database.</p>
<p>This vignette uses <em>KEGG</em> pathways in human as an example.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsTopology</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/retrieve_topology.html">retrieve_topology</a></span><span class="op">(</span>database <span class="op">=</span> <span class="st">"kegg"</span>, species <span class="op">=</span> <span class="st">"hsapiens"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsTopology</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "kegg.Glycolysis / Gluconeogenesis"            </span></span>
<span><span class="co">## [2] "kegg.Citrate cycle (TCA cycle)"               </span></span>
<span><span class="co">## [3] "kegg.Pentose phosphate pathway"               </span></span>
<span><span class="co">## [4] "kegg.Pentose and glucuronate interconversions"</span></span>
<span><span class="co">## [5] "kegg.Fructose and mannose metabolism"         </span></span>
<span><span class="co">## [6] "kegg.Galactose metabolism"</span></span></code></pre>
<p>If only selected biological processes are of interest to your
research, it’s possible to only retrieve the topologies of those
pathways by specifying keywords. For example, to retrieve all
metabolism-related <em>KEGG</em> pathways:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gsTopology_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/retrieve_topology.html">retrieve_topology</a></span><span class="op">(</span></span>
<span>  database <span class="op">=</span> <span class="st">"kegg"</span>,</span>
<span>  species <span class="op">=</span> <span class="st">"hsapiens"</span>, </span>
<span>  keyword <span class="op">=</span> <span class="st">"metabolism"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gsTopology_sub</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "kegg.Fructose and mannose metabolism"             </span></span>
<span><span class="co">##  [2] "kegg.Galactose metabolism"                        </span></span>
<span><span class="co">##  [3] "kegg.Ascorbate and aldarate metabolism"           </span></span>
<span><span class="co">##  [4] "kegg.Purine metabolism"                           </span></span>
<span><span class="co">##  [5] "kegg.Caffeine metabolism"                         </span></span>
<span><span class="co">##  [6] "kegg.Pyrimidine metabolism"                       </span></span>
<span><span class="co">##  [7] "kegg.Alanine, aspartate and glutamate metabolism" </span></span>
<span><span class="co">##  [8] "kegg.Glycine, serine and threonine metabolism"    </span></span>
<span><span class="co">##  [9] "kegg.Cysteine and methionine metabolism"          </span></span>
<span><span class="co">## [10] "kegg.Arginine and proline metabolism"             </span></span>
<span><span class="co">## [11] "kegg.Histidine metabolism"                        </span></span>
<span><span class="co">## [12] "kegg.Tyrosine metabolism"                         </span></span>
<span><span class="co">## [13] "kegg.Phenylalanine metabolism"                    </span></span>
<span><span class="co">## [14] "kegg.Tryptophan metabolism"                       </span></span>
<span><span class="co">## [15] "kegg.beta-Alanine metabolism"                     </span></span>
<span><span class="co">## [16] "kegg.Taurine and hypotaurine metabolism"          </span></span>
<span><span class="co">## [17] "kegg.Phosphonate and phosphinate metabolism"      </span></span>
<span><span class="co">## [18] "kegg.Selenocompound metabolism"                   </span></span>
<span><span class="co">## [19] "kegg.Glutathione metabolism"                      </span></span>
<span><span class="co">## [20] "kegg.Starch and sucrose metabolism"               </span></span>
<span><span class="co">## [21] "kegg.Amino sugar and nucleotide sugar metabolism" </span></span>
<span><span class="co">## [22] "kegg.Glycerolipid metabolism"                     </span></span>
<span><span class="co">## [23] "kegg.Inositol phosphate metabolism"               </span></span>
<span><span class="co">## [24] "kegg.Glycerophospholipid metabolism"              </span></span>
<span><span class="co">## [25] "kegg.Ether lipid metabolism"                      </span></span>
<span><span class="co">## [26] "kegg.Arachidonic acid metabolism"                 </span></span>
<span><span class="co">## [27] "kegg.Linoleic acid metabolism"                    </span></span>
<span><span class="co">## [28] "kegg.alpha-Linolenic acid metabolism"             </span></span>
<span><span class="co">## [29] "kegg.Sphingolipid metabolism"                     </span></span>
<span><span class="co">## [30] "kegg.Pyruvate metabolism"                         </span></span>
<span><span class="co">## [31] "kegg.Glyoxylate and dicarboxylate metabolism"     </span></span>
<span><span class="co">## [32] "kegg.Propanoate metabolism"                       </span></span>
<span><span class="co">## [33] "kegg.Butanoate metabolism"                        </span></span>
<span><span class="co">## [34] "kegg.Thiamine metabolism"                         </span></span>
<span><span class="co">## [35] "kegg.Riboflavin metabolism"                       </span></span>
<span><span class="co">## [36] "kegg.Vitamin B6 metabolism"                       </span></span>
<span><span class="co">## [37] "kegg.Nicotinate and nicotinamide metabolism"      </span></span>
<span><span class="co">## [38] "kegg.Lipoic acid metabolism"                      </span></span>
<span><span class="co">## [39] "kegg.Retinol metabolism"                          </span></span>
<span><span class="co">## [40] "kegg.Porphyrin metabolism"                        </span></span>
<span><span class="co">## [41] "kegg.Nitrogen metabolism"                         </span></span>
<span><span class="co">## [42] "kegg.Sulfur metabolism"                           </span></span>
<span><span class="co">## [43] "kegg.Metabolism of xenobiotics by cytochrome P450"</span></span>
<span><span class="co">## [44] "kegg.Drug metabolism - cytochrome P450"           </span></span>
<span><span class="co">## [45] "kegg.Drug metabolism - other enzymes"             </span></span>
<span><span class="co">## [46] "kegg.Carbon metabolism"                           </span></span>
<span><span class="co">## [47] "kegg.2-Oxocarboxylic acid metabolism"             </span></span>
<span><span class="co">## [48] "kegg.Fatty acid metabolism"                       </span></span>
<span><span class="co">## [49] "kegg.Nucleotide metabolism"                       </span></span>
<span><span class="co">## [50] "kegg.Cholesterol metabolism"                      </span></span>
<span><span class="co">## [51] "kegg.Central carbon metabolism in cancer"         </span></span>
<span><span class="co">## [52] "kegg.Choline metabolism in cancer"</span></span></code></pre>
<p>It is also possible to provide multiple databases’ names and/or
multiple keywords for a focused analysis.</p>
<pre><code><span><span class="co">##   [1] "kegg.Fructose and mannose metabolism"                                           </span></span>
<span><span class="co">##   [2] "kegg.Galactose metabolism"                                                      </span></span>
<span><span class="co">##   [3] "kegg.Ascorbate and aldarate metabolism"                                         </span></span>
<span><span class="co">##   [4] "kegg.Purine metabolism"                                                         </span></span>
<span><span class="co">##   [5] "kegg.Caffeine metabolism"                                                       </span></span>
<span><span class="co">##   [6] "kegg.Pyrimidine metabolism"                                                     </span></span>
<span><span class="co">##   [7] "kegg.Alanine, aspartate and glutamate metabolism"                               </span></span>
<span><span class="co">##   [8] "kegg.Glycine, serine and threonine metabolism"                                  </span></span>
<span><span class="co">##   [9] "kegg.Cysteine and methionine metabolism"                                        </span></span>
<span><span class="co">##  [10] "kegg.Arginine and proline metabolism"                                           </span></span>
<span><span class="co">##  [11] "kegg.Histidine metabolism"                                                      </span></span>
<span><span class="co">##  [12] "kegg.Tyrosine metabolism"                                                       </span></span>
<span><span class="co">##  [13] "kegg.Phenylalanine metabolism"                                                  </span></span>
<span><span class="co">##  [14] "kegg.Tryptophan metabolism"                                                     </span></span>
<span><span class="co">##  [15] "kegg.beta-Alanine metabolism"                                                   </span></span>
<span><span class="co">##  [16] "kegg.Taurine and hypotaurine metabolism"                                        </span></span>
<span><span class="co">##  [17] "kegg.Phosphonate and phosphinate metabolism"                                    </span></span>
<span><span class="co">##  [18] "kegg.Selenocompound metabolism"                                                 </span></span>
<span><span class="co">##  [19] "kegg.Glutathione metabolism"                                                    </span></span>
<span><span class="co">##  [20] "kegg.Starch and sucrose metabolism"                                             </span></span>
<span><span class="co">##  [21] "kegg.Amino sugar and nucleotide sugar metabolism"                               </span></span>
<span><span class="co">##  [22] "kegg.Glycerolipid metabolism"                                                   </span></span>
<span><span class="co">##  [23] "kegg.Inositol phosphate metabolism"                                             </span></span>
<span><span class="co">##  [24] "kegg.Glycerophospholipid metabolism"                                            </span></span>
<span><span class="co">##  [25] "kegg.Ether lipid metabolism"                                                    </span></span>
<span><span class="co">##  [26] "kegg.Arachidonic acid metabolism"                                               </span></span>
<span><span class="co">##  [27] "kegg.Linoleic acid metabolism"                                                  </span></span>
<span><span class="co">##  [28] "kegg.alpha-Linolenic acid metabolism"                                           </span></span>
<span><span class="co">##  [29] "kegg.Sphingolipid metabolism"                                                   </span></span>
<span><span class="co">##  [30] "kegg.Pyruvate metabolism"                                                       </span></span>
<span><span class="co">##  [31] "kegg.Glyoxylate and dicarboxylate metabolism"                                   </span></span>
<span><span class="co">##  [32] "kegg.Propanoate metabolism"                                                     </span></span>
<span><span class="co">##  [33] "kegg.Butanoate metabolism"                                                      </span></span>
<span><span class="co">##  [34] "kegg.Thiamine metabolism"                                                       </span></span>
<span><span class="co">##  [35] "kegg.Riboflavin metabolism"                                                     </span></span>
<span><span class="co">##  [36] "kegg.Vitamin B6 metabolism"                                                     </span></span>
<span><span class="co">##  [37] "kegg.Nicotinate and nicotinamide metabolism"                                    </span></span>
<span><span class="co">##  [38] "kegg.Lipoic acid metabolism"                                                    </span></span>
<span><span class="co">##  [39] "kegg.Retinol metabolism"                                                        </span></span>
<span><span class="co">##  [40] "kegg.Porphyrin metabolism"                                                      </span></span>
<span><span class="co">##  [41] "kegg.Nitrogen metabolism"                                                       </span></span>
<span><span class="co">##  [42] "kegg.Sulfur metabolism"                                                         </span></span>
<span><span class="co">##  [43] "kegg.Metabolism of xenobiotics by cytochrome P450"                              </span></span>
<span><span class="co">##  [44] "kegg.Drug metabolism - cytochrome P450"                                         </span></span>
<span><span class="co">##  [45] "kegg.Drug metabolism - other enzymes"                                           </span></span>
<span><span class="co">##  [46] "kegg.Carbon metabolism"                                                         </span></span>
<span><span class="co">##  [47] "kegg.2-Oxocarboxylic acid metabolism"                                           </span></span>
<span><span class="co">##  [48] "kegg.Fatty acid metabolism"                                                     </span></span>
<span><span class="co">##  [49] "kegg.Nucleotide metabolism"                                                     </span></span>
<span><span class="co">##  [50] "kegg.Estrogen signaling pathway"                                                </span></span>
<span><span class="co">##  [51] "kegg.Cholesterol metabolism"                                                    </span></span>
<span><span class="co">##  [52] "kegg.Central carbon metabolism in cancer"                                       </span></span>
<span><span class="co">##  [53] "kegg.Choline metabolism in cancer"                                              </span></span>
<span><span class="co">##  [54] "reactome.Metabolism"                                                            </span></span>
<span><span class="co">##  [55] "reactome.Inositol phosphate metabolism"                                         </span></span>
<span><span class="co">##  [56] "reactome.PI Metabolism"                                                         </span></span>
<span><span class="co">##  [57] "reactome.Phospholipid metabolism"                                               </span></span>
<span><span class="co">##  [58] "reactome.Nucleotide metabolism"                                                 </span></span>
<span><span class="co">##  [59] "reactome.Sulfur amino acid metabolism"                                          </span></span>
<span><span class="co">##  [60] "reactome.Glycosaminoglycan metabolism"                                          </span></span>
<span><span class="co">##  [61] "reactome.Integration of energy metabolism"                                      </span></span>
<span><span class="co">##  [62] "reactome.Keratan sulfate/keratin metabolism"                                    </span></span>
<span><span class="co">##  [63] "reactome.Heparan sulfate/heparin (HS-GAG) metabolism"                           </span></span>
<span><span class="co">##  [64] "reactome.Glycosphingolipid metabolism"                                          </span></span>
<span><span class="co">##  [65] "reactome.Lipoprotein metabolism"                                                </span></span>
<span><span class="co">##  [66] "reactome.Chondroitin sulfate/dermatan sulfate metabolism"                       </span></span>
<span><span class="co">##  [67] "reactome.Porphyrin metabolism"                                                  </span></span>
<span><span class="co">##  [68] "reactome.Estrogen biosynthesis"                                                 </span></span>
<span><span class="co">##  [69] "reactome.Bile acid and bile salt metabolism"                                    </span></span>
<span><span class="co">##  [70] "reactome.Non-coding RNA Metabolism"                                             </span></span>
<span><span class="co">##  [71] "reactome.Metabolism of steroid hormones"                                        </span></span>
<span><span class="co">##  [72] "reactome.Cobalamin (Cbl, vitamin B12) transport and metabolism"                 </span></span>
<span><span class="co">##  [73] "reactome.Metabolism of folate and pterines"                                     </span></span>
<span><span class="co">##  [74] "reactome.Biotin transport and metabolism"                                       </span></span>
<span><span class="co">##  [75] "reactome.Vitamin D (calciferol) metabolism"                                     </span></span>
<span><span class="co">##  [76] "reactome.Nicotinate metabolism"                                                 </span></span>
<span><span class="co">##  [77] "reactome.Vitamin C (ascorbate) metabolism"                                      </span></span>
<span><span class="co">##  [78] "reactome.Vitamin B2 (riboflavin) metabolism"                                    </span></span>
<span><span class="co">##  [79] "reactome.Metabolism of water-soluble vitamins and cofactors"                    </span></span>
<span><span class="co">##  [80] "reactome.Metabolism of vitamins and cofactors"                                  </span></span>
<span><span class="co">##  [81] "reactome.Vitamin B5 (pantothenate) metabolism"                                  </span></span>
<span><span class="co">##  [82] "reactome.Carnitine metabolism"                                                  </span></span>
<span><span class="co">##  [83] "reactome.Metabolism of nitric oxide: NOS3 activation and regulation"            </span></span>
<span><span class="co">##  [84] "reactome.Metabolism of Angiotensinogen to Angiotensins"                         </span></span>
<span><span class="co">##  [85] "reactome.alpha-linolenic (omega3) and linoleic (omega6) acid metabolism"        </span></span>
<span><span class="co">##  [86] "reactome.Linoleic acid (LA) metabolism"                                         </span></span>
<span><span class="co">##  [87] "reactome.alpha-linolenic acid (ALA) metabolism"                                 </span></span>
<span><span class="co">##  [88] "reactome.Metabolism of amine-derived hormones"                                  </span></span>
<span><span class="co">##  [89] "reactome.Arachidonic acid metabolism"                                           </span></span>
<span><span class="co">##  [90] "reactome.Hyaluronan metabolism"                                                 </span></span>
<span><span class="co">##  [91] "reactome.Metabolism of ingested SeMet, Sec, MeSec into H2Se"                    </span></span>
<span><span class="co">##  [92] "reactome.Selenoamino acid metabolism"                                           </span></span>
<span><span class="co">##  [93] "reactome.Peptide hormone metabolism"                                            </span></span>
<span><span class="co">##  [94] "reactome.Defects in vitamin and cofactor metabolism"                            </span></span>
<span><span class="co">##  [95] "reactome.Defects in biotin (Btn) metabolism"                                    </span></span>
<span><span class="co">##  [96] "reactome.Metabolism of polyamines"                                              </span></span>
<span><span class="co">##  [97] "reactome.Diseases associated with glycosaminoglycan metabolism"                 </span></span>
<span><span class="co">##  [98] "reactome.Glyoxylate metabolism and glycine degradation"                         </span></span>
<span><span class="co">##  [99] "reactome.Peroxisomal lipid metabolism"                                          </span></span>
<span><span class="co">## [100] "reactome.Metabolism of proteins"                                                </span></span>
<span><span class="co">## [101] "reactome.Regulation of lipid metabolism by PPARalpha"                           </span></span>
<span><span class="co">## [102] "reactome.Sialic acid metabolism"                                                </span></span>
<span><span class="co">## [103] "reactome.Sphingolipid metabolism"                                               </span></span>
<span><span class="co">## [104] "reactome.Metabolism of lipids"                                                  </span></span>
<span><span class="co">## [105] "reactome.Fructose metabolism"                                                   </span></span>
<span><span class="co">## [106] "reactome.Diseases of carbohydrate metabolism"                                   </span></span>
<span><span class="co">## [107] "reactome.Diseases of metabolism"                                                </span></span>
<span><span class="co">## [108] "reactome.Surfactant metabolism"                                                 </span></span>
<span><span class="co">## [109] "reactome.Metabolism of fat-soluble vitamins"                                    </span></span>
<span><span class="co">## [110] "reactome.Pyruvate metabolism"                                                   </span></span>
<span><span class="co">## [111] "reactome.Glucose metabolism"                                                    </span></span>
<span><span class="co">## [112] "reactome.Creatine metabolism"                                                   </span></span>
<span><span class="co">## [113] "reactome.Amino acid and derivative metabolism"                                  </span></span>
<span><span class="co">## [114] "reactome.Carbohydrate metabolism"                                               </span></span>
<span><span class="co">## [115] "reactome.Pyruvate metabolism and Citric Acid (TCA) cycle"                       </span></span>
<span><span class="co">## [116] "reactome.Ketone body metabolism"                                                </span></span>
<span><span class="co">## [117] "reactome.RUNX1 regulates estrogen receptor mediated transcription"              </span></span>
<span><span class="co">## [118] "reactome.Metabolism of RNA"                                                     </span></span>
<span><span class="co">## [119] "reactome.Metabolism of steroids"                                                </span></span>
<span><span class="co">## [120] "reactome.Phenylalanine and tyrosine metabolism"                                 </span></span>
<span><span class="co">## [121] "reactome.Aspartate and asparagine metabolism"                                   </span></span>
<span><span class="co">## [122] "reactome.Phenylalanine metabolism"                                              </span></span>
<span><span class="co">## [123] "reactome.Glutamate and glutamine metabolism"                                    </span></span>
<span><span class="co">## [124] "reactome.Fatty acid metabolism"                                                 </span></span>
<span><span class="co">## [125] "reactome.Metabolism of cofactors"                                               </span></span>
<span><span class="co">## [126] "reactome.Triglyceride metabolism"                                               </span></span>
<span><span class="co">## [127] "reactome.Glycogen metabolism"                                                   </span></span>
<span><span class="co">## [128] "reactome.Extra-nuclear estrogen signaling"                                      </span></span>
<span><span class="co">## [129] "reactome.Estrogen-dependent gene expression"                                    </span></span>
<span><span class="co">## [130] "reactome.Regulation of glycolysis by fructose 2,6-bisphosphate metabolism"      </span></span>
<span><span class="co">## [131] "reactome.Estrogen-stimulated signaling through PRKCZ"                           </span></span>
<span><span class="co">## [132] "reactome.Estrogen-dependent nuclear events downstream of ESR-membrane signaling"</span></span>
<span><span class="co">## [133] "reactome.Retinoid metabolism and transport"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="score-single-sample-pathway-perturbation">Score single sample pathway perturbation<a class="anchor" aria-label="anchor" href="#score-single-sample-pathway-perturbation"></a>
</h3>
<p>Once the expression matrix, sample metadata and pathway topologies
are all ready, gene-wise single-sample perturbation scores can be
computed within each sample:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genePertScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/raw_gene_pert.html">raw_gene_pert</a></span><span class="op">(</span><span class="va">weightedFC</span><span class="op">$</span><span class="va">weighted_logFC</span>, <span class="va">gsTopology</span><span class="op">)</span></span></code></pre></div>
<p>and summed to derive pathway perturbation scores for each pathway in
each treated samples.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathwayPertScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pathway_pert.html">pathway_pert</a></span><span class="op">(</span><span class="va">genePertScore</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pathwayPertScore</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##           sample        score                                        gs_name</span></span>
<span><span class="co">## 1 E2+R5020_N2_48  0.006717358 kegg.EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 2    R5020_N2_48  0.006718738 kegg.EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 3 E2+R5020_N3_48  0.002785146 kegg.EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 4    R5020_N3_48 -0.004030015 kegg.EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 5 E2+R5020_P4_48  0.004087122 kegg.EGFR tyrosine kinase inhibitor resistance</span></span>
<span><span class="co">## 6    R5020_P4_48  0.015609886 kegg.EGFR tyrosine kinase inhibitor resistance</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="generate-null-distributions-of-perturbation-scores">Generate null distributions of perturbation scores<a class="anchor" aria-label="anchor" href="#generate-null-distributions-of-perturbation-scores"></a>
</h3>
<p>To derive the empirical p-values for each single-sample perturbation
scores and normalize the raw scores for comparing overall treatment
effects, null distributions of scores for each pathway are generated
through a sample-label permutation approach.</p>
<p>In the permutation, all sample labels will be randomly shuffled and
put into permuted pairs. Permuted single-sample logFCs will be
calculated for each permuted pair of samples, while the reminding
pathway perturbation scoring algorithm remains unchanged. Unless
otherwise specified through the <code>NB</code> parameter, all possible
permuted pairs will be used to construct the null distributions of
perturbation scores.</p>
<p>The output of the <code><a href="../reference/generate_permuted_scores.html">generate_permuted_scores()</a></code> function is
a list where each element is a vector of permuted perturbation scores
for one specific pathway.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">permutedScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_permuted_scores.html">generate_permuted_scores</a></span><span class="op">(</span></span>
<span>  expreMatrix  <span class="op">=</span> <span class="va">logCPM_example</span>, </span>
<span>  gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>  weight <span class="op">=</span> <span class="va">weightedFC</span><span class="op">$</span><span class="va">weight</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="test-significant-perturbation-on">Test significant perturbation on<a class="anchor" aria-label="anchor" href="#test-significant-perturbation-on"></a>
</h3>
<div class="section level4">
<h4 id="single-sample-level">single-sample level<a class="anchor" aria-label="anchor" href="#single-sample-level"></a>
</h4>
<p>After the empirical null distributions are generated, the median and
mad of each distribution will be calculated for each pathway to convert
the test single-sample perturbation scores derived from the
<code>compute_perturbation_score()</code> function to robust z-scores:
<span class="math inline">\((Score - Median)/MAD\)</span>.</p>
<p>Two-sided p-values associated with each perturbation scores are also
computed by the proportion of permuted scores that are more extreme than
the test perturbation score within each pathway. Raw p-values will be
corrected for multiple-testing using a user-defined approach. The
default is <code>fdr</code>.</p>
<p>In a data with N samples, the total number of possible permuted pairs
of samples is <span class="math inline">\(N \times (N-1)\)</span>. When
the sample size is small, small p-values cannot be accurately estimated
so the p-values returned by the <code><a href="../reference/normalise_by_permu.html">normalise_by_permu()</a></code>
function should be intepreted with caution.</p>
<p>The <code><a href="../reference/normalise_by_permu.html">normalise_by_permu()</a></code> function requires the test
perturbation scores and permuted perturbation scores as input. Users can
choose to sort the output by p-values, gene-set names or sample
names.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normalisedScores</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalise_by_permu.html">normalise_by_permu</a></span><span class="op">(</span><span class="va">permutedScore</span>, <span class="va">pathwayPertScore</span>, sortBy <span class="op">=</span> <span class="st">"pvalue"</span><span class="op">)</span></span></code></pre></div>
<p>In this example dataset, none of the pathway was considered to be
significantly perturbed within individual samples using a FDR cut-off of
0.05.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">adjPvalue</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">## [1] MAD       MEDIAN    gs_name   sample    score     robustZ   pvalue   </span></span>
<span><span class="co">## [8] adjPvalue</span></span>
<span><span class="co">## &lt;0 rows&gt; (or 0-length row.names)</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="treatment-level">treatment-level<a class="anchor" aria-label="anchor" href="#treatment-level"></a>
</h4>
<p>In addition to testing pathway perturbations at single-sample level,
normalised perturbation scores can also be used to model mean treatment
effects within a group, such as within each treatment group. An
advantage of this method is that it has a high level of flexibility that
allows us to incorporate confounding factors as co-factors or
co-variates to offset their effects.</p>
<p>For example, in the example data-set, samples were collected from
patients with different progesterone receptor (PR) statuses. Knowing
that PR status would affect how tumour tissues responded to estrogen
and/or progesterone treatments, we can include it as a cofactor to
offset its confounding effect.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="va">normalisedScores</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">metadata_example</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span>f <span class="op">=</span> <span class="va">.</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co">#.["Estrogen signaling pathway"] %&gt;%</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">robustZ</span> <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">treatment</span> <span class="op">+</span> <span class="va">PR</span>, data <span class="op">=</span> <span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">summary</span><span class="op">)</span></span>
<span><span class="va">treat_sig</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span>, </span>
<span>  <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">fit</span><span class="op">[[</span><span class="va">x</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">coefficients</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="va">.</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span>,<span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">Estimate</span>, pvalue <span class="op">=</span> <span class="va">`Pr(&gt;|t|)`</span> <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"Treatment"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>      <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>        gs_name <span class="op">=</span> <span class="va">x</span>, </span>
<span>        FDR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">pvalue</span>, <span class="st">"fdr"</span><span class="op">)</span>, </span>
<span>        Treatment <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_remove.html" class="external-link">str_remove_all</a></span><span class="op">(</span><span class="va">Treatment</span>, <span class="st">"treatment"</span><span class="op">)</span></span>
<span>      <span class="op">)</span> </span>
<span>  <span class="op">}</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<p>Results from the linear modelling revealed pathways that were on
average perturbed due to each treatment:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_at</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/vars.html" class="external-link">vars</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Treatment"</span>, <span class="st">"gs_name"</span><span class="op">)</span><span class="op">)</span>, <span class="va">as.factor</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate_all.html" class="external-link">mutate_if</a></span><span class="op">(</span><span class="va">is.numeric</span>, <span class="va">sprintf</span>, fmt <span class="op">=</span> <span class="st">'%#.4f'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Direction <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">Estimate</span> <span class="op">&lt;</span> <span class="fl">0</span>, <span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>        <span class="va">Treatment</span>, `Perturbation Score` <span class="op">=</span> <span class="va">Estimate</span>, <span class="va">Direction</span>,</span>
<span>        `Gene-set name` <span class="op">=</span> <span class="va">gs_name</span>, </span>
<span>        `P-value` <span class="op">=</span> <span class="va">pvalue</span>, </span>
<span>        <span class="va">FDR</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span></span>
<span>        filter <span class="op">=</span> <span class="st">"top"</span>, </span>
<span>        options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>            columnDefs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>targets <span class="op">=</span> <span class="st">"Direction"</span>, visible <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span>        <span class="op">)</span>, </span>
<span>        caption <span class="op">=</span> <span class="fu">htmltools</span><span class="fu">::</span><span class="va"><a href="https://rstudio.github.io/htmltools/reference/builder.html" class="external-link">tags</a></span><span class="op">$</span><span class="fu">caption</span><span class="op">(</span></span>
<span>                  <span class="fu">htmltools</span><span class="fu">::</span><span class="fu"><a href="https://rstudio.github.io/htmltools/reference/builder.html" class="external-link">em</a></span><span class="op">(</span></span>
<span>                      <span class="st">"Pathways that were significant perturbed within each treatment group."</span><span class="op">)</span></span>
<span>              <span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/formatCurrency.html" class="external-link">formatStyle</a></span><span class="op">(</span></span>
<span>        <span class="st">'Perturbation Score'</span>, <span class="st">'Direction'</span>,</span>
<span>        color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/DT/man/styleInterval.html" class="external-link">styleEqual</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Inhibited"</span>, <span class="st">"Activation"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-786fcedd189b0f3ed645" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-786fcedd189b0f3ed645">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;E2+R5020&quot;,&quot;R5020&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;kegg.Aldosterone synthesis and secretion&quot;,&quot;kegg.Antigen processing and presentation&quot;,&quot;kegg.Cellular senescence&quot;,&quot;kegg.Cholinergic synapse&quot;,&quot;kegg.ECM-receptor interaction&quot;,&quot;kegg.Estrogen signaling pathway&quot;,&quot;kegg.Focal adhesion&quot;,&quot;kegg.Human papillomavirus infection&quot;,&quot;kegg.Malaria&quot;,&quot;kegg.Melanoma&quot;,&quot;kegg.Mineral absorption&quot;,&quot;kegg.Neutrophil extracellular trap formation&quot;,&quot;kegg.Ovarian steroidogenesis&quot;,&quot;kegg.PI3K-Akt signaling pathway&quot;,&quot;kegg.Prostate cancer&quot;,&quot;kegg.Renin secretion&quot;,&quot;kegg.Small cell lung cancer&quot;,&quot;kegg.Staphylococcus aureus infection&quot;,&quot;kegg.Tight junction&quot;,&quot;kegg.Toxoplasmosis&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","caption":"<caption>\n  <em>Pathways that were significant perturbed within each treatment group.<\/em>\n<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35"],["E2+R5020","R5020","E2+R5020","R5020","E2+R5020","E2+R5020","R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","R5020","E2+R5020","E2+R5020","R5020"],["0.9969","1.4583","-0.7242","-0.6766","-0.8466","-1.1493","-0.7256","1.4955","1.4649","1.6074","1.6250","1.9078","1.2672","1.7030","0.9490","1.1622","1.2220","1.5219","2.0326","2.0493","-0.7208","-0.7154","1.2055","1.0689","2.0848","2.1275","1.3158","1.4883","-1.3585","1.1037","0.5258","0.4949","-1.2176","-1.1206","-1.0147"],["Activation","Activation","Inhibited","Inhibited","Inhibited","Inhibited","Inhibited","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Activation","Inhibited","Inhibited","Activation","Activation","Activation","Activation","Activation","Activation","Inhibited","Activation","Activation","Activation","Inhibited","Inhibited","Inhibited"],["kegg.Aldosterone synthesis and secretion","kegg.Aldosterone synthesis and secretion","kegg.Antigen processing and presentation","kegg.Antigen processing and presentation","kegg.Cellular senescence","kegg.Cholinergic synapse","kegg.Cholinergic synapse","kegg.ECM-receptor interaction","kegg.Estrogen signaling pathway","kegg.Estrogen signaling pathway","kegg.Focal adhesion","kegg.Focal adhesion","kegg.Human papillomavirus infection","kegg.Human papillomavirus infection","kegg.Malaria","kegg.Malaria","kegg.Melanoma","kegg.Melanoma","kegg.Mineral absorption","kegg.Mineral absorption","kegg.Neutrophil extracellular trap formation","kegg.Neutrophil extracellular trap formation","kegg.Ovarian steroidogenesis","kegg.Ovarian steroidogenesis","kegg.PI3K-Akt signaling pathway","kegg.PI3K-Akt signaling pathway","kegg.Prostate cancer","kegg.Prostate cancer","kegg.Renin secretion","kegg.Small cell lung cancer","kegg.Staphylococcus aureus infection","kegg.Staphylococcus aureus infection","kegg.Tight junction","kegg.Toxoplasmosis","kegg.Toxoplasmosis"],["0.0128","0.0018","0.0377","0.0483","0.0156","0.0011","0.0120","0.0131","0.0014","0.0008","0.0272","0.0138","0.0233","0.0060","0.0250","0.0103","0.0085","0.0028","0.0017","0.0017","0.0285","0.0294","0.0259","0.0412","0.0351","0.0325","0.0310","0.0187","0.0098","0.0165","0.0342","0.0428","0.0056","0.0150","0.0230"],["0.0128","0.0037","0.0483","0.0483","0.0313","0.0022","0.0120","0.0263","0.0014","0.0014","0.0272","0.0272","0.0233","0.0120","0.0250","0.0206","0.0085","0.0055","0.0017","0.0017","0.0294","0.0294","0.0412","0.0412","0.0351","0.0351","0.0310","0.0310","0.0195","0.0330","0.0428","0.0428","0.0112","0.0230","0.0230"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Treatment<\/th>\n      <th>Perturbation Score<\/th>\n      <th>Direction<\/th>\n      <th>Gene-set name<\/th>\n      <th>P-value<\/th>\n      <th>FDR<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":3,"visible":false},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[3]; $(this.api().cell(row, 2).node()).css({'color':value == \"Inhibited\" ? \"blue\" : value == \"Activation\" ? \"red\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script><p>It is not surprising to see that the estrogen signalling pathway was
significantly activated under both R5020 (progesterone) and E2
(estrogen) +R5020 treatments.</p>
<div class="section level5">
<h5 id="visualise-genes-contributions-to-pathway-perturbation">Visualise genes’ contributions to pathway perturbation<a class="anchor" aria-label="anchor" href="#visualise-genes-contributions-to-pathway-perturbation"></a>
</h5>
<p>If there’s a specific pathway that we would like to dig deeper into
and explore which pathway genes played a key role in its perturbation,
for example, activation of the “Ovarian steroidogenesis”, we can plot
the gene-level perturbation scores of genes that are constantly highly
perturbed or highly variable in that pathway as a heatmap using the
<code><a href="../reference/plot_gene_contribution.html">plot_gene_contribution()</a></code> function.</p>
<p>From the heatmap below that we can see that the activation of this
pathway was strongly driven by two genes: ENTREZID:3630 and
ENTREZID:3479.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_gene_contribution.html">plot_gene_contribution</a></span><span class="op">(</span></span>
<span>    genePertMatr  <span class="op">=</span> <span class="va">genePertScore</span><span class="op">$</span><span class="va">`kegg.Ovarian steroidogenesis`</span>,</span>
<span>    filterBy <span class="op">=</span> <span class="st">"mean"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-4-1.png" alt="*Gene-level perturbation scores of genes with top 15 highest absolute mean gene-wise perturbation scores in the Ovarian steroidogenesis gene-set.*" width="768"><p class="caption">
<em>Gene-level perturbation scores of genes with top 15 highest absolute
mean gene-wise perturbation scores in the Ovarian steroidogenesis
gene-set.</em>
</p>
</div>
<p>By default, genes’ entrez IDs are used and plotted as row names,
which may not be very informative. So the row names could be overwritten
by providing a <code>data.frame</code> mapping entrez IDs to other
identifiers through the <code>mapRownameTo</code> parameter.</p>
<p>Mapping between different gene identifiers could be achieved through
the <code>mapIDs()</code> function from the Bioconductor package <a href="https://bioconductor.org/packages/release/bioc/html/AnnotationDbi.html" class="external-link"><code>AnnotationDbi</code></a>.
But to reduce the compiling time of this vignette, mappings between
entrez IDs and gene names as available in Ensembl Release 101 have been
provided as a <code>data.frame</code> called
<code>entrez2name</code>.</p>
<p>Note that if the mapping information was provided and the mapping was
only successful for some genes but not the others, only genes that have
been mapped successfully will be plotted.</p>
<p>Since <code><a href="../reference/plot_gene_contribution.html">plot_gene_contribution()</a></code> is built on
<code>pheatmap</code>, which provides a practical column annotation
feature, the <code><a href="../reference/plot_gene_contribution.html">plot_gene_contribution()</a></code> function also allow a
<code>data.frame</code> storing annotation information to be provided.
We can annotate each column (ie. each sample) by the direction of
pathway perturbation in that sample or any other sample metadata, such
as their treatments.</p>
<p>In this example, we first create a <code>data.frame</code> storing
the pathway perturbation scores of the “Ovarian steroidogenesis” pathway
in each sample and their treatments.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annotation_df</span> <span class="op">&lt;-</span> <span class="va">pathwayPertScore</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">gs_name</span> <span class="op">==</span> <span class="st">"kegg.Ovarian steroidogenesis"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>        `Pathway-level` <span class="op">=</span>  <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span></span>
<span>            <span class="va">score</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">"Activated"</span>, <span class="st">"Inhibited"</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sample</span>, <span class="va">`Pathway-level`</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>        <span class="va">.</span>,  <span class="va">metadata_example</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>            <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sample</span>, Treatment <span class="op">=</span> <span class="va">treatment</span><span class="op">)</span>, </span>
<span>        unmatched <span class="op">=</span> <span class="st">"drop"</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>The annotation <code>data.frame</code> was provided to the
<code><a href="../reference/plot_gene_contribution.html">plot_gene_contribution()</a></code> function through the
<code>annotation_df</code> parameter. Colors of the annotation could be
customised through <code><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap::pheatmap()</a></code>’s
<code>annotation_colors</code> parameter.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"entrez2name.rda"</span>, package <span class="op">=</span> <span class="st">"sSNAPPY"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_gene_contribution.html">plot_gene_contribution</a></span><span class="op">(</span></span>
<span>    genePertMatr  <span class="op">=</span> <span class="va">genePertScore</span><span class="op">$</span><span class="va">`kegg.Ovarian steroidogenesis`</span>,</span>
<span>    annotation_df <span class="op">=</span> <span class="va">annotation_df</span>,</span>
<span>    filterBy <span class="op">=</span> <span class="st">"mean"</span>, </span>
<span>    mapEntrezID <span class="op">=</span> <span class="va">entrez2name</span>,</span>
<span>    annotation_colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        Treatment <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>R5020 <span class="op">=</span> <span class="st">"black"</span>, `E2+R5020` <span class="op">=</span> <span class="st">"white"</span><span class="op">)</span>,</span>
<span>        `Pathway-level` <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Activated <span class="op">=</span> <span class="st">"darkred"</span>, Inhibited <span class="op">=</span> <span class="st">"lightskyblue"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-5-1.png" alt="*Gene-level perturbation scores of genes with top 15 absolute mean gene-wise perturbation scores in the Ovarian steroidogenesis gene-set.Genes playing the most important roles in pathway activation was INS, CGA, GNAS.*" width="768"><p class="caption">
<em>Gene-level perturbation scores of genes with top 15 absolute mean
gene-wise perturbation scores in the Ovarian steroidogenesis
gene-set.Genes playing the most important roles in pathway activation
was INS, CGA, GNAS.</em>
</p>
</div>
</div>
<div class="section level5">
<h5 id="visualise-overlap-between-gene-sets-as-networks">Visualise overlap between gene-sets as networks<a class="anchor" aria-label="anchor" href="#visualise-overlap-between-gene-sets-as-networks"></a>
</h5>
<p>Visualising significantly perturbed biological pathways as a network,
where edges between gene-sets reflect how much overlap they share, is an
useful approach for demonstrating the connections between biological
processes. The <code><a href="../reference/plot_gs_network.html">plot_gs_network()</a></code> function in this package
allows an easy construction of such network by taking the
<code><a href="../reference/normalise_by_permu.html">normalise_by_permu()</a></code> function’s output as direct input and
allowing flexible customisation.</p>
<p>Nodes in the network plots could be colored by the predicted
direction of perturbation (i.e.. sign of robust z-score). Results of
group-level perturbation can also be visualised using the
<code><a href="../reference/plot_gs_network.html">plot_gs_network()</a></code> function.</p>
<p>The function allows you to customize the layout, colour, edge
transparency and other aesthetics of the graph. More information can be
found on the help page (<code><a href="../reference/plot_gs_network.html">?plot_gs_network</a></code>). The output of
the graph is a <code>ggplot</code> object and the theme of it can be
changed just as any other <code>ggplot</code> figures.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">Estimate</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">"Activated"</span>, <span class="st">"Inhibited"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/plot_gs_network.html">plot_gs_network</a></span><span class="op">(</span></span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorBy <span class="op">=</span> <span class="st">"status"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span> </span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">Estimate</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">"Activated"</span>, <span class="st">"Inhibited"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/plot_gs_network.html">plot_gs_network</a></span><span class="op">(</span></span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorBy <span class="op">=</span> <span class="st">"status"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span></span>
<span>        panel.grid <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>By default, <code><a href="../reference/plot_gs_network.html">plot_gs_network()</a></code> function does not include
nodes that are not connected to any other nodes, which could be
overwritten by setting the <code>plotIsolated</code> parameter to
TURE.</p>
</div>
<div class="section level5">
<h5 id="visualise-community-structure-in-the-gene-set-network">Visualise community structure in the gene-set network<a class="anchor" aria-label="anchor" href="#visualise-community-structure-in-the-gene-set-network"></a>
</h5>
<p>When a large number of pathways were perturbed, it is hard to answer
the question “What key biological processes were perturbed?” solely by
looking at all the pathway names. To solve that, we can use the
<code><a href="../reference/plot_community.html">plot_community()</a></code> function to apply a community detection
algorithm to the network we had above, and annotate each community by
the biological process that most pathways assigned to that community
belong to.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/plot_community.html">plot_community</a></span><span class="op">(</span></span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorBy <span class="op">=</span> <span class="st">"community"</span>, </span>
<span>        color_lg_title <span class="op">=</span> <span class="st">"Community"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-7-1.png" alt="*Pathways significantly perturbed by the E2+R5020 combintation treatment, annotated by the main biological processes each pathways belong to.*" width="1536"><p class="caption">
<em>Pathways significantly perturbed by the E2+R5020 combintation
treatment, annotated by the main biological processes each pathways
belong to.</em>
</p>
</div>
<p><code>sSNAPPY</code> was built in with categorisations of
<em>KEGG</em> pathways. But to annotate pathways retrieved from other
databases, customized annotation <code>data.frame</code> could be
provided through the <code>gsAnnotation</code> parameter.</p>
</div>
<div class="section level5">
<h5 id="visualise-genes-included-in-perturbed-pathways-networks">Visualise genes included in perturbed pathways networks<a class="anchor" aria-label="anchor" href="#visualise-genes-included-in-perturbed-pathways-networks"></a>
</h5>
<p>If we want to not only know if two pathways are connected but also
the genes connecting those pathways, we can use the
<code><a href="../reference/plot_gs2gene.html">plot_gs2gene()</a></code> function instead:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/plot_gs2gene.html">plot_gs2gene</a></span><span class="op">(</span></span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorGsBy <span class="op">=</span> <span class="st">"Estimate"</span>, </span>
<span>        labelGene <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>        geneNodeSize  <span class="op">=</span> <span class="fl">1</span>, </span>
<span>        edgeAlpha <span class="op">=</span> <span class="fl">0.1</span>, </span>
<span>        gsNameSize <span class="op">=</span> <span class="fl">2</span>,</span>
<span>        filterGeneBy <span class="op">=</span> <span class="fl">3</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradient2</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-8-1.png" alt="*Pathways significantly perturbed by the E2+R5020 combintation treatment and genes contained in at least 3 of those pathways.*" width="1152"><p class="caption">
<em>Pathways significantly perturbed by the E2+R5020 combintation
treatment and genes contained in at least 3 of those pathways.</em>
</p>
</div>
<p>However, since there are a large number of genes in each pathway, the
plot above was quite messy and it was unrealistic to plot all genes’
names. So it is recommend to filter pathway genes by their perturbation
scores or logFCs.</p>
<p>For example, we can rank genes by the absolute values of their mean
single-sample logFCs and only focus on genes that were ranked in the top
200 of the list.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meanFC</span> <span class="op">&lt;-</span> <span class="va">weightedFC</span><span class="op">$</span><span class="va">weighted_logFC</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">.</span><span class="op">[</span>, <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html" class="external-link">str_detect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, <span class="st">"E2"</span>, negate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">mean</span> <span class="op">)</span></span>
<span><span class="va">top200_gene</span> <span class="op">&lt;-</span> <span class="va">meanFC</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span>decreasing <span class="op">=</span> <span class="cn">TRUE</span>, <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">.</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">200</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">top200_FC</span> <span class="op">&lt;-</span> <span class="va">meanFC</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">.</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">top200_gene</span><span class="op">]</span></span>
<span><span class="va">top200_FC</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">top200_FC</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">"Up-Regulated"</span>, <span class="st">"Down-Regulated"</span><span class="op">)</span></span></code></pre></div>
<p>When we provide genes’ logFCs as a named vector through the
<code>geneFC</code> parameter, only pathway genes with logFCs provided
will be plotted and gene nodes will be colored by genes’ directions of
changes. The names of the logFC vector must be genes’ entrez IDs in the
format of “ENTREZID:XXXX”, as pathway topology matrices retrieved
through <code><a href="../reference/retrieve_topology.html">retrieve_topology()</a></code> always use entrez IDs as
identifiers.</p>
<p>However, it is not going to be informative to label genes with their
entrez IDs. So the same as the <code><a href="../reference/plot_gene_contribution.html">plot_gene_contribution()</a></code>
function, we can provide a <code>data.frame</code> to match genes’
entrez IDs to our chosen identifiers through the
<code>mapEntrezID</code> parameter in the <code><a href="../reference/plot_gs2gene.html">plot_gs2gene()</a></code>
function too.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">treat_sig</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">FDR</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">Treatment</span> <span class="op">==</span> <span class="st">"E2+R5020"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">Estimate</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="st">"Activated"</span>, <span class="st">"Inhibited"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/plot_gs2gene.html">plot_gs2gene</a></span><span class="op">(</span></span>
<span>        gsTopology <span class="op">=</span> <span class="va">gsTopology</span>, </span>
<span>        colorGsBy <span class="op">=</span> <span class="st">"status"</span>, </span>
<span>        geneFC <span class="op">=</span> <span class="va">top200_FC</span>, </span>
<span>        mapEntrezID <span class="op">=</span> <span class="va">entrez2name</span>, </span>
<span>        gsNameSize <span class="op">=</span> <span class="fl">3</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkred"</span>, <span class="st">"lightskyblue"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"blue"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="sSNAPPY_files/figure-html/unnamed-chunk-9-1.png" alt="*Pathways significantly perturbed by the E2+R5020 combintation treatment, and pathway genes with top 200 magnitudes of changes among all E2+R5020-treated samples. Both pathways and genes were colored by their directions of changes.*" width="960"><p class="caption">
<em>Pathways significantly perturbed by the E2+R5020 combintation
treatment, and pathway genes with top 200 magnitudes of changes among
all E2+R5020-treated samples. Both pathways and genes were colored by
their directions of changes.</em>
</p>
</div>
<p>We can also filters genes’ by their contributions to a pathway
perturbations.To help with that, <code>sSNAPPY</code> provides an option
to rank genes’ perturbation scores within each sample for each
pathway.</p>
<p>If in a given pathway, both positive and negative gene-wise
perturbation scores exist, positive and negative scores are ranked
separately, where the larger a positive rank, the more the gene
contributed to the pathway’s activation, and the smaller a negative
rank, the more the gene contributed to the pathways’ inhibition.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneRank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rank_gene_pert.html">rank_gene_pert</a></span><span class="op">(</span><span class="va">genePertScore</span>, <span class="va">gsTopology</span><span class="op">)</span></span></code></pre></div>
<p>Depending on the biological question, it could be interesting to plot
only the pathway genes with the same directions of changes as the
pathway they belonged to, and ignoring those genes that were
antagonizing the pathway perturbation.</p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Sales G, Calura E, Cavalieri D, Romualdi C (2012). “graphite - a
Bioconductor package to convert pathway topology to gene network.” BMC
Bioinformatics. <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-13-20" class="external-link uri">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-13-20</a>.</li>
<li>Tarca, Adi Laurentiu et al. (2009). “A novel signaling pathway
impact analysis.” Bioinformatics vol. 25,1 : 75-82. <a href="doi:10.1093/bioinformatics/btn577" class="uri">doi:10.1093/bioinformatics/btn577</a>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] graphite_1.46.0  htmltools_0.5.5  DT_0.28          cowplot_1.1.1   </span></span>
<span><span class="co">##  [5] magrittr_2.0.3   lubridate_1.9.2  forcats_1.0.0    stringr_1.5.0   </span></span>
<span><span class="co">##  [9] dplyr_1.1.2      purrr_1.0.1      readr_2.1.4      tidyr_1.3.0     </span></span>
<span><span class="co">## [13] tibble_3.2.1     tidyverse_2.0.0  sSNAPPY_1.5.1    ggplot2_3.4.2   </span></span>
<span><span class="co">## [17] BiocStyle_2.28.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] DBI_1.1.3                   bitops_1.0-7               </span></span>
<span><span class="co">##   [3] gridExtra_2.3               rlang_1.1.1                </span></span>
<span><span class="co">##   [5] matrixStats_1.0.0           compiler_4.3.1             </span></span>
<span><span class="co">##   [7] RSQLite_2.3.1               mgcv_1.8-42                </span></span>
<span><span class="co">##   [9] png_0.1-8                   systemfonts_1.0.4          </span></span>
<span><span class="co">##  [11] vctrs_0.6.3                 reshape2_1.4.4             </span></span>
<span><span class="co">##  [13] pkgconfig_2.0.3             crayon_1.5.2               </span></span>
<span><span class="co">##  [15] fastmap_1.1.1               ellipsis_0.3.2             </span></span>
<span><span class="co">##  [17] XVector_0.40.0              labeling_0.4.2             </span></span>
<span><span class="co">##  [19] ggraph_2.1.0                pander_0.6.5               </span></span>
<span><span class="co">##  [21] utf8_1.2.3                  rmarkdown_2.23             </span></span>
<span><span class="co">##  [23] tzdb_0.4.0                  graph_1.78.0               </span></span>
<span><span class="co">##  [25] ragg_1.2.5                  bit_4.0.5                  </span></span>
<span><span class="co">##  [27] xfun_0.39                   zlibbioc_1.46.0            </span></span>
<span><span class="co">##  [29] cachem_1.0.8                GenomeInfoDb_1.36.1        </span></span>
<span><span class="co">##  [31] jsonlite_1.8.7              blob_1.2.4                 </span></span>
<span><span class="co">##  [33] highr_0.10                  DelayedArray_0.26.6        </span></span>
<span><span class="co">##  [35] tweenr_2.0.2                R6_2.5.1                   </span></span>
<span><span class="co">##  [37] bslib_0.5.0                 stringi_1.7.12             </span></span>
<span><span class="co">##  [39] RColorBrewer_1.1-3          limma_3.56.2               </span></span>
<span><span class="co">##  [41] GenomicRanges_1.52.0        jquerylib_0.1.4            </span></span>
<span><span class="co">##  [43] Rcpp_1.0.11                 bookdown_0.34              </span></span>
<span><span class="co">##  [45] SummarizedExperiment_1.30.2 knitr_1.43                 </span></span>
<span><span class="co">##  [47] IRanges_2.34.1              splines_4.3.1              </span></span>
<span><span class="co">##  [49] timechange_0.2.0            Matrix_1.5-4.1             </span></span>
<span><span class="co">##  [51] igraph_1.5.0.1              tidyselect_1.2.0           </span></span>
<span><span class="co">##  [53] abind_1.4-5                 yaml_2.3.7                 </span></span>
<span><span class="co">##  [55] viridis_0.6.4               curl_5.0.1                 </span></span>
<span><span class="co">##  [57] plyr_1.8.8                  lattice_0.21-8             </span></span>
<span><span class="co">##  [59] Biobase_2.60.0              withr_2.5.0                </span></span>
<span><span class="co">##  [61] KEGGREST_1.40.0             evaluate_0.21              </span></span>
<span><span class="co">##  [63] desc_1.4.2                  polyclip_1.10-4            </span></span>
<span><span class="co">##  [65] Biostrings_2.68.1           pillar_1.9.0               </span></span>
<span><span class="co">##  [67] BiocManager_1.30.21.1       MatrixGenerics_1.12.2      </span></span>
<span><span class="co">##  [69] stats4_4.3.1                generics_0.1.3             </span></span>
<span><span class="co">##  [71] rprojroot_2.0.3             RCurl_1.98-1.12            </span></span>
<span><span class="co">##  [73] hms_1.1.3                   S4Vectors_0.38.1           </span></span>
<span><span class="co">##  [75] munsell_0.5.0               scales_1.2.1               </span></span>
<span><span class="co">##  [77] gtools_3.9.4                glue_1.6.2                 </span></span>
<span><span class="co">##  [79] pheatmap_1.0.12             tools_4.3.1                </span></span>
<span><span class="co">##  [81] locfit_1.5-9.8              fs_1.6.3                   </span></span>
<span><span class="co">##  [83] graphlayouts_1.0.0          tidygraph_1.2.3            </span></span>
<span><span class="co">##  [85] grid_4.3.1                  crosstalk_1.2.0            </span></span>
<span><span class="co">##  [87] AnnotationDbi_1.62.2        edgeR_3.42.4               </span></span>
<span><span class="co">##  [89] colorspace_2.1-0            nlme_3.1-162               </span></span>
<span><span class="co">##  [91] GenomeInfoDbData_1.2.10     ggforce_0.4.1              </span></span>
<span><span class="co">##  [93] cli_3.6.1                   rappdirs_0.3.3             </span></span>
<span><span class="co">##  [95] textshaping_0.3.6           fansi_1.0.4                </span></span>
<span><span class="co">##  [97] S4Arrays_1.0.5              viridisLite_0.4.2          </span></span>
<span><span class="co">##  [99] gtable_0.3.3                sass_0.4.7                 </span></span>
<span><span class="co">## [101] digest_0.6.33               BiocGenerics_0.46.0        </span></span>
<span><span class="co">## [103] ggrepel_0.9.3               htmlwidgets_1.6.2          </span></span>
<span><span class="co">## [105] org.Hs.eg.db_3.17.0         farver_2.1.1               </span></span>
<span><span class="co">## [107] memoise_2.0.1               pkgdown_2.0.7              </span></span>
<span><span class="co">## [109] lifecycle_1.0.3             httr_1.4.6                 </span></span>
<span><span class="co">## [111] bit64_4.0.5                 MASS_7.3-60</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://github.com/Wenjun-Liu" class="external-link">Wenjun Liu</a>, Stephen Pederson.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
